{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# This notebook is to collect Data for assessing whether time since TB infection predictor generalizes to Adolescent Acquisition study, or whether current positive predictions are from distribution mismatch between RNA-seq and microarray"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Load Libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "library(MetaIntegrator)\n",
    "library(GEOquery)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Pull the TB datasets together"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Functions to help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "gses.array = c(\n",
    "\"GSE19491\",\n",
    "\n",
    "\n",
    "\"GSE37250\",\n",
    "\"GSE39939\",\n",
    "\"GSE39940\",\n",
    "\n",
    "'GSE42825', 'GSE42826' ,'GSE42827', 'GSE42830', 'GSE42831', 'GSE42832'\n",
    ")\n",
    "\n",
    "\n",
    "test.gses = gses.array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "make.total.dataset = function(data.dir, gses) {\n",
    "    pheno.list = list()\n",
    "    exprs.list = list()\n",
    "    for (GSE in gses) {\n",
    "        gse.dir = paste(data.dir, GSE, sep=\"/\")\n",
    "        gene_file = paste(gse.dir, \"/\", GSE, \"_genelevel.csv\", sep=\"\")\n",
    "        pheno_file = paste(gse.dir, \"/\",  GSE, \"_pheno.csv\", sep=\"\")\n",
    "        \n",
    "        exprs = read.csv(gene_file, row.names=1, header=T)\n",
    "        pheno = read.csv(pheno_file, row.names=1, header=T)\n",
    "        \n",
    "        \n",
    "        exprs.list[[GSE]] = exprs\n",
    "        pheno.list[[GSE]] = pheno\n",
    "    }\n",
    "    \n",
    "    results = list()\n",
    "    results$exprs = exprs.list\n",
    "    results$pheno.list = pheno.list\n",
    "    return(results)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "data.dir = \"/master/rault/bitbucket/deep_tb/GEO_Warsinske_TB/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Read in the GSEs data for test set (validation of recent exposure predictor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data = make.total.dataset(data.dir, test.gses)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "List of 2\n",
      " $ exprs     :List of 10\n",
      "  ..$ GSE19491:'data.frame':\t15836 obs. of  498 variables:\n",
      "  .. ..$ GSM484368: num [1:15836] 14.29 9.51 6.16 7.9 11.51 ...\n",
      "  .. ..$ GSM484369: num [1:15836] 14.12 9.74 6.1 7.8 11.02 ...\n",
      "  .. ..$ GSM484370: num [1:15836] 13.81 9.59 6.23 7.85 11.43 ...\n",
      "  .. ..$ GSM484371: num [1:15836] 13.94 9.68 6.3 7.82 11.59 ...\n",
      "  .. ..$ GSM484372: num [1:15836] 14.08 9.47 6.02 7.8 11.51 ...\n",
      "  .. ..$ GSM484373: num [1:15836] 14.24 9.7 6.05 8.11 11.41 ...\n",
      "  .. ..$ GSM484374: num [1:15836] 14.3 9.49 6.04 7.65 11.71 ...\n",
      "  .. ..$ GSM484375: num [1:15836] 14.32 9.5 5.95 7.58 11.85 ...\n",
      "  .. ..$ GSM484376: num [1:15836] 13.96 9.22 5.92 7.81 11.33 ...\n",
      "  .. ..$ GSM484377: num [1:15836] 14.25 9.25 6.02 7.82 11.8 ...\n",
      "  .. ..$ GSM484378: num [1:15836] 14.11 9.99 5.78 7.82 11.88 ...\n",
      "  .. ..$ GSM484379: num [1:15836] 13.96 9.37 6.21 7.66 11.57 ...\n",
      "  .. ..$ GSM484380: num [1:15836] 13.76 10.16 6.06 7.88 11.32 ...\n",
      "  .. ..$ GSM484381: num [1:15836] 13.65 9.92 5.93 7.94 11.85 ...\n",
      "  .. ..$ GSM484382: num [1:15836] 14.03 9.62 6.01 7.95 12.04 ...\n",
      "  .. ..$ GSM484383: num [1:15836] 13.68 10.46 6.08 7.79 11.61 ...\n",
      "  .. ..$ GSM484384: num [1:15836] 13.64 9.45 6.23 7.92 11.55 ...\n",
      "  .. ..$ GSM484385: num [1:15836] 13.97 9.84 5.95 7.85 11.92 ...\n",
      "  .. ..$ GSM484386: num [1:15836] 14.08 9.84 5.81 7.72 12.2 ...\n",
      "  .. ..$ GSM484387: num [1:15836] 13.91 10.47 6.18 7.73 11.8 ...\n",
      "  .. ..$ GSM484388: num [1:15836] 13.67 9.85 6 7.66 11.48 ...\n",
      "  .. ..$ GSM484389: num [1:15836] 13.84 9.59 6.06 7.72 11.93 ...\n",
      "  .. ..$ GSM484390: num [1:15836] 13.67 9.73 5.89 7.88 11.55 ...\n",
      "  .. ..$ GSM484391: num [1:15836] 13.51 9.54 6.24 7.76 11.76 ...\n",
      "  .. ..$ GSM484392: num [1:15836] 13.97 9.64 6.1 7.81 11.78 ...\n",
      "  .. ..$ GSM484393: num [1:15836] 13.86 9.63 6 7.77 11.77 ...\n",
      "  .. ..$ GSM484394: num [1:15836] 13.76 9.89 5.93 7.75 12.21 ...\n",
      "  .. ..$ GSM484395: num [1:15836] 13.23 10.03 6.03 7.73 11.4 ...\n",
      "  .. ..$ GSM484396: num [1:15836] 13.37 9.47 6.16 7.88 11.54 ...\n",
      "  .. ..$ GSM484397: num [1:15836] 13.91 9.36 6.12 8.06 12.01 ...\n",
      "  .. ..$ GSM484398: num [1:15836] 13.77 10.16 6.05 7.85 11.49 ...\n",
      "  .. ..$ GSM484399: num [1:15836] 13.72 9.8 6.04 7.85 11.39 ...\n",
      "  .. ..$ GSM484400: num [1:15836] 14.02 9.66 6.1 7.87 12.14 ...\n",
      "  .. ..$ GSM484448: num [1:15836] 14.23 9.32 6.28 7.97 10.69 ...\n",
      "  .. ..$ GSM484449: num [1:15836] 13.65 10.04 5.99 8.13 9.6 ...\n",
      "  .. ..$ GSM484450: num [1:15836] 14.47 9.15 5.96 7.84 10.58 ...\n",
      "  .. ..$ GSM484451: num [1:15836] 13.99 9.36 6.03 7.78 9.6 ...\n",
      "  .. ..$ GSM484452: num [1:15836] 14.27 9.11 5.95 7.95 10.17 ...\n",
      "  .. ..$ GSM484453: num [1:15836] 14.34 9.56 6.1 7.85 10.66 ...\n",
      "  .. ..$ GSM484454: num [1:15836] 13.94 9.69 6.15 8.03 10.42 ...\n",
      "  .. ..$ GSM484455: num [1:15836] 14.36 9.27 6.21 7.86 11.02 ...\n",
      "  .. ..$ GSM484456: num [1:15836] 14.05 9.88 6.15 7.96 9.92 ...\n",
      "  .. ..$ GSM484457: num [1:15836] 14.53 9.41 6.03 7.67 10.77 ...\n",
      "  .. ..$ GSM484458: num [1:15836] 14.54 9.18 6.15 7.57 9.48 ...\n",
      "  .. ..$ GSM484459: num [1:15836] 14.13 9.35 6.05 7.96 9.18 ...\n",
      "  .. ..$ GSM484460: num [1:15836] 14.91 8.97 5.94 8 10.42 ...\n",
      "  .. ..$ GSM484461: num [1:15836] 14.6 9.23 6.16 8.01 10.75 ...\n",
      "  .. ..$ GSM484462: num [1:15836] 14.72 8.82 6.12 7.87 9.67 ...\n",
      "  .. ..$ GSM484463: num [1:15836] 14.32 8.72 6.18 7.91 9.32 ...\n",
      "  .. ..$ GSM484464: num [1:15836] 14.36 8.74 6.11 8.02 10.7 ...\n",
      "  .. ..$ GSM484465: num [1:15836] 14.92 9.19 5.68 7.54 10.26 ...\n",
      "  .. ..$ GSM484466: num [1:15836] 14.24 9.13 6.21 7.96 10.33 ...\n",
      "  .. ..$ GSM484467: num [1:15836] 14.74 9 5.97 7.72 10.54 ...\n",
      "  .. ..$ GSM484468: num [1:15836] 14.23 9.09 6.06 7.94 10.53 ...\n",
      "  .. ..$ GSM484469: num [1:15836] 14.4 9.37 6.29 8.06 10.47 ...\n",
      "  .. ..$ GSM484470: num [1:15836] 13.9 9.49 6.25 7.98 10.89 ...\n",
      "  .. ..$ GSM484471: num [1:15836] 14.17 9.18 6.09 8.01 10.61 ...\n",
      "  .. ..$ GSM484472: num [1:15836] 13.7 9.47 6.1 7.88 9.54 ...\n",
      "  .. ..$ GSM484473: num [1:15836] 14.45 9.41 6.05 7.95 10.86 ...\n",
      "  .. ..$ GSM484474: num [1:15836] 14.09 9.52 5.82 7.73 9.74 ...\n",
      "  .. ..$ GSM484475: num [1:15836] 14.26 9.9 6.2 7.75 10.11 ...\n",
      "  .. ..$ GSM484476: num [1:15836] 14.31 9.04 6.26 8.05 10.88 ...\n",
      "  .. ..$ GSM484477: num [1:15836] 13.96 9.12 6.06 7.9 10.53 ...\n",
      "  .. ..$ GSM484478: num [1:15836] 14.34 9.09 6.11 7.9 10.65 ...\n",
      "  .. ..$ GSM484479: num [1:15836] 14.41 9.11 6.19 7.64 10.96 ...\n",
      "  .. ..$ GSM484480: num [1:15836] 14.45 9.14 6.04 7.65 11.11 ...\n",
      "  .. ..$ GSM484481: num [1:15836] 14.46 9.29 6.45 7.7 11.11 ...\n",
      "  .. ..$ GSM484482: num [1:15836] 14.38 9.33 6.16 7.98 10.91 ...\n",
      "  .. ..$ GSM484483: num [1:15836] 14.34 9.22 6.23 7.85 11.13 ...\n",
      "  .. ..$ GSM484484: num [1:15836] 14.33 9.23 5.89 7.94 11.33 ...\n",
      "  .. ..$ GSM484485: num [1:15836] 13.91 9.71 6.17 7.98 10.58 ...\n",
      "  .. ..$ GSM484486: num [1:15836] 14.18 9.06 5.83 7.77 10.45 ...\n",
      "  .. ..$ GSM484487: num [1:15836] 14.44 9.05 6.18 7.74 10.56 ...\n",
      "  .. ..$ GSM484488: num [1:15836] 14.46 9.09 5.95 7.87 11.16 ...\n",
      "  .. ..$ GSM484489: num [1:15836] 14.35 9.08 6.03 7.73 11.02 ...\n",
      "  .. ..$ GSM484500: num [1:15836] 14.23 10.65 6.18 8.43 10.96 ...\n",
      "  .. ..$ GSM484501: num [1:15836] 14.58 9.88 6.06 8.4 11.65 ...\n",
      "  .. ..$ GSM484502: num [1:15836] 14.4 10.41 6.14 8.39 11.46 ...\n",
      "  .. ..$ GSM484503: num [1:15836] 14.22 9.9 5.83 8.53 11.44 ...\n",
      "  .. ..$ GSM484504: num [1:15836] 14.98 9.8 6.15 8.38 12 ...\n",
      "  .. ..$ GSM484505: num [1:15836] 14.52 9.89 6.2 8.5 11.88 ...\n",
      "  .. ..$ GSM484506: num [1:15836] 14.59 9.49 5.94 8.39 11.73 ...\n",
      "  .. ..$ GSM484507: num [1:15836] 14.64 9.42 6.33 8.18 11.97 ...\n",
      "  .. ..$ GSM484508: num [1:15836] 14.6 10 6.28 8.52 12.34 ...\n",
      "  .. ..$ GSM484509: num [1:15836] 14.61 9.89 6.05 8.36 12.14 ...\n",
      "  .. ..$ GSM484510: num [1:15836] 14.55 9.78 6.04 8.21 12.04 ...\n",
      "  .. ..$ GSM484511: num [1:15836] 14.73 10.01 6.12 8.52 11.88 ...\n",
      "  .. ..$ GSM484512: num [1:15836] 14.66 9.94 6.28 8.57 11.83 ...\n",
      "  .. ..$ GSM484513: num [1:15836] 14.27 10.01 6.38 8.6 11.49 ...\n",
      "  .. ..$ GSM484514: num [1:15836] 14.79 10 6.09 8.45 12.05 ...\n",
      "  .. ..$ GSM484515: num [1:15836] 14.2 10.39 6.09 8.38 11.49 ...\n",
      "  .. ..$ GSM484516: num [1:15836] 14.04 10.47 6.07 8.36 11.6 ...\n",
      "  .. ..$ GSM484517: num [1:15836] 14.09 9.73 6.05 8.22 11.34 ...\n",
      "  .. ..$ GSM484518: num [1:15836] 14.83 9.94 6.41 8.32 12.42 ...\n",
      "  .. ..$ GSM484519: num [1:15836] 14.54 10.29 6.26 8.66 11.48 ...\n",
      "  .. ..$ GSM484520: num [1:15836] 15.09 9.54 6.19 8.35 11.8 ...\n",
      "  .. ..$ GSM484521: num [1:15836] 14.82 9.77 6.11 8.35 11.85 ...\n",
      "  .. ..$ GSM484522: num [1:15836] 14.81 9.79 5.85 8.27 11.76 ...\n",
      "  .. ..$ GSM484523: num [1:15836] 14.73 9.95 6.2 8.48 11.91 ...\n",
      "  .. .. [list output truncated]\n",
      "  ..$ GSE37250:'data.frame':\t17879 obs. of  537 variables:\n",
      "  .. ..$ GSM914353: num [1:17879] 13.82 13.05 6.91 9.5 14.03 ...\n",
      "  .. ..$ GSM914354: num [1:17879] 13.82 13.16 6.76 9.17 14.22 ...\n",
      "  .. ..$ GSM914355: num [1:17879] 13.95 13.55 6.91 9.42 14.37 ...\n",
      "  .. ..$ GSM914356: num [1:17879] 13.84 13.41 6.94 9.43 14.3 ...\n",
      "  .. ..$ GSM914357: num [1:17879] 13.73 13.15 6.82 9.36 14.23 ...\n",
      "  .. ..$ GSM914358: num [1:17879] 14.22 12.71 6.7 9.37 14.25 ...\n",
      "  .. ..$ GSM914359: num [1:17879] 14.19 12.57 6.43 9.46 14.4 ...\n",
      "  .. ..$ GSM914360: num [1:17879] 13.72 13.26 6.76 9.25 13.5 ...\n",
      "  .. ..$ GSM914361: num [1:17879] 13.15 13.35 6.76 9.39 14.37 ...\n",
      "  .. ..$ GSM914362: num [1:17879] 14.11 13.26 7.04 9.41 14.36 ...\n",
      "  .. ..$ GSM914363: num [1:17879] 13.31 13.13 6.19 9.3 14.43 ...\n",
      "  .. ..$ GSM914364: num [1:17879] 13.75 12.78 6.32 9.34 14.82 ...\n",
      "  .. ..$ GSM914365: num [1:17879] 13.05 12.85 6.51 9.31 14.43 ...\n",
      "  .. ..$ GSM914366: num [1:17879] 13.53 13 6.96 9.28 14.86 ...\n",
      "  .. ..$ GSM914367: num [1:17879] 13.81 13.07 7.09 9.31 13.83 ...\n",
      "  .. ..$ GSM914368: num [1:17879] 13.52 13.54 6.71 9.39 13.91 ...\n",
      "  .. ..$ GSM914369: num [1:17879] 13.34 13.24 6.82 9.78 14.15 ...\n",
      "  .. ..$ GSM914370: num [1:17879] 13.63 12.99 6.94 9.49 13.77 ...\n",
      "  .. ..$ GSM914371: num [1:17879] 13.33 13.47 6.67 9.13 14.24 ...\n",
      "  .. ..$ GSM914372: num [1:17879] 13.83 12.63 7.05 9.32 14.35 ...\n",
      "  .. ..$ GSM914373: num [1:17879] 13.44 12.84 7.37 9.45 13.89 ...\n",
      "  .. ..$ GSM914374: num [1:17879] 13.4 13.23 6.85 9.26 14.21 ...\n",
      "  .. ..$ GSM914375: num [1:17879] 13.87 12.81 7.22 9.37 14.15 ...\n",
      "  .. ..$ GSM914376: num [1:17879] 13.41 13.02 7.11 9.34 13.91 ...\n",
      "  .. ..$ GSM914377: num [1:17879] 13.78 13.12 7.25 9.39 13.86 ...\n",
      "  .. ..$ GSM914378: num [1:17879] 13.58 13.14 6.98 9.43 14.01 ...\n",
      "  .. ..$ GSM914379: num [1:17879] 14.11 12.88 6.92 9.63 13.83 ...\n",
      "  .. ..$ GSM914380: num [1:17879] 13.69 13.04 6.92 9.85 14.35 ...\n",
      "  .. ..$ GSM914381: num [1:17879] 13.4 12.2 6.76 9.12 13.7 ...\n",
      "  .. ..$ GSM914382: num [1:17879] 13.95 12.76 7.19 9.49 14.08 ...\n",
      "  .. ..$ GSM914383: num [1:17879] 14.21 13.06 6.79 9.36 13.28 ...\n",
      "  .. ..$ GSM914384: num [1:17879] 13.87 12.91 6.89 9.21 13.2 ...\n",
      "  .. ..$ GSM914385: num [1:17879] 13.81 13.1 7.03 9.34 12.53 ...\n",
      "  .. ..$ GSM914386: num [1:17879] 13.82 13.43 7.14 9.39 12.68 ...\n",
      "  .. ..$ GSM914387: num [1:17879] 13.6 14 7.03 9.23 14.07 ...\n",
      "  .. ..$ GSM914388: num [1:17879] 14.24 12.74 7.46 9.09 13.95 ...\n",
      "  .. ..$ GSM914389: num [1:17879] 13.91 13.47 7.13 9.37 14.2 ...\n",
      "  .. ..$ GSM914390: num [1:17879] 13.43 12.87 7.08 9.1 13.49 ...\n",
      "  .. ..$ GSM914391: num [1:17879] 13.47 12.96 7.31 9.56 14.6 ...\n",
      "  .. ..$ GSM914392: num [1:17879] 13.76 12.74 6.74 9.38 14.37 ...\n",
      "  .. ..$ GSM914393: num [1:17879] 14.17 12.96 7.24 9.35 14.51 ...\n",
      "  .. ..$ GSM914394: num [1:17879] 13.58 12.9 7.06 9.37 14.64 ...\n",
      "  .. ..$ GSM914395: num [1:17879] 13.95 12.8 7.01 9.35 14.67 ...\n",
      "  .. ..$ GSM914396: num [1:17879] 13.5 13.14 6.99 9.37 14.53 ...\n",
      "  .. ..$ GSM914397: num [1:17879] 13.81 12.75 6.33 9.25 14.9 ...\n",
      "  .. ..$ GSM914398: num [1:17879] 13.53 12.82 6.71 9.27 14.91 ...\n",
      "  .. ..$ GSM914399: num [1:17879] 14.09 12.5 6.63 9.3 13.87 ...\n",
      "  .. ..$ GSM914400: num [1:17879] 13.31 12.94 7.06 9.57 14.08 ...\n",
      "  .. ..$ GSM914401: num [1:17879] 14.14 12.69 7.31 9.24 14.28 ...\n",
      "  .. ..$ GSM914402: num [1:17879] 14.07 12.9 6.94 9.31 14.44 ...\n",
      "  .. ..$ GSM914403: num [1:17879] 13.65 12.7 6.7 9.26 14.62 ...\n",
      "  .. ..$ GSM914404: num [1:17879] 13.63 12.9 6.64 9.23 14.46 ...\n",
      "  .. ..$ GSM914405: num [1:17879] 13.42 13.22 6.69 9.49 14.41 ...\n",
      "  .. ..$ GSM914406: num [1:17879] 13.39 13.66 6.59 9.32 13.92 ...\n",
      "  .. ..$ GSM914407: num [1:17879] 13.69 12.91 6.46 9.37 14.59 ...\n",
      "  .. ..$ GSM914408: num [1:17879] 13.27 13.44 6.59 9.22 14.54 ...\n",
      "  .. ..$ GSM914409: num [1:17879] 13.35 13.41 6.57 9.54 14.47 ...\n",
      "  .. ..$ GSM914410: num [1:17879] 13.47 12.93 6.39 9.34 14.44 ...\n",
      "  .. ..$ GSM914411: num [1:17879] 13.4 13.25 6.31 9.62 13.82 ...\n",
      "  .. ..$ GSM914412: num [1:17879] 13.2 13.61 6.57 9.63 13.82 ...\n",
      "  .. ..$ GSM914413: num [1:17879] 13.5 13.05 6.56 9.45 14.39 ...\n",
      "  .. ..$ GSM914414: num [1:17879] 13.96 12.61 6.83 9.18 14.94 ...\n",
      "  .. ..$ GSM914415: num [1:17879] 13.76 13.14 6.52 9.38 14.51 ...\n",
      "  .. ..$ GSM914416: num [1:17879] 13.46 12.69 6.71 9.28 14.76 ...\n",
      "  .. ..$ GSM914417: num [1:17879] 14.06 13.15 6.82 9.14 14.75 ...\n",
      "  .. ..$ GSM914418: num [1:17879] 13.67 13.58 7.08 9.45 13.88 ...\n",
      "  .. ..$ GSM914419: num [1:17879] 12.68 13.61 6.63 9.52 12.96 ...\n",
      "  .. ..$ GSM914420: num [1:17879] 13.3 13.74 6.81 9.36 13.83 ...\n",
      "  .. ..$ GSM914421: num [1:17879] 13.81 13.25 7.12 9.49 14.05 ...\n",
      "  .. ..$ GSM914422: num [1:17879] 13.01 13.23 6.97 9.23 14.29 ...\n",
      "  .. ..$ GSM914423: num [1:17879] 13.48 13.41 6.88 9.45 14.57 ...\n",
      "  .. ..$ GSM914424: num [1:17879] 13.3 13.24 6.79 9.54 14.18 ...\n",
      "  .. ..$ GSM914425: num [1:17879] 13.47 13.26 6.24 9.37 13.93 ...\n",
      "  .. ..$ GSM914426: num [1:17879] 13.25 13.14 6.66 9.18 13.56 ...\n",
      "  .. ..$ GSM914427: num [1:17879] 13.52 12.96 6.81 9.31 14.47 ...\n",
      "  .. ..$ GSM914428: num [1:17879] 13.62 13.44 6.62 9.47 13.93 ...\n",
      "  .. ..$ GSM914429: num [1:17879] 13.25 13.66 7.25 9.39 14.05 ...\n",
      "  .. ..$ GSM914430: num [1:17879] 13.62 12.95 6.94 9.35 14.6 ...\n",
      "  .. ..$ GSM914431: num [1:17879] 13.11 13.58 7.15 9.09 14.28 ...\n",
      "  .. ..$ GSM914432: num [1:17879] 13.97 12.75 6.77 9.36 14.4 ...\n",
      "  .. ..$ GSM914433: num [1:17879] 13.55 13.36 6.99 9.41 14.19 ...\n",
      "  .. ..$ GSM914434: num [1:17879] 13.99 12.43 6.85 9.36 13.5 ...\n",
      "  .. ..$ GSM914435: num [1:17879] 13.71 13.34 6.86 9.33 12.47 ...\n",
      "  .. ..$ GSM914436: num [1:17879] 13.81 13.17 6.75 9.45 12.44 ...\n",
      "  .. ..$ GSM914437: num [1:17879] 13.74 13.38 7.08 9.34 12.88 ...\n",
      "  .. ..$ GSM914438: num [1:17879] 13.69 13.44 6.81 9.61 13.98 ...\n",
      "  .. ..$ GSM914439: num [1:17879] 13.36 13.21 6.55 9.38 13.82 ...\n",
      "  .. ..$ GSM914440: num [1:17879] 13.84 12.72 6.55 9.26 14.26 ...\n",
      "  .. ..$ GSM914441: num [1:17879] 13.57 13.44 6.89 9.55 14.03 ...\n",
      "  .. ..$ GSM914442: num [1:17879] 13.51 13.33 6.61 9.2 14.37 ...\n",
      "  .. ..$ GSM914443: num [1:17879] 13.75 13.23 6.95 9.63 14.22 ...\n",
      "  .. ..$ GSM914444: num [1:17879] 14.03 13.16 7.08 9.49 14.53 ...\n",
      "  .. ..$ GSM914445: num [1:17879] 14.07 13.15 7 9.31 14.86 ...\n",
      "  .. ..$ GSM914446: num [1:17879] 13.66 13.21 6.15 9.3 14.73 ...\n",
      "  .. ..$ GSM914447: num [1:17879] 13.84 12.66 6.94 9.36 14.78 ...\n",
      "  .. ..$ GSM914448: num [1:17879] 13.42 12.77 6.81 9.24 14.58 ...\n",
      "  .. ..$ GSM914449: num [1:17879] 14.05 13.26 6.94 9.44 14.69 ...\n",
      "  .. ..$ GSM914450: num [1:17879] 13.07 13.06 6.81 9.19 14.24 ...\n",
      "  .. ..$ GSM914451: num [1:17879] 14.03 12.92 6.91 9.35 14.08 ...\n",
      "  .. .. [list output truncated]\n",
      "  ..$ GSE39939:'data.frame':\t17879 obs. of  157 variables:\n",
      "  .. ..$ GSM1249177: num [1:17879] 13.98 12.84 6.67 9.35 11.45 ...\n",
      "  .. ..$ GSM1249178: num [1:17879] 14.04 12.96 6.57 9.26 13.22 ...\n",
      "  .. ..$ GSM1249179: num [1:17879] 13.36 12.84 6.5 8.96 12.65 ...\n",
      "  .. ..$ GSM1249180: num [1:17879] 13.64 12.92 6.65 9.2 11.99 ...\n",
      "  .. ..$ GSM1249181: num [1:17879] 14.27 12.91 6.97 9.63 13.23 ...\n",
      "  .. ..$ GSM1249182: num [1:17879] 13.66 13.4 6.8 9.05 11.87 ...\n",
      "  .. ..$ GSM1249183: num [1:17879] 14 12.73 7.27 9.52 12.03 ...\n",
      "  .. ..$ GSM1249184: num [1:17879] 13.61 13.24 6.96 9.58 10.9 ...\n",
      "  .. ..$ GSM1249185: num [1:17879] 13.99 12.76 6.89 9.14 13.06 ...\n",
      "  .. ..$ GSM1249186: num [1:17879] 13.79 13.02 6.47 9.19 11.63 ...\n",
      "  .. ..$ GSM1249187: num [1:17879] 14.19 12.89 7.2 9.28 13.44 ...\n",
      "  .. ..$ GSM1249188: num [1:17879] 14.22 13.3 6.91 9.29 13.46 ...\n",
      "  .. ..$ GSM1249189: num [1:17879] 14.49 12.62 6.3 9.02 13.37 ...\n",
      "  .. ..$ GSM1249190: num [1:17879] 14.08 12.13 6.62 9.25 14.14 ...\n",
      "  .. ..$ GSM1249191: num [1:17879] 14.25 12.37 6.89 9.06 14.63 ...\n",
      "  .. ..$ GSM1249192: num [1:17879] 14.23 12.83 7.28 8.98 12.57 ...\n",
      "  .. ..$ GSM1249193: num [1:17879] 14.32 12.89 7.15 8.93 12.88 ...\n",
      "  .. ..$ GSM1249194: num [1:17879] 14.21 13.18 7.11 9.33 12 ...\n",
      "  .. ..$ GSM1249195: num [1:17879] 14.07 12.84 6.84 9.15 13.7 ...\n",
      "  .. ..$ GSM1249196: num [1:17879] 14.22 12.08 6.92 8.97 13.53 ...\n",
      "  .. ..$ GSM1249197: num [1:17879] 14.14 12.08 6.85 9.08 13.86 ...\n",
      "  .. ..$ GSM1249198: num [1:17879] 14.08 12.2 6.69 9.14 13.32 ...\n",
      "  .. ..$ GSM1249199: num [1:17879] 14.32 12.45 6.63 9.35 13.9 ...\n",
      "  .. ..$ GSM1249200: num [1:17879] 14.28 12.74 7.18 9.28 12.65 ...\n",
      "  .. ..$ GSM1249201: num [1:17879] 14.42 12.09 7.24 9.49 12.87 ...\n",
      "  .. ..$ GSM1249202: num [1:17879] 14.12 12.64 6.89 9.17 12.64 ...\n",
      "  .. ..$ GSM1249203: num [1:17879] 14.41 12.31 6.92 9.01 13.15 ...\n",
      "  .. ..$ GSM1249204: num [1:17879] 13.73 12.32 6.89 9.34 12.41 ...\n",
      "  .. ..$ GSM1249205: num [1:17879] 14.34 12.71 6.53 9.2 12.95 ...\n",
      "  .. ..$ GSM1249206: num [1:17879] 14.3 12.57 6.85 9.1 12.82 ...\n",
      "  .. ..$ GSM1249207: num [1:17879] 14.22 12.42 6.77 9.5 12.73 ...\n",
      "  .. ..$ GSM1249208: num [1:17879] 14.27 13.27 6.84 9.3 12.47 ...\n",
      "  .. ..$ GSM1249209: num [1:17879] 14.11 12.43 6.89 9.19 13.57 ...\n",
      "  .. ..$ GSM1249210: num [1:17879] 14.03 12.91 7.09 9 12.27 ...\n",
      "  .. ..$ GSM1249211: num [1:17879] 14.45 13.09 6.46 9.23 12.62 ...\n",
      "  .. ..$ GSM1249212: num [1:17879] 14.52 12.38 6.8 9.11 12.89 ...\n",
      "  .. ..$ GSM1249213: num [1:17879] 13.89 13.59 7.26 9.14 11.97 ...\n",
      "  .. ..$ GSM1249214: num [1:17879] 14.18 12.54 6.87 9.2 13.16 ...\n",
      "  .. ..$ GSM1249215: num [1:17879] 14.44 12.93 6.99 8.88 13.12 ...\n",
      "  .. ..$ GSM1249216: num [1:17879] 14.21 13.34 6.81 9.68 11.41 ...\n",
      "  .. ..$ GSM1249217: num [1:17879] 14.05 13.02 6.63 9.34 10.98 ...\n",
      "  .. ..$ GSM1249218: num [1:17879] 14.17 12.87 6.25 9.21 13.22 ...\n",
      "  .. ..$ GSM1249219: num [1:17879] 13.99 12.53 6.59 9.49 12.16 ...\n",
      "  .. ..$ GSM1249220: num [1:17879] 14.07 12.57 6.7 9.43 12.34 ...\n",
      "  .. ..$ GSM1249221: num [1:17879] 14.14 12.18 6.68 9.22 12.54 ...\n",
      "  .. ..$ GSM1249222: num [1:17879] 14.13 12.93 6.58 9.37 12.95 ...\n",
      "  .. ..$ GSM1249223: num [1:17879] 14.41 12.92 6.76 9.17 12.93 ...\n",
      "  .. ..$ GSM1249224: num [1:17879] 14.01 12.65 6.49 9.28 12.2 ...\n",
      "  .. ..$ GSM1249225: num [1:17879] 14.23 12.3 6.85 9.24 14.26 ...\n",
      "  .. ..$ GSM1249226: num [1:17879] 13.92 12.57 7.31 9.39 12.02 ...\n",
      "  .. ..$ GSM1249227: num [1:17879] 13.83 13.48 6.92 9.26 11.2 ...\n",
      "  .. ..$ GSM1249228: num [1:17879] 14.08 12.55 6.73 9.17 12.41 ...\n",
      "  .. ..$ GSM1249229: num [1:17879] 14.52 12.18 6.72 9.52 12.96 ...\n",
      "  .. ..$ GSM1249230: num [1:17879] 14.23 12.85 7.06 9.55 12.28 ...\n",
      "  .. ..$ GSM1249231: num [1:17879] 13.92 12.15 6.69 9.32 11.99 ...\n",
      "  .. ..$ GSM1249232: num [1:17879] 13.98 12.98 7.16 9.25 13.02 ...\n",
      "  .. ..$ GSM1249233: num [1:17879] 14.09 12.99 7.2 9.11 12.91 ...\n",
      "  .. ..$ GSM1249234: num [1:17879] 14.19 12.29 6.4 9.26 13.47 ...\n",
      "  .. ..$ GSM1249235: num [1:17879] 14.07 12.79 6.96 9.43 11.91 ...\n",
      "  .. ..$ GSM1249236: num [1:17879] 14.39 12.25 7.07 9.25 13.39 ...\n",
      "  .. ..$ GSM1249237: num [1:17879] 14.26 12.21 6.96 9.14 12.47 ...\n",
      "  .. ..$ GSM1249238: num [1:17879] 14.38 12.11 6.93 9.36 12.91 ...\n",
      "  .. ..$ GSM1249239: num [1:17879] 14.23 12.44 6.72 9.3 12.79 ...\n",
      "  .. ..$ GSM1249240: num [1:17879] 14.21 12.55 6.81 9.17 12.85 ...\n",
      "  .. ..$ GSM1249241: num [1:17879] 14.07 12.87 6.72 8.98 12.66 ...\n",
      "  .. ..$ GSM1249242: num [1:17879] 14.32 12.46 6.95 9.27 13 ...\n",
      "  .. ..$ GSM1249243: num [1:17879] 13.56 12.62 6.86 8.93 13.92 ...\n",
      "  .. ..$ GSM1249244: num [1:17879] 13.61 12.84 6.8 9.23 12.45 ...\n",
      "  .. ..$ GSM1249245: num [1:17879] 14.18 12.35 7 9.08 14.97 ...\n",
      "  .. ..$ GSM1249246: num [1:17879] 13.94 13.51 6.74 9.11 12.43 ...\n",
      "  .. ..$ GSM1249247: num [1:17879] 13.71 13.37 6.67 9.23 12.27 ...\n",
      "  .. ..$ GSM1249248: num [1:17879] 13.08 12.72 7.55 9.12 12.7 ...\n",
      "  .. ..$ GSM1249249: num [1:17879] 13.76 13.01 6.78 9.08 12.75 ...\n",
      "  .. ..$ GSM1249250: num [1:17879] 14.22 12.58 6.78 9.46 12.6 ...\n",
      "  .. ..$ GSM1249251: num [1:17879] 14.47 12.48 6.71 9.1 12.99 ...\n",
      "  .. ..$ GSM1249252: num [1:17879] 14.15 12.6 7.19 9.17 13.39 ...\n",
      "  .. ..$ GSM1249253: num [1:17879] 13.69 12.91 7.23 9.08 14.22 ...\n",
      "  .. ..$ GSM1249254: num [1:17879] 13.87 12.98 7.13 9.24 11.67 ...\n",
      "  .. ..$ GSM1249255: num [1:17879] 13.84 12.76 6.97 9.01 11.56 ...\n",
      "  .. ..$ GSM1249256: num [1:17879] 14.15 12.59 7.38 9.06 12.81 ...\n",
      "  .. ..$ GSM1249257: num [1:17879] 13.94 13.11 6.94 9.18 12.1 ...\n",
      "  .. ..$ GSM1249258: num [1:17879] 14.27 12.54 6.76 9.4 13.2 ...\n",
      "  .. ..$ GSM1249259: num [1:17879] 13.85 12.84 7.01 9.34 12.31 ...\n",
      "  .. ..$ GSM1249260: num [1:17879] 13.89 12.74 6.49 9.15 12.92 ...\n",
      "  .. ..$ GSM1249261: num [1:17879] 13.83 12.1 6.8 9.13 12.36 ...\n",
      "  .. ..$ GSM1249262: num [1:17879] 14.22 12.24 7.22 9.22 12.83 ...\n",
      "  .. ..$ GSM1249263: num [1:17879] 14.3 12.65 6.74 9.11 13.36 ...\n",
      "  .. ..$ GSM1249264: num [1:17879] 14.41 12.59 6.6 9.46 12.89 ...\n",
      "  .. ..$ GSM1249265: num [1:17879] 14.26 12.86 6.91 9.41 12.88 ...\n",
      "  .. ..$ GSM1249266: num [1:17879] 14.42 12.62 7.24 9.39 13.57 ...\n",
      "  .. ..$ GSM1249267: num [1:17879] 14.46 12.24 6.77 9.17 14.76 ...\n",
      "  .. ..$ GSM1249268: num [1:17879] 14 12.47 6.75 9.33 12.93 ...\n",
      "  .. ..$ GSM1249269: num [1:17879] 14.41 12.53 6.94 9.11 12.92 ...\n",
      "  .. ..$ GSM1249270: num [1:17879] 14.33 12.48 6.84 9.24 13.01 ...\n",
      "  .. ..$ GSM1249271: num [1:17879] 14.5 12.15 7.03 9.11 13.48 ...\n",
      "  .. ..$ GSM1249272: num [1:17879] 14.29 12.5 7.41 9.26 12.03 ...\n",
      "  .. ..$ GSM1249273: num [1:17879] 14.29 12.31 6.58 9.38 11.48 ...\n",
      "  .. ..$ GSM1249274: num [1:17879] 14.49 12.24 7.4 9.04 11.85 ...\n",
      "  .. ..$ GSM1249275: num [1:17879] 14.4 12.09 7.09 9.13 13.13 ...\n",
      "  .. .. [list output truncated]\n",
      "  ..$ GSE39940:'data.frame':\t17879 obs. of  334 variables:\n",
      "  .. ..$ GSM981627: num [1:17879] 14.23 12.92 7.09 9.55 13.72 ...\n",
      "  .. ..$ GSM981628: num [1:17879] 14.28 13.3 7.13 9.59 13.49 ...\n",
      "  .. ..$ GSM981629: num [1:17879] 13.99 12.72 7.17 9.35 14.45 ...\n",
      "  .. ..$ GSM981630: num [1:17879] 14.29 12.93 6.96 9.77 13.83 ...\n",
      "  .. ..$ GSM981631: num [1:17879] 13.85 13.03 7.2 9.36 13.76 ...\n",
      "  .. ..$ GSM981632: num [1:17879] 14.36 12.93 7 9.64 14.61 ...\n",
      "  .. ..$ GSM981633: num [1:17879] 14.56 13.19 6.56 9.3 14.54 ...\n",
      "  .. ..$ GSM981634: num [1:17879] 13.93 13.33 6.68 9.27 13.4 ...\n",
      "  .. ..$ GSM981635: num [1:17879] 13.81 13.4 6.94 9.53 12.51 ...\n",
      "  .. ..$ GSM981636: num [1:17879] 14.33 12.95 6.67 9.23 13.11 ...\n",
      "  .. ..$ GSM981637: num [1:17879] 13.9 13.18 6.49 9.61 12.87 ...\n",
      "  .. ..$ GSM981638: num [1:17879] 14.39 13.41 6.84 9.57 13.45 ...\n",
      "  .. ..$ GSM981639: num [1:17879] 14.33 13.21 6.83 9.55 13.22 ...\n",
      "  .. ..$ GSM981640: num [1:17879] 13.88 13.3 6.78 9.3 13.06 ...\n",
      "  .. ..$ GSM981641: num [1:17879] 14.33 12.6 6.78 9.58 13.6 ...\n",
      "  .. ..$ GSM981642: num [1:17879] 14.31 13.28 7.22 9.35 14.01 ...\n",
      "  .. ..$ GSM981643: num [1:17879] 14.2 13.15 6.73 9.52 13.85 ...\n",
      "  .. ..$ GSM981644: num [1:17879] 14.18 13.12 7.33 9.41 12.72 ...\n",
      "  .. ..$ GSM981645: num [1:17879] 13.95 13.74 6.69 9.61 12.87 ...\n",
      "  .. ..$ GSM981646: num [1:17879] 13.91 13.04 6.73 9.52 12.71 ...\n",
      "  .. ..$ GSM981647: num [1:17879] 14.37 13.01 6.82 9.48 13.06 ...\n",
      "  .. ..$ GSM981648: num [1:17879] 14.34 13.1 6.88 9.46 12.89 ...\n",
      "  .. ..$ GSM981649: num [1:17879] 13.82 13.82 6.59 9.42 12.43 ...\n",
      "  .. ..$ GSM981650: num [1:17879] 14.34 12.96 6.94 9.44 13.26 ...\n",
      "  .. ..$ GSM981651: num [1:17879] 14.53 12.78 6.99 9.44 13.06 ...\n",
      "  .. ..$ GSM981652: num [1:17879] 13.73 13.93 6.78 9.17 11.96 ...\n",
      "  .. ..$ GSM981653: num [1:17879] 14.05 13.21 6.89 9.77 12.8 ...\n",
      "  .. ..$ GSM981654: num [1:17879] 14.13 13.13 7.03 9.24 12.47 ...\n",
      "  .. ..$ GSM981655: num [1:17879] 14.43 12.88 6.42 9.7 12.36 ...\n",
      "  .. ..$ GSM981656: num [1:17879] 14.25 13.4 6.58 9.21 11.99 ...\n",
      "  .. ..$ GSM981657: num [1:17879] 14.04 13.06 6.95 9.2 12.59 ...\n",
      "  .. ..$ GSM981658: num [1:17879] 14.3 13.24 6.68 9.75 11.41 ...\n",
      "  .. ..$ GSM981659: num [1:17879] 14.48 12.88 7.21 9.42 12.1 ...\n",
      "  .. ..$ GSM981660: num [1:17879] 14.3 13.26 6.67 9.61 10.73 ...\n",
      "  .. ..$ GSM981661: num [1:17879] 13.9 12.5 6.99 9.53 10.67 ...\n",
      "  .. ..$ GSM981662: num [1:17879] 14.39 12.92 6.38 9.13 13.23 ...\n",
      "  .. ..$ GSM981663: num [1:17879] 13.95 13.34 6.71 9.16 13.62 ...\n",
      "  .. ..$ GSM981664: num [1:17879] 14.25 13.15 7.08 9.5 13.36 ...\n",
      "  .. ..$ GSM981665: num [1:17879] 13.73 13.72 6.62 9.21 13.8 ...\n",
      "  .. ..$ GSM981666: num [1:17879] 13.91 13.51 6.47 9.5 13.08 ...\n",
      "  .. ..$ GSM981667: num [1:17879] 12.02 13.7 6.77 9.36 12.96 ...\n",
      "  .. ..$ GSM981668: num [1:17879] 13.84 13.14 6.82 9.25 13.24 ...\n",
      "  .. ..$ GSM981669: num [1:17879] 14.43 13.05 6.69 9.42 13.63 ...\n",
      "  .. ..$ GSM981670: num [1:17879] 13.81 13.13 6.71 9.54 14.56 ...\n",
      "  .. ..$ GSM981671: num [1:17879] 14.18 13.06 7.27 9.36 14.27 ...\n",
      "  .. ..$ GSM981672: num [1:17879] 13.5 13.08 6.76 9.23 14.46 ...\n",
      "  .. ..$ GSM981673: num [1:17879] 14.11 13.32 6.79 9.19 14.49 ...\n",
      "  .. ..$ GSM981674: num [1:17879] 14.31 12.83 6.8 9.24 14.42 ...\n",
      "  .. ..$ GSM981675: num [1:17879] 13.76 13.63 6.49 9.54 14.43 ...\n",
      "  .. ..$ GSM981676: num [1:17879] 14.11 12.24 6.82 9.05 14.74 ...\n",
      "  .. ..$ GSM981677: num [1:17879] 14.4 12.76 6.69 9.15 14.98 ...\n",
      "  .. ..$ GSM981678: num [1:17879] 13.59 14.06 6.76 9.31 13.68 ...\n",
      "  .. ..$ GSM981679: num [1:17879] 13.85 13.31 6.58 9.23 14.62 ...\n",
      "  .. ..$ GSM981680: num [1:17879] 14.06 13.15 6.78 9.17 14.17 ...\n",
      "  .. ..$ GSM981681: num [1:17879] 14.09 13.01 6.56 9.05 14.39 ...\n",
      "  .. ..$ GSM981682: num [1:17879] 14.01 13.2 6.83 9.3 14.38 ...\n",
      "  .. ..$ GSM981683: num [1:17879] 14.24 12.6 6.65 9.1 14.74 ...\n",
      "  .. ..$ GSM981684: num [1:17879] 13.63 13.83 7.01 9.46 11.98 ...\n",
      "  .. ..$ GSM981685: num [1:17879] 14.17 12.89 6.62 9.17 14.73 ...\n",
      "  .. ..$ GSM981686: num [1:17879] 13.98 12.68 6.27 9.01 14.16 ...\n",
      "  .. ..$ GSM981687: num [1:17879] 13.71 12.87 6.5 9.64 12.8 ...\n",
      "  .. ..$ GSM981688: num [1:17879] 14.32 12.81 6.4 9.38 13.54 ...\n",
      "  .. ..$ GSM981689: num [1:17879] 14.5 12.65 6.85 9.36 13.68 ...\n",
      "  .. ..$ GSM981690: num [1:17879] 14.31 13.26 6.81 9.34 13.75 ...\n",
      "  .. ..$ GSM981691: num [1:17879] 14.17 13.36 6.76 9.55 13.49 ...\n",
      "  .. ..$ GSM981692: num [1:17879] 14.03 13.21 6.82 9.26 13.51 ...\n",
      "  .. ..$ GSM981693: num [1:17879] 14.37 12.91 6.91 9.29 13.49 ...\n",
      "  .. ..$ GSM981694: num [1:17879] 14.44 12.03 6.66 9.36 13.67 ...\n",
      "  .. ..$ GSM981695: num [1:17879] 14.24 12.79 6.58 9.38 13.6 ...\n",
      "  .. ..$ GSM981696: num [1:17879] 14.4 13.17 6.81 9.43 13.26 ...\n",
      "  .. ..$ GSM981697: num [1:17879] 14.36 12.89 6.97 9.71 13.47 ...\n",
      "  .. ..$ GSM981698: num [1:17879] 14.27 13.19 7.01 9.51 12.92 ...\n",
      "  .. ..$ GSM981699: num [1:17879] 13.93 13.47 6.53 9.44 12.76 ...\n",
      "  .. ..$ GSM981700: num [1:17879] 14.21 13.01 6.54 9.3 12.41 ...\n",
      "  .. ..$ GSM981701: num [1:17879] 14.36 13.36 6.56 9.45 12.96 ...\n",
      "  .. ..$ GSM981702: num [1:17879] 14.17 13.06 6.89 9.51 12.69 ...\n",
      "  .. ..$ GSM981703: num [1:17879] 14.14 13.02 6.93 9.43 12.55 ...\n",
      "  .. ..$ GSM981704: num [1:17879] 14.27 13.29 6.53 9.66 12.45 ...\n",
      "  .. ..$ GSM981705: num [1:17879] 14.41 13.28 6.53 9.34 12.34 ...\n",
      "  .. ..$ GSM981706: num [1:17879] 14.02 13.48 6.92 9.37 11.98 ...\n",
      "  .. ..$ GSM981707: num [1:17879] 14.1 13.14 7.24 9.26 11.93 ...\n",
      "  .. ..$ GSM981708: num [1:17879] 14.17 12.53 7.23 9.38 12.74 ...\n",
      "  .. ..$ GSM981709: num [1:17879] 14.28 13.12 7.03 9.41 12.77 ...\n",
      "  .. ..$ GSM981710: num [1:17879] 14.06 13.23 6.54 9.22 11.32 ...\n",
      "  .. ..$ GSM981711: num [1:17879] 13.65 12.69 6.98 9.62 12.03 ...\n",
      "  .. ..$ GSM981712: num [1:17879] 14.59 12.88 7.07 9.56 12.54 ...\n",
      "  .. ..$ GSM981713: num [1:17879] 14.5 13.34 6.92 9.68 12.08 ...\n",
      "  .. ..$ GSM981714: num [1:17879] 14.29 13.15 6.67 9.39 12.78 ...\n",
      "  .. ..$ GSM981715: num [1:17879] 14.24 13.04 6.72 8.87 12.43 ...\n",
      "  .. ..$ GSM981716: num [1:17879] 13.96 13.16 6.52 9.33 12.69 ...\n",
      "  .. ..$ GSM981717: num [1:17879] 14.28 12.78 6.93 9.28 13.48 ...\n",
      "  .. ..$ GSM981718: num [1:17879] 14.39 13.32 6.75 9.15 13.59 ...\n",
      "  .. ..$ GSM981719: num [1:17879] 14.33 13.3 6.38 9.12 13.25 ...\n",
      "  .. ..$ GSM981720: num [1:17879] 13.95 13.65 6.8 9.25 12.91 ...\n",
      "  .. ..$ GSM981721: num [1:17879] 13.73 13.52 6.62 9.26 14.1 ...\n",
      "  .. ..$ GSM981722: num [1:17879] 13.92 13.75 7.26 8.99 13.26 ...\n",
      "  .. ..$ GSM981723: num [1:17879] 14.13 13.14 6.72 9.46 13.46 ...\n",
      "  .. ..$ GSM981724: num [1:17879] 13.98 13.35 6.95 9.31 13.29 ...\n",
      "  .. ..$ GSM981725: num [1:17879] 13.86 13.6 6.98 9.18 13.59 ...\n",
      "  .. .. [list output truncated]\n",
      "  ..$ GSE42825:'data.frame':\t11147 obs. of  42 variables:\n",
      "  .. ..$ GSM1050732: num [1:11147] 13.95 10.11 5.04 6.6 11.06 ...\n",
      "  .. ..$ GSM1050733: num [1:11147] 13.81 9.65 5.39 6.39 11.32 ...\n",
      "  .. ..$ GSM1050734: num [1:11147] 14.49 9.97 5.16 6.88 11.95 ...\n",
      "  .. ..$ GSM1050735: num [1:11147] 13.57 10.3 5.44 6.72 11.01 ...\n",
      "  .. ..$ GSM1050736: num [1:11147] 14.05 10.08 4.85 6.69 10.14 ...\n",
      "  .. ..$ GSM1050737: num [1:11147] 13.66 9.55 4.95 6.55 10.53 ...\n",
      "  .. ..$ GSM1050738: num [1:11147] 14.07 9.93 5.22 6.57 10.75 ...\n",
      "  .. ..$ GSM1050739: num [1:11147] 13.8 10.34 4.94 6.68 10.51 ...\n",
      "  .. ..$ GSM1050740: num [1:11147] 14 10.05 5.25 6.71 11.27 ...\n",
      "  .. ..$ GSM1050741: num [1:11147] 14.18 9.94 5.52 6.4 10.98 ...\n",
      "  .. ..$ GSM1050742: num [1:11147] 14.01 10.24 5.56 6.83 11.21 ...\n",
      "  .. ..$ GSM1050743: num [1:11147] 14.08 9.9 5.07 6.48 11.18 ...\n",
      "  .. ..$ GSM1050744: num [1:11147] 14.41 9.62 4.98 6.5 10.97 ...\n",
      "  .. ..$ GSM1050745: num [1:11147] 13.96 10.16 5.32 6.82 10.86 ...\n",
      "  .. ..$ GSM1050746: num [1:11147] 13.98 9.87 5.27 6.39 11.33 ...\n",
      "  .. ..$ GSM1050747: num [1:11147] 14.13 9.68 5.52 6.43 11.43 ...\n",
      "  .. ..$ GSM1050748: num [1:11147] 14.08 10.15 5.38 6.89 10.81 ...\n",
      "  .. ..$ GSM1050749: num [1:11147] 14.01 10.04 5.21 6.57 11.4 ...\n",
      "  .. ..$ GSM1050750: num [1:11147] 14.11 9.76 5.68 6.51 11.19 ...\n",
      "  .. ..$ GSM1050751: num [1:11147] 14.03 9.71 5.45 6.29 11.04 ...\n",
      "  .. ..$ GSM1050752: num [1:11147] 14 9.76 5.93 6.6 11.51 ...\n",
      "  .. ..$ GSM1050753: num [1:11147] 14.01 9.5 5.08 6.73 11.33 ...\n",
      "  .. ..$ GSM1050754: num [1:11147] 14.02 10.17 5.28 6.66 11.4 ...\n",
      "  .. ..$ GSM1050755: num [1:11147] 14.1 9.7 5.65 6.66 11.16 ...\n",
      "  .. ..$ GSM1050756: num [1:11147] 14.16 9.84 5.08 6.45 11.35 ...\n",
      "  .. ..$ GSM1050757: num [1:11147] 14.22 9.63 5.03 6.51 11.18 ...\n",
      "  .. ..$ GSM1050758: num [1:11147] 13.84 9.97 5.44 6.37 11.02 ...\n",
      "  .. ..$ GSM1050759: num [1:11147] 13.95 9.89 5.45 6.76 11.14 ...\n",
      "  .. ..$ GSM1050760: num [1:11147] 13.93 10.04 5.26 6.73 10.95 ...\n",
      "  .. ..$ GSM1050761: num [1:11147] 13.92 10.23 5.06 6.8 11.04 ...\n",
      "  .. ..$ GSM1050762: num [1:11147] 13.98 9.69 5.21 6.85 10.84 ...\n",
      "  .. ..$ GSM1050763: num [1:11147] 14.09 9.77 5.11 6.67 11.35 ...\n",
      "  .. ..$ GSM1050764: num [1:11147] 14.08 9.39 5.25 6.78 11.51 ...\n",
      "  .. ..$ GSM1050765: num [1:11147] 14.06 9.71 5.01 6.63 11.21 ...\n",
      "  .. ..$ GSM1050766: num [1:11147] 14.21 10.36 5.48 6.88 11.2 ...\n",
      "  .. ..$ GSM1050767: num [1:11147] 13.94 10.12 5.13 6.74 11.36 ...\n",
      "  .. ..$ GSM1050768: num [1:11147] 14.09 9.51 4.98 6.86 10.87 ...\n",
      "  .. ..$ GSM1050769: num [1:11147] 13.93 9.76 5.24 6.41 11.12 ...\n",
      "  .. ..$ GSM1050770: num [1:11147] 14.17 10.12 5.27 6.54 11.09 ...\n",
      "  .. ..$ GSM1050771: num [1:11147] 14.09 9.61 5.95 7.02 11.21 ...\n",
      "  .. ..$ GSM1050772: num [1:11147] 13.98 9.83 4.72 6.62 11.43 ...\n",
      "  .. ..$ GSM1050773: num [1:11147] 14.09 9.8 5.46 6.79 11.08 ...\n",
      "  ..$ GSE42826:'data.frame':\t17879 obs. of  102 variables:\n",
      "  .. ..$ GSM1050774: num [1:17879] 12.92 10.35 5.53 7.68 12.71 ...\n",
      "  .. ..$ GSM1050775: num [1:17879] 12.71 9.92 5.09 7.53 11.9 ...\n",
      "  .. ..$ GSM1050776: num [1:17879] 12.21 9.71 5.12 7.47 11.12 ...\n",
      "  .. ..$ GSM1050777: num [1:17879] 11.67 10.48 5.41 7.63 10.32 ...\n",
      "  .. ..$ GSM1050778: num [1:17879] 12.71 10.02 5.23 7.74 11.82 ...\n",
      "  .. ..$ GSM1050779: num [1:17879] 12.91 10.6 5.34 7.79 11.75 ...\n",
      "  .. ..$ GSM1050780: num [1:17879] 12.28 9.67 4.6 7.37 11.56 ...\n",
      "  .. ..$ GSM1050781: num [1:17879] 11.82 9.32 5.26 7.11 10.93 ...\n",
      "  .. ..$ GSM1050782: num [1:17879] 11.64 9.09 5.44 7.2 10.89 ...\n",
      "  .. ..$ GSM1050783: num [1:17879] 12.22 9.84 5.06 7.67 11.47 ...\n",
      "  .. ..$ GSM1050784: num [1:17879] 12.37 9.87 5.54 7.68 11.46 ...\n",
      "  .. ..$ GSM1050785: num [1:17879] 12.75 10.03 5.62 7.66 11.83 ...\n",
      "  .. ..$ GSM1050786: num [1:17879] 11.74 8.72 5.11 7.36 10.54 ...\n",
      "  .. ..$ GSM1050787: num [1:17879] 12.08 9.95 5.04 7.54 11.65 ...\n",
      "  .. ..$ GSM1050788: num [1:17879] 11.75 8.94 5.33 7.21 11.19 ...\n",
      "  .. ..$ GSM1050789: num [1:17879] 11.97 9.37 5.03 7.21 11.14 ...\n",
      "  .. ..$ GSM1050790: num [1:17879] 12.34 9.81 5.22 7.69 11.52 ...\n",
      "  .. ..$ GSM1050791: num [1:17879] 11.89 9.57 5.13 7.49 11.15 ...\n",
      "  .. ..$ GSM1050792: num [1:17879] 11.82 9.53 5.68 7.57 11.12 ...\n",
      "  .. ..$ GSM1050793: num [1:17879] 12.53 9.85 5.29 7.79 11.75 ...\n",
      "  .. ..$ GSM1050794: num [1:17879] 12.87 10.57 5.23 7.77 12.2 ...\n",
      "  .. ..$ GSM1050795: num [1:17879] 12.56 10.22 5.01 7.75 11.79 ...\n",
      "  .. ..$ GSM1050796: num [1:17879] 11.52 9.72 5.29 6.98 11.07 ...\n",
      "  .. ..$ GSM1050797: num [1:17879] 12.12 10.08 5.56 7.34 11.44 ...\n",
      "  .. ..$ GSM1050798: num [1:17879] 12.13 9.64 5.25 7.7 9.94 ...\n",
      "  .. ..$ GSM1050799: num [1:17879] 11.78 9.02 5.25 7.04 10.14 ...\n",
      "  .. ..$ GSM1050800: num [1:17879] 12.66 9.51 5.15 7.51 11.69 ...\n",
      "  .. ..$ GSM1050801: num [1:17879] 11.47 9.91 5.63 7.28 9.64 ...\n",
      "  .. ..$ GSM1050802: num [1:17879] 12.56 10.33 5.2 8.08 11.18 ...\n",
      "  .. ..$ GSM1050803: num [1:17879] 12.55 9.97 5.01 7.61 11.71 ...\n",
      "  .. ..$ GSM1050804: num [1:17879] 12.68 10.34 5.5 7.84 11.53 ...\n",
      "  .. ..$ GSM1050805: num [1:17879] 12.35 9.38 5.18 7.41 11.41 ...\n",
      "  .. ..$ GSM1050806: num [1:17879] 11.92 9.42 5.24 6.93 10.98 ...\n",
      "  .. ..$ GSM1050807: num [1:17879] 11.67 9.72 5.32 7.17 10.39 ...\n",
      "  .. ..$ GSM1050808: num [1:17879] 12.71 10.53 5 7.76 11.98 ...\n",
      "  .. ..$ GSM1050809: num [1:17879] 12.22 9.09 5.18 7.4 10.99 ...\n",
      "  .. ..$ GSM1050810: num [1:17879] 12.41 9.66 5.36 7.89 11.25 ...\n",
      "  .. ..$ GSM1050811: num [1:17879] 12.13 10.3 5.31 7.49 10.7 ...\n",
      "  .. ..$ GSM1050812: num [1:17879] 12.57 9.79 5.42 7.77 11.08 ...\n",
      "  .. ..$ GSM1050813: num [1:17879] 12.15 10.06 5.26 7.29 11.11 ...\n",
      "  .. ..$ GSM1050814: num [1:17879] 11.12 9.55 5.41 7.11 9.81 ...\n",
      "  .. ..$ GSM1050815: num [1:17879] 12.1 9.07 5.93 7.1 10.37 ...\n",
      "  .. ..$ GSM1050816: num [1:17879] 12.7 10.12 5.54 7.76 11.42 ...\n",
      "  .. ..$ GSM1050817: num [1:17879] 12.13 10.01 5.13 7.46 10.71 ...\n",
      "  .. ..$ GSM1050818: num [1:17879] 11.35 10.17 5.27 7.62 10.14 ...\n",
      "  .. ..$ GSM1050819: num [1:17879] 12.17 9.57 5.42 7.25 11.14 ...\n",
      "  .. ..$ GSM1050820: num [1:17879] 12.81 9.97 5.48 7.9 11.81 ...\n",
      "  .. ..$ GSM1050821: num [1:17879] 12.58 9.64 5.5 7.23 11.87 ...\n",
      "  .. ..$ GSM1050822: num [1:17879] 11.22 9.06 4.94 6.99 9.82 ...\n",
      "  .. ..$ GSM1050823: num [1:17879] 12.43 9.63 5.63 7.42 10.88 ...\n",
      "  .. ..$ GSM1050824: num [1:17879] 11.94 9.16 5.37 7.2 10.77 ...\n",
      "  .. ..$ GSM1050825: num [1:17879] 12.45 9.95 5.27 7.95 10.9 ...\n",
      "  .. ..$ GSM1050826: num [1:17879] 12.17 9.79 5.01 7.64 10.94 ...\n",
      "  .. ..$ GSM1050827: num [1:17879] 12.74 10.18 5.37 7.73 11.32 ...\n",
      "  .. ..$ GSM1050828: num [1:17879] 12.61 10.36 5.39 7.78 11.46 ...\n",
      "  .. ..$ GSM1050829: num [1:17879] 12.59 9.5 5.56 7.24 11.63 ...\n",
      "  .. ..$ GSM1050830: num [1:17879] 11.52 10.09 5.38 7.48 10.06 ...\n",
      "  .. ..$ GSM1050831: num [1:17879] 12.08 9.18 4.97 7.05 10.93 ...\n",
      "  .. ..$ GSM1050832: num [1:17879] 11.98 9.28 5.13 7.34 11.11 ...\n",
      "  .. ..$ GSM1050833: num [1:17879] 12.14 9.24 5.21 7.32 11.26 ...\n",
      "  .. ..$ GSM1050834: num [1:17879] 11.92 10.13 5.35 7.58 10.63 ...\n",
      "  .. ..$ GSM1050835: num [1:17879] 11.69 9.5 5.23 7.42 10.61 ...\n",
      "  .. ..$ GSM1050836: num [1:17879] 11.8 9.85 4.91 7.63 11.03 ...\n",
      "  .. ..$ GSM1050837: num [1:17879] 12.77 9.79 5.4 7.69 11.98 ...\n",
      "  .. ..$ GSM1050838: num [1:17879] 12.58 10.04 5.41 7.54 11.93 ...\n",
      "  .. ..$ GSM1050839: num [1:17879] 12.5 9.7 5.16 7.57 11.36 ...\n",
      "  .. ..$ GSM1050840: num [1:17879] 11.32 10.21 5.22 7.45 10.06 ...\n",
      "  .. ..$ GSM1050841: num [1:17879] 11.72 8.91 5.02 6.84 10.83 ...\n",
      "  .. ..$ GSM1050842: num [1:17879] 12.04 9.55 4.98 7.41 11.54 ...\n",
      "  .. ..$ GSM1050843: num [1:17879] 12.26 10.15 4.87 8.01 11.44 ...\n",
      "  .. ..$ GSM1050844: num [1:17879] 12.13 9.42 4.86 7.41 11.24 ...\n",
      "  .. ..$ GSM1050845: num [1:17879] 12.59 10.14 5.26 7.93 11.22 ...\n",
      "  .. ..$ GSM1050846: num [1:17879] 12.31 9.61 5.04 7.38 11.68 ...\n",
      "  .. ..$ GSM1050847: num [1:17879] 12.19 9.92 5.06 7.25 11.8 ...\n",
      "  .. ..$ GSM1050848: num [1:17879] 12.13 9.74 5.66 7.83 11 ...\n",
      "  .. ..$ GSM1050849: num [1:17879] 12.68 9.84 5.38 7.72 12.18 ...\n",
      "  .. ..$ GSM1050850: num [1:17879] 11.99 9.02 5 7.12 11.26 ...\n",
      "  .. ..$ GSM1050851: num [1:17879] 11.98 9.49 5.46 7.28 11.31 ...\n",
      "  .. ..$ GSM1050852: num [1:17879] 12.2 9.39 4.98 7.4 11.91 ...\n",
      "  .. ..$ GSM1050853: num [1:17879] 12.52 9.96 5.11 7.61 11.91 ...\n",
      "  .. ..$ GSM1050854: num [1:17879] 11.63 8.66 5.19 6.89 10.59 ...\n",
      "  .. ..$ GSM1050855: num [1:17879] 11.77 10.04 5.04 7.81 10.77 ...\n",
      "  .. ..$ GSM1050856: num [1:17879] 11.64 9.77 5.12 7.06 10.89 ...\n",
      "  .. ..$ GSM1050857: num [1:17879] 12.13 9.99 5.35 7.21 11.53 ...\n",
      "  .. ..$ GSM1050858: num [1:17879] 12.58 9.92 4.79 7.68 12.24 ...\n",
      "  .. ..$ GSM1050859: num [1:17879] 12.05 9.43 5.45 7.74 11.65 ...\n",
      "  .. ..$ GSM1050860: num [1:17879] 11.92 9.2 5.02 7.16 11.49 ...\n",
      "  .. ..$ GSM1050861: num [1:17879] 11.88 9.23 5.12 6.9 11.35 ...\n",
      "  .. ..$ GSM1050862: num [1:17879] 12.65 10.43 5.97 8.26 12.2 ...\n",
      "  .. ..$ GSM1050863: num [1:17879] 11.32 10.54 5.06 7.54 9.97 ...\n",
      "  .. ..$ GSM1050864: num [1:17879] 12.07 10.05 5.75 7.67 11.06 ...\n",
      "  .. ..$ GSM1050865: num [1:17879] 12.18 9.79 5.61 7.82 11.12 ...\n",
      "  .. ..$ GSM1050866: num [1:17879] 12 10.18 5.39 7.98 10.79 ...\n",
      "  .. ..$ GSM1050867: num [1:17879] 11.9 10.19 5.09 7.52 10.81 ...\n",
      "  .. ..$ GSM1050868: num [1:17879] 11.89 10 5.44 7.61 10.56 ...\n",
      "  .. ..$ GSM1050869: num [1:17879] 11.87 10.33 5.27 7.58 10.79 ...\n",
      "  .. ..$ GSM1050870: num [1:17879] 12.2 9.77 5.04 7.52 10.95 ...\n",
      "  .. ..$ GSM1050871: num [1:17879] 12.09 10.15 5.28 7.65 10.86 ...\n",
      "  .. ..$ GSM1050872: num [1:17879] 12.07 9.76 4.99 7.59 11.16 ...\n",
      "  .. .. [list output truncated]\n",
      "  ..$ GSE42827:'data.frame':\t17879 obs. of  10 variables:\n",
      "  .. ..$ GSM1050876: num [1:17879] 11.99 9.9 5.35 7.66 11.11 ...\n",
      "  .. ..$ GSM1050877: num [1:17879] 11.69 10.46 5.45 7.91 10.57 ...\n",
      "  .. ..$ GSM1050878: num [1:17879] 11.69 10.5 5.73 7.59 10.31 ...\n",
      "  .. ..$ GSM1050879: num [1:17879] 11.17 10.45 5.94 7.33 9.54 ...\n",
      "  .. ..$ GSM1050880: num [1:17879] 11.85 11.05 5.31 7.89 11.23 ...\n",
      "  .. ..$ GSM1050881: num [1:17879] 11.73 9.59 5.52 7.43 11.15 ...\n",
      "  .. ..$ GSM1050882: num [1:17879] 11.86 9.51 5.57 7.64 11.19 ...\n",
      "  .. ..$ GSM1050883: num [1:17879] 12.18 9.94 5.61 7.49 11.81 ...\n",
      "  .. ..$ GSM1050884: num [1:17879] 11.92 9.85 5.74 7.6 11.41 ...\n",
      "  .. ..$ GSM1050885: num [1:17879] 11.05 10.76 5.83 7.74 10.29 ...\n",
      "  ..$ GSE42830:'data.frame':\t17879 obs. of  95 variables:\n",
      "  .. ..$ GSM1050928: num [1:17879] 12.43 9.82 5.32 7.51 11.15 ...\n",
      "  .. ..$ GSM1050929: num [1:17879] 11.46 9.23 5.17 7.48 10.29 ...\n",
      "  .. ..$ GSM1050930: num [1:17879] 11.91 10.32 4.74 7.66 10.79 ...\n",
      "  .. ..$ GSM1050931: num [1:17879] 11.78 10.05 5.91 7.28 10.86 ...\n",
      "  .. ..$ GSM1050932: num [1:17879] 12.07 9.76 5.56 7.63 11.88 ...\n",
      "  .. ..$ GSM1050933: num [1:17879] 11.87 9.98 5.67 7.79 11.25 ...\n",
      "  .. ..$ GSM1050934: num [1:17879] 12.21 10.12 5.38 7.82 11.1 ...\n",
      "  .. ..$ GSM1050935: num [1:17879] 12.09 10.03 5.89 7.88 11.52 ...\n",
      "  .. ..$ GSM1050936: num [1:17879] 12.47 10.38 5.2 7.86 11.82 ...\n",
      "  .. ..$ GSM1050937: num [1:17879] 11.62 9.22 5.54 7.34 11.25 ...\n",
      "  .. ..$ GSM1050938: num [1:17879] 12.1 9.85 5.53 7.71 11.73 ...\n",
      "  .. ..$ GSM1050939: num [1:17879] 11.98 9.81 5.46 7.52 10.97 ...\n",
      "  .. ..$ GSM1050940: num [1:17879] 11.97 9.85 5.46 7.83 10.97 ...\n",
      "  .. ..$ GSM1050941: num [1:17879] 12.09 9.67 5.52 7.67 11.24 ...\n",
      "  .. ..$ GSM1050942: num [1:17879] 11.58 10.64 5.39 7.82 11.19 ...\n",
      "  .. ..$ GSM1050943: num [1:17879] 11.35 10.24 5.54 7.05 9.88 ...\n",
      "  .. ..$ GSM1050944: num [1:17879] 10.98 9.46 5.78 7.03 9.64 ...\n",
      "  .. ..$ GSM1050945: num [1:17879] 11.57 10.63 5.54 7.84 10.35 ...\n",
      "  .. ..$ GSM1050946: num [1:17879] 11.61 9.65 5.69 7.72 10.61 ...\n",
      "  .. ..$ GSM1050947: num [1:17879] 11.79 9.96 5.3 7.39 10.93 ...\n",
      "  .. ..$ GSM1050948: num [1:17879] 11.38 9.34 5.18 7.03 9.79 ...\n",
      "  .. ..$ GSM1050949: num [1:17879] 12.06 9.82 5.18 7.76 10.63 ...\n",
      "  .. ..$ GSM1050950: num [1:17879] 11.59 9.79 5.1 7.75 9.99 ...\n",
      "  .. ..$ GSM1050951: num [1:17879] 11.35 9.35 5.42 7.68 10.31 ...\n",
      "  .. ..$ GSM1050952: num [1:17879] 11.3 9.76 5.42 7.45 9.77 ...\n",
      "  .. ..$ GSM1050953: num [1:17879] 11.63 9.52 5.45 7.41 10.18 ...\n",
      "  .. ..$ GSM1050954: num [1:17879] 11.99 9.39 5.26 7.53 10.61 ...\n",
      "  .. ..$ GSM1050955: num [1:17879] 11.35 9.69 5.44 7.12 9.77 ...\n",
      "  .. ..$ GSM1050956: num [1:17879] 12.25 9.78 5.6 7.66 10.93 ...\n",
      "  .. ..$ GSM1050957: num [1:17879] 11.8 9.06 5.01 7.18 10.39 ...\n",
      "  .. ..$ GSM1050958: num [1:17879] 11.61 10.02 5.54 7.38 9.96 ...\n",
      "  .. ..$ GSM1050959: num [1:17879] 12.45 10.1 5.55 7.96 11.13 ...\n",
      "  .. ..$ GSM1050960: num [1:17879] 11.68 10.46 5.36 7.88 10.57 ...\n",
      "  .. ..$ GSM1050961: num [1:17879] 11.61 10.47 5.77 7.56 10.47 ...\n",
      "  .. ..$ GSM1050962: num [1:17879] 12.02 9.44 5.74 7.45 10.68 ...\n",
      "  .. ..$ GSM1050963: num [1:17879] 11.68 10.5 5.66 7.55 10.31 ...\n",
      "  .. ..$ GSM1050964: num [1:17879] 12.26 9.58 5.78 7.58 11.11 ...\n",
      "  .. ..$ GSM1050965: num [1:17879] 10.98 10.32 5.66 7.24 9.32 ...\n",
      "  .. ..$ GSM1050966: num [1:17879] 10.83 10.11 5.13 7.02 8.9 ...\n",
      "  .. ..$ GSM1050967: num [1:17879] 11.99 10.34 5.74 7.9 10.41 ...\n",
      "  .. ..$ GSM1050968: num [1:17879] 11.74 9.5 5.74 7.55 10.63 ...\n",
      "  .. ..$ GSM1050969: num [1:17879] 11.68 9.64 6.26 7.23 10.68 ...\n",
      "  .. ..$ GSM1050970: num [1:17879] 11.23 9.36 5.36 7.08 10 ...\n",
      "  .. ..$ GSM1050971: num [1:17879] 11.77 9.9 5.57 7.58 10.29 ...\n",
      "  .. ..$ GSM1050972: num [1:17879] 11.62 9.29 5.57 7.47 11.02 ...\n",
      "  .. ..$ GSM1050973: num [1:17879] 11.7 9.7 5.5 7.4 10.7 ...\n",
      "  .. ..$ GSM1050974: num [1:17879] 11.15 10.45 5.88 7.29 9.54 ...\n",
      "  .. ..$ GSM1050975: num [1:17879] 11.92 9.71 5.26 7.5 10.97 ...\n",
      "  .. ..$ GSM1050976: num [1:17879] 11.91 9.65 5.13 7.18 10.92 ...\n",
      "  .. ..$ GSM1050977: num [1:17879] 12.02 10.08 5.46 7.55 10.8 ...\n",
      "  .. ..$ GSM1050978: num [1:17879] 11.65 9.15 5.3 7.44 10.88 ...\n",
      "  .. ..$ GSM1050979: num [1:17879] 11.27 9.29 5.21 7.09 10.24 ...\n",
      "  .. ..$ GSM1050980: num [1:17879] 11.77 10.57 5.67 7.56 10.43 ...\n",
      "  .. ..$ GSM1050981: num [1:17879] 12.23 9.84 5.66 7.66 12.07 ...\n",
      "  .. ..$ GSM1050982: num [1:17879] 11.62 9.89 5.44 7.69 10.57 ...\n",
      "  .. ..$ GSM1050983: num [1:17879] 12.13 9.96 6.02 7.71 11.34 ...\n",
      "  .. ..$ GSM1050984: num [1:17879] 11.98 9.32 5.58 7.37 11.35 ...\n",
      "  .. ..$ GSM1050985: num [1:17879] 11.95 10.01 5.42 7.79 11.36 ...\n",
      "  .. ..$ GSM1050986: num [1:17879] 11.44 9.28 5.54 7.16 10.6 ...\n",
      "  .. ..$ GSM1050987: num [1:17879] 12.01 9.77 5.62 7.55 11.55 ...\n",
      "  .. ..$ GSM1050988: num [1:17879] 11.74 9.82 5.34 7.69 10.8 ...\n",
      "  .. ..$ GSM1050989: num [1:17879] 11.3 9.45 5.33 7.27 10.81 ...\n",
      "  .. ..$ GSM1050990: num [1:17879] 11.84 11.05 5.22 7.86 11.23 ...\n",
      "  .. ..$ GSM1050991: num [1:17879] 11.72 9.05 5.51 7.23 11.08 ...\n",
      "  .. ..$ GSM1050992: num [1:17879] 12.21 9.74 5.95 7.81 12.05 ...\n",
      "  .. ..$ GSM1050993: num [1:17879] 11.58 10.13 5.42 7.61 11.03 ...\n",
      "  .. ..$ GSM1050994: num [1:17879] 11.42 9.83 5.08 7.32 10.44 ...\n",
      "  .. ..$ GSM1050995: num [1:17879] 11.62 9.66 5.38 7.21 11.34 ...\n",
      "  .. ..$ GSM1050996: num [1:17879] 11.61 9.97 5.21 7.39 10.67 ...\n",
      "  .. ..$ GSM1050997: num [1:17879] 11.68 10.05 5.1 7.47 11.29 ...\n",
      "  .. ..$ GSM1050998: num [1:17879] 12.16 9.95 5.45 7.5 11.71 ...\n",
      "  .. ..$ GSM1050999: num [1:17879] 11.46 9.09 5.49 6.93 11.21 ...\n",
      "  .. ..$ GSM1051000: num [1:17879] 12.09 9.65 5.53 7.41 12.23 ...\n",
      "  .. ..$ GSM1051001: num [1:17879] 10.96 9.31 5.15 6.79 10.29 ...\n",
      "  .. ..$ GSM1051002: num [1:17879] 11.17 9.27 5.05 7.32 10.5 ...\n",
      "  .. ..$ GSM1051003: num [1:17879] 11.8 9.71 5.5 7.44 10.97 ...\n",
      "  .. ..$ GSM1051004: num [1:17879] 12.22 10.48 5.65 7.75 11.82 ...\n",
      "  .. ..$ GSM1051005: num [1:17879] 11.32 10.39 5.8 7.01 10.49 ...\n",
      "  .. ..$ GSM1051006: num [1:17879] 11.06 9.09 5.45 6.92 10.81 ...\n",
      "  .. ..$ GSM1051007: num [1:17879] 11.69 9.49 5.07 7.16 10.89 ...\n",
      "  .. ..$ GSM1051008: num [1:17879] 12.03 10.16 5.49 7.37 11.45 ...\n",
      "  .. ..$ GSM1051009: num [1:17879] 11.27 9.4 5.34 7.05 10.85 ...\n",
      "  .. ..$ GSM1051010: num [1:17879] 11.88 9.26 5.65 7.49 11.34 ...\n",
      "  .. ..$ GSM1051011: num [1:17879] 11.98 9.73 5.19 7.69 11.67 ...\n",
      "  .. ..$ GSM1051012: num [1:17879] 11.5 9.15 5.04 7.67 11.67 ...\n",
      "  .. ..$ GSM1051013: num [1:17879] 11.85 9.41 5.23 7.39 11.32 ...\n",
      "  .. ..$ GSM1051014: num [1:17879] 10.98 9.35 5.6 7.06 10.55 ...\n",
      "  .. ..$ GSM1051015: num [1:17879] 11.91 9.76 5.42 7.76 11.81 ...\n",
      "  .. ..$ GSM1051016: num [1:17879] 11.58 9.18 5.65 7.11 10.9 ...\n",
      "  .. ..$ GSM1051017: num [1:17879] 11.82 9.57 5.62 7.28 11.44 ...\n",
      "  .. ..$ GSM1051018: num [1:17879] 11.7 11 5.5 7.6 11.5 ...\n",
      "  .. ..$ GSM1051019: num [1:17879] 11.03 10.76 5.77 7.72 10.29 ...\n",
      "  .. ..$ GSM1051020: num [1:17879] 11.6 9.48 5.15 7.49 11.51 ...\n",
      "  .. ..$ GSM1051021: num [1:17879] 11.52 9.81 5.04 7.43 9.95 ...\n",
      "  .. ..$ GSM1051022: num [1:17879] 11.86 9.64 5.21 7.46 11.02 ...\n",
      "  ..$ GSE42831:'data.frame':\t17879 obs. of  17 variables:\n",
      "  .. ..$ GSM1051023: num [1:17879] 10.04 8.98 5.34 6.9 8.98 ...\n",
      "  .. ..$ GSM1051024: num [1:17879] 11.72 9.45 5.41 7.72 10.53 ...\n",
      "  .. ..$ GSM1051025: num [1:17879] 11.95 9.88 5.33 7.36 10.63 ...\n",
      "  .. ..$ GSM1051026: num [1:17879] 10.61 9.25 4.93 7.12 9.43 ...\n",
      "  .. ..$ GSM1051027: num [1:17879] 12.41 10.38 5.12 7.77 11.51 ...\n",
      "  .. ..$ GSM1051028: num [1:17879] 10.67 9.55 5.13 7.32 9.41 ...\n",
      "  .. ..$ GSM1051029: num [1:17879] 12.34 9.81 5.08 7.62 11.52 ...\n",
      "  .. ..$ GSM1051030: num [1:17879] 11.95 9.97 4.93 7.74 10.38 ...\n",
      "  .. ..$ GSM1051031: num [1:17879] 12.56 10.33 5.06 8.02 11.18 ...\n",
      "  .. ..$ GSM1051032: num [1:17879] 11.66 9.71 5.19 7.09 10.38 ...\n",
      "  .. ..$ GSM1051033: num [1:17879] 11.47 9.41 4.94 7.07 9.74 ...\n",
      "  .. ..$ GSM1051034: num [1:17879] 12.04 9.79 5.17 7.48 10.18 ...\n",
      "  .. ..$ GSM1051035: num [1:17879] 12.13 9.62 5.17 7.51 10.8 ...\n",
      "  .. ..$ GSM1051036: num [1:17879] 11.2 9.05 4.77 6.92 9.81 ...\n",
      "  .. ..$ GSM1051037: num [1:17879] 12.16 9.78 4.85 7.58 10.94 ...\n",
      "  .. ..$ GSM1051038: num [1:17879] 11.34 9.53 5.17 7.28 10.25 ...\n",
      "  .. ..$ GSM1051039: num [1:17879] 10.57 9.86 4.85 7.27 9.55 ...\n",
      "  ..$ GSE42832:'data.frame':\t12296 obs. of  90 variables:\n",
      "  .. ..$ GSM1051040: num [1:12296] 11.3 8.52 5.44 5.97 8.67 ...\n",
      "  .. ..$ GSM1051041: num [1:12296] 12.17 9.15 6.11 6.32 10.73 ...\n",
      "  .. ..$ GSM1051042: num [1:12296] 11.31 9.37 5.79 5.87 10.12 ...\n",
      "  .. ..$ GSM1051043: num [1:12296] 11.4 8.27 5.91 5.9 10.08 ...\n",
      "  .. ..$ GSM1051044: num [1:12296] 11.89 8.85 5.77 6.33 10.49 ...\n",
      "  .. ..$ GSM1051045: num [1:12296] 11.54 9.11 5.85 6.21 9.73 ...\n",
      "  .. ..$ GSM1051046: num [1:12296] 11.08 8.75 5.28 5.97 9.04 ...\n",
      "  .. ..$ GSM1051047: num [1:12296] 11.24 7.87 5.9 5.85 9.59 ...\n",
      "  .. ..$ GSM1051048: num [1:12296] 10.96 7.89 5.79 5.38 9.15 ...\n",
      "  .. ..$ GSM1051049: num [1:12296] 11.55 9.35 5.85 5.8 10.02 ...\n",
      "  .. ..$ GSM1051050: num [1:12296] 11.86 8.86 5.84 6 10.31 ...\n",
      "  .. ..$ GSM1051051: num [1:12296] 8.93 8.29 5.19 5.46 6.56 ...\n",
      "  .. ..$ GSM1051052: num [1:12296] 11.97 9.1 6.46 6.28 10.51 ...\n",
      "  .. ..$ GSM1051053: num [1:12296] 11.04 8.34 5.58 6.11 9.33 ...\n",
      "  .. ..$ GSM1051054: num [1:12296] 11.05 8.86 5.77 5.68 9.83 ...\n",
      "  .. ..$ GSM1051055: num [1:12296] 11.7 8.26 6.34 5.91 10.49 ...\n",
      "  .. ..$ GSM1051056: num [1:12296] 11.77 8.47 5.94 5.7 10.26 ...\n",
      "  .. ..$ GSM1051057: num [1:12296] 8.39 8.19 5.08 5.83 6.11 ...\n",
      "  .. ..$ GSM1051058: num [1:12296] 12.58 9.14 6.76 6.34 10.93 ...\n",
      "  .. ..$ GSM1051059: num [1:12296] 11.82 8.54 6.39 5.98 10.27 ...\n",
      "  .. ..$ GSM1051060: num [1:12296] 9.22 8.37 5.69 5.59 6.83 ...\n",
      "  .. ..$ GSM1051061: num [1:12296] 11.9 9.72 5.47 6.36 9.95 ...\n",
      "  .. ..$ GSM1051062: num [1:12296] 10.55 7.88 5.63 5.68 8.44 ...\n",
      "  .. ..$ GSM1051063: num [1:12296] 11.57 9 6.25 6.13 9.42 ...\n",
      "  .. ..$ GSM1051064: num [1:12296] 12.24 9.4 6.28 6.13 10.52 ...\n",
      "  .. ..$ GSM1051065: num [1:12296] 11.82 8.95 5.71 5.91 10.18 ...\n",
      "  .. ..$ GSM1051066: num [1:12296] 11.91 9.57 6.49 6.83 10.68 ...\n",
      "  .. ..$ GSM1051067: num [1:12296] 11.78 9.8 5.73 7.18 10.52 ...\n",
      "  .. ..$ GSM1051068: num [1:12296] 12.26 9.21 5.72 6.41 10.46 ...\n",
      "  .. ..$ GSM1051069: num [1:12296] 12.44 9.39 6.72 6.23 10.68 ...\n",
      "  .. ..$ GSM1051070: num [1:12296] 11.19 8.12 5.77 5.6 9.9 ...\n",
      "  .. ..$ GSM1051071: num [1:12296] 11.21 8.63 5.5 5.76 10.15 ...\n",
      "  .. ..$ GSM1051072: num [1:12296] 12.71 9.14 6.85 6.36 10.93 ...\n",
      "  .. ..$ GSM1051073: num [1:12296] 11.98 10.06 5.38 6.96 10.4 ...\n",
      "  .. ..$ GSM1051074: num [1:12296] 9.3 8.19 5.81 5.81 6.83 ...\n",
      "  .. ..$ GSM1051075: num [1:12296] 11.48 8.68 5.87 5.75 10.24 ...\n",
      "  .. ..$ GSM1051076: num [1:12296] 12.06 10.03 5.35 6.6 10.37 ...\n",
      "  .. ..$ GSM1051077: num [1:12296] 12.19 9.59 5.35 6.67 10.61 ...\n",
      "  .. ..$ GSM1051078: num [1:12296] 11.3 9.16 5.24 6.04 9.85 ...\n",
      "  .. ..$ GSM1051079: num [1:12296] 11.55 8.34 5.72 5.8 10.06 ...\n",
      "  .. ..$ GSM1051080: num [1:12296] 10.66 8.28 5.73 5.68 8.54 ...\n",
      "  .. ..$ GSM1051081: num [1:12296] 9.82 8.98 5.68 6.4 7.91 ...\n",
      "  .. ..$ GSM1051082: num [1:12296] 11.79 9.03 5.68 5.72 10.33 ...\n",
      "  .. ..$ GSM1051083: num [1:12296] 8.28 7.86 5.24 5.53 6.57 ...\n",
      "  .. ..$ GSM1051084: num [1:12296] 12.25 9.62 5.82 7.07 10.44 ...\n",
      "  .. ..$ GSM1051085: num [1:12296] 11.93 9.14 6.28 6.03 10.74 ...\n",
      "  .. ..$ GSM1051086: num [1:12296] 9.08 8.75 5.84 6.03 7.19 ...\n",
      "  .. ..$ GSM1051087: num [1:12296] 11.15 9.1 5.89 6.38 9.51 ...\n",
      "  .. ..$ GSM1051088: num [1:12296] 11.56 8.69 5.62 5.86 9.97 ...\n",
      "  .. ..$ GSM1051089: num [1:12296] 8.63 7.83 5.4 5.45 6.44 ...\n",
      "  .. ..$ GSM1051090: num [1:12296] 11.04 8.37 5.59 5.87 9.26 ...\n",
      "  .. ..$ GSM1051091: num [1:12296] 11.74 8.84 6.01 6.15 10.95 ...\n",
      "  .. ..$ GSM1051092: num [1:12296] 11.77 9.54 5.62 6.28 10.64 ...\n",
      "  .. ..$ GSM1051093: num [1:12296] 10 7.48 4.78 5.64 7.76 ...\n",
      "  .. ..$ GSM1051094: num [1:12296] 11.31 9.53 5 6.64 9.47 ...\n",
      "  .. ..$ GSM1051095: num [1:12296] 11.85 8.46 6.05 5.91 10.42 ...\n",
      "  .. ..$ GSM1051096: num [1:12296] 11.71 8.34 6.01 5.88 10.21 ...\n",
      "  .. ..$ GSM1051097: num [1:12296] 12.71 9.72 6.76 6.65 11.28 ...\n",
      "  .. ..$ GSM1051098: num [1:12296] 11.92 9.29 5.43 6.09 10.22 ...\n",
      "  .. ..$ GSM1051099: num [1:12296] 11.85 9.65 5.8 6.33 9.66 ...\n",
      "  .. ..$ GSM1051100: num [1:12296] 12.55 8.55 5.17 5.92 11.06 ...\n",
      "  .. ..$ GSM1051101: num [1:12296] 11.67 9.82 5.76 6.92 10.45 ...\n",
      "  .. ..$ GSM1051102: num [1:12296] 11.69 9.5 5.5 6.47 10.18 ...\n",
      "  .. ..$ GSM1051103: num [1:12296] 9.61 8.67 5.49 5.82 6.73 ...\n",
      "  .. ..$ GSM1051104: num [1:12296] 11.9 8.95 5.97 6.29 10.2 ...\n",
      "  .. ..$ GSM1051105: num [1:12296] 12.25 9.13 5.73 6.05 10.66 ...\n",
      "  .. ..$ GSM1051106: num [1:12296] 11.12 8.45 5.74 6.04 9.42 ...\n",
      "  .. ..$ GSM1051107: num [1:12296] 12.29 8.74 6.6 5.93 11.06 ...\n",
      "  .. ..$ GSM1051108: num [1:12296] 11.8 9.69 6.04 6.41 10.54 ...\n",
      "  .. ..$ GSM1051109: num [1:12296] 11.39 8.64 5.76 5.64 9.78 ...\n",
      "  .. ..$ GSM1051110: num [1:12296] 11.05 9.15 5.07 5.75 9.74 ...\n",
      "  .. ..$ GSM1051111: num [1:12296] 9.48 9.18 5.64 6.01 8.14 ...\n",
      "  .. ..$ GSM1051112: num [1:12296] 8.58 8.26 5.12 5.71 6.74 ...\n",
      "  .. ..$ GSM1051113: num [1:12296] 11.5 7.76 6 5.79 9.91 ...\n",
      "  .. ..$ GSM1051114: num [1:12296] 11.1 8.88 6.09 5.99 9.22 ...\n",
      "  .. ..$ GSM1051115: num [1:12296] 8.94 8.29 4.94 5.56 6.53 ...\n",
      "  .. ..$ GSM1051116: num [1:12296] 12.09 8.95 6.11 5.97 10.69 ...\n",
      "  .. ..$ GSM1051117: num [1:12296] 12 9.19 6.5 6.05 10.79 ...\n",
      "  .. ..$ GSM1051118: num [1:12296] 9.87 8.3 5.89 6.26 6.85 ...\n",
      "  .. ..$ GSM1051119: num [1:12296] 9.66 8.14 5.33 6.01 7.21 ...\n",
      "  .. ..$ GSM1051120: num [1:12296] 11.59 8.52 5.09 5.97 9.83 ...\n",
      "  .. ..$ GSM1051121: num [1:12296] 11.83 9.98 5.55 6.6 9.89 ...\n",
      "  .. ..$ GSM1051122: num [1:12296] 11.1 8.79 4.96 5.84 9.82 ...\n",
      "  .. ..$ GSM1051123: num [1:12296] 9.46 8.22 5.44 5.88 7.63 ...\n",
      "  .. ..$ GSM1051124: num [1:12296] 11.08 8.47 5.93 6.09 9.37 ...\n",
      "  .. ..$ GSM1051125: num [1:12296] 11.34 9.35 5.38 6.57 10.25 ...\n",
      "  .. ..$ GSM1051126: num [1:12296] 10.82 8.52 5.27 5.94 9.35 ...\n",
      "  .. ..$ GSM1051127: num [1:12296] 11.64 9.42 5.31 6.56 10.37 ...\n",
      "  .. ..$ GSM1051128: num [1:12296] 12.04 10.12 5.71 7.03 10.58 ...\n",
      "  .. ..$ GSM1051129: num [1:12296] 11.49 8.66 5.89 5.71 9.1 ...\n",
      " $ pheno.list:List of 10\n",
      "  ..$ GSE19491:'data.frame':\t498 obs. of  68 variables:\n",
      "  .. ..$ title                                : Factor w/ 498 levels \"CON_LON_test035\",..: 2 8 14 20 26 28 30 32 36 34 ...\n",
      "  .. ..$ geo_accession                        : Factor w/ 498 levels \"GSM484368\",\"GSM484369\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                               : Factor w/ 1 level \"Public on Aug 11 2010\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date                      : Factor w/ 4 levels \"Dec 11 2009\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date                     : Factor w/ 2 levels \"Aug 11 2010\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                                 : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count                        : int [1:498] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1                      : Factor w/ 16 levels \"CD4+ cells from healthy control\",..: 10 10 10 10 10 10 10 10 10 10 ...\n",
      "  .. ..$ organism_ch1                         : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1                  : Factor w/ 85 levels \"age: 0\",\"age: 1\",..: 26 16 57 46 36 26 20 28 18 49 ...\n",
      "  .. ..$ characteristics_ch1.1                : Factor w/ 2 levels \"gender: Female\",..: 2 2 1 1 1 1 1 2 1 2 ...\n",
      "  .. ..$ characteristics_ch1.2                : Factor w/ 13 levels \"ethnicity: Afican American\",..: 13 13 12 12 6 6 13 11 4 6 ...\n",
      "  .. ..$ characteristics_ch1.3                : Factor w/ 15 levels \"healthy control: ASLE\",..: 6 6 6 6 6 6 6 6 6 6 ...\n",
      "  .. ..$ characteristics_ch1.4                : Factor w/ 6 levels \"\",\"geographical region: London\",..: 4 4 4 4 4 4 4 4 4 4 ...\n",
      "  .. ..$ characteristics_ch1.5                : Factor w/ 9 levels \"\",\"bcg vaccinated: No\",..: 6 6 6 6 6 6 6 6 6 6 ...\n",
      "  .. ..$ characteristics_ch1.6                : Factor w/ 7 levels \"\",\"isolate sensitivity: Fully_Sensitive\",..: 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ characteristics_ch1.7                : Factor w/ 34 levels \"\",\"bcg vaccinated: No\",..: 3 3 3 3 3 3 3 3 3 3 ...\n",
      "  .. ..$ characteristics_ch1.8                : Factor w/ 12 levels \"\",\"exposure_latent: Contact_Screen\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.9                : Factor w/ 8 levels \"\",\"index_case_disease_site: \",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.10               : Factor w/ 6 levels \"\",\"smear of index case: \",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.11               : Factor w/ 7 levels \"\",\"modal x-ray grade: \",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.12               : Factor w/ 7 levels \"\",\"sputum_smear: +\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.13               : Factor w/ 4 levels \"\",\"sputum_culture: n.a\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.14               : Factor w/ 7 levels \"\",\"bal_smear: +\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.15               : Factor w/ 4 levels \"\",\"bal_culture: n.a\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.16               : Factor w/ 8 levels \"\",\"isolate_sensitivity: Fully_sensitive\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1                         : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1                 : Factor w/ 5 levels \"Ambion Temput Isolation kit\",..: 5 5 5 5 5 5 5 5 5 5 ...\n",
      "  .. ..$ label_ch1                            : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1                   : Factor w/ 3 levels \"Biotinylated cRNA were generated following manufacturer protocols using the Illumina TotalPrep Amplification Kit from Ambion\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1                            : int [1:498] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol                         : Factor w/ 2 levels \"Hybridize 12-Sample Beadchip\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol                        : Factor w/ 1 level \"SentrixScanVersion 2.3.0.13  (This information is in the .XML file)\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ description                          : Factor w/ 26 levels \"CD4+ cells from healthy control recruited in London. The subject was negative for skin test and IGRA test.\",..: 11 11 11 11 11 11 10 11 10 11 ...\n",
      "  .. ..$ data_processing                      : Factor w/ 2 levels \"The data were analysed using BeadStudio version 1.5.1.3 using the average chip normalization method.\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id                          : Factor w/ 1 level \"GPL6947\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name                         : Factor w/ 1 level \"Damien,,Chaussabel\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_email                        : Factor w/ 1 level \"DChaussabel@benaroyaresearch.org\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute                    : Factor w/ 1 level \"Baylor Institute for Immunology Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address                      : Factor w/ 1 level \"3434 Live Oak\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city                         : Factor w/ 1 level \"Dallas\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_state                        : Factor w/ 1 level \"TX\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code              : int [1:498] 75204 75204 75204 75204 75204 75204 75204 75204 75204 75204 ...\n",
      "  .. ..$ contact_country                      : Factor w/ 1 level \"USA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file                   : Factor w/ 498 levels \"ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484368/suppl/GSM484368_CON_LON_test035_long_0_022.txt.gz\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ data_row_count                       : int [1:498] 48803 48803 48803 48803 48803 48803 48803 48803 48803 48803 ...\n",
      "  .. ..$ relation                             : Factor w/ 98 levels \"\",\"Reanalyzed by: GSM2886035\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ age.ch1                              : Factor w/ 85 levels \"0\",\"1\",\"10\",\"11\",..: 26 16 57 46 36 26 20 28 18 49 ...\n",
      "  .. ..$ bal_culture.ch1                      : Factor w/ 3 levels \"n.a\",\"no_growth_6_weeks\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ bal_smear.ch1                        : Factor w/ 6 levels \"+\",\"++\",\"+++\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ bcg.vaccinated.ch1                   : Factor w/ 4 levels \"No\",\"Not Known\",..: 4 4 4 4 4 4 4 4 4 4 ...\n",
      "  .. ..$ ethnicity.ch1                        : Factor w/ 13 levels \"Afican American\",..: 13 13 12 12 6 6 13 11 4 6 ...\n",
      "  .. ..$ exposure_latent.ch1                  : Factor w/ 5 levels \"Contact_Screen\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ gender.ch1                           : Factor w/ 2 levels \"Female\",\"Male\": 2 2 1 1 1 1 1 2 1 2 ...\n",
      "  .. ..$ geographical.region.ch1              : Factor w/ 2 levels \"London\",\"South Africa\": NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ healthy.control.ch1                  : Factor w/ 4 levels \"ASLE\",\"pSLE\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ illness.ch1                          : Factor w/ 11 levels \"ASLE\",\"Control\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. ..$ index_case_disease_site.ch1          : Factor w/ 5 levels \"\",\"extrapulm\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ isolate.sensitivity.ch1              : Factor w/ 3 levels \"Fully_Sensitive\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. ..$ isolate_sensitivity.ch1              : Factor w/ 7 levels \"Fully_sensitive\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ modal.x.ray.grade.ch1                : Factor w/ 6 levels \"\",\"Advanced\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ region.of.birth.ch1                  : Factor w/ 3 levels \"non_UK_born\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ smear.of.index.case.ch1              : Factor w/ 5 levels \"\",\"n.a\",\"Sm-\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ sputum.smear.status.ch1              : Factor w/ 4 levels \"n.a\",\"Negative\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ sputum_culture.ch1                   : Factor w/ 4 levels \"M.tuberculosis\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ sputum_smear.ch1                     : Factor w/ 11 levels \"+\",\"++\",\"+++\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. ..$ time.post.initiation.of.treatment.ch1: Factor w/ 3 levels \"0_months\",\"12_months\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ tst.ch1                              : Factor w/ 31 levels \"0\",\"10\",\"11\",..: NA NA NA NA NA NA NA NA NA NA ...\n",
      "  ..$ GSE37250:'data.frame':\t537 obs. of  38 variables:\n",
      "  .. ..$ title                  : Factor w/ 537 levels \"active tuberculosis HIV negative Malawi WB_1_M_39619\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 537 levels \"GSM914353\",\"GSM914354\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 23 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Apr 13 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Dec 17 2018\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:537] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 3 levels \"Whole Blood from patient with Active TB\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 3 levels \"disease state: active tuberculosis\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 2 levels \"hiv status: HIV negative\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 2 levels \"geographical region: Malawi\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 1 level \"tissue: whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"Standard PAXgene blood RNA extraction kit protocol (Qiagen).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:537] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridisation protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ description            : Factor w/ 537 levels \"Sample name: 5483347021_A\",..: 1 13 25 81 92 68 103 115 127 139 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using Genome Studio, with background subtracted and quantile normalised.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Victoria,,Wright\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_email          : Factor w/ 1 level \"v.wright@imperial.ac.uk\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_laboratory     : Factor w/ 1 level \"Wright Fleming Institute\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_department     : Factor w/ 1 level \"Paediatrics\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"Imperial College London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"Norfolk Place\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"W2 1PG\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:537] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 3 levels \"active tuberculosis\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ geographical.region.ch1: Factor w/ 2 levels \"Malawi\",\"South Africa\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ hiv.status.ch1         : Factor w/ 2 levels \"HIV negative\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE39939:'data.frame':\t157 obs. of  35 variables:\n",
      "  .. ..$ title                  : Factor w/ 157 levels \"WB_20_KIL_100091\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 157 levels \"GSM1249177\",\"GSM1249178\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on May 01 2014\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Oct 23 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"May 01 2014\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:157] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 3 levels \"Whole Blood from patient with Active TB\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 5 levels \"illness: active tuberculosis (culture confirmed)\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 2 levels \"hiv status: HIV negative\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 1 level \"geographical region: Kenya\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"Standard PAXgene blood RNA extraction kit protocol (Qiagen).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:157] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridisation protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using Genome Studio, with background subtracted and quantile normalised.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Victoria,,Wright\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_email          : Factor w/ 1 level \"v.wright@imperial.ac.uk\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_laboratory     : Factor w/ 1 level \"Wright Fleming Institute\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_department     : Factor w/ 1 level \"Paediatrics\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"Imperial College London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"Norfolk Place\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"W2 1PG\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:157] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ geographical.region.ch1: Factor w/ 1 level \"Kenya\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ hiv.status.ch1         : Factor w/ 2 levels \"HIV negative\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. ..$ illness.ch1            : Factor w/ 5 levels \"active tuberculosis (culture confirmed)\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE39940:'data.frame':\t334 obs. of  37 variables:\n",
      "  .. ..$ title                  : Factor w/ 334 levels \"WB_20_M_40012\",..: 145 146 147 148 149 150 151 152 153 154 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 334 levels \"GSM981627\",\"GSM981628\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on May 01 2014\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Aug 07 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"May 01 2014\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:334] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 3 levels \"Whole Blood from patient with Active TB\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 3 levels \"disease status: active tuberculosis\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 2 levels \"hiv status: HIV negative\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 2 levels \"geographical region: Malawi\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 334 levels \"barcode: 6116725094_A\",..: 236 63 64 53 54 55 56 228 229 230 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"Standard PAXgene blood RNA extraction kit protocol (Qiagen).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:334] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridisation protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using Genome Studio, with background subtracted and quantile normalised.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Victoria,,Wright\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_email          : Factor w/ 1 level \"v.wright@imperial.ac.uk\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_laboratory     : Factor w/ 1 level \"Wright Fleming Institute\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_department     : Factor w/ 1 level \"Paediatrics\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"Imperial College London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"Norfolk Place\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"W2 1PG\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:334] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ barcode.ch1            : Factor w/ 334 levels \"6116725094_A\",..: 236 63 64 53 54 55 56 228 229 230 ...\n",
      "  .. ..$ disease.status.ch1     : Factor w/ 3 levels \"active tuberculosis\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ geographical.region.ch1: Factor w/ 2 levels \"Malawi\",\"South Africa\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ hiv.status.ch1         : Factor w/ 2 levels \"HIV negative\",..: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE42825:'data.frame':\t42 obs. of  37 variables:\n",
      "  .. ..$ title                  : Factor w/ 42 levels \"Control Sample 10 validation set\",..: 35 25 36 37 38 39 40 41 42 1 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 42 levels \"GSM1050732\",\"GSM1050733\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:42] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 1 level \"Whole Blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 3 levels \"gender: F\",\"gender: M\",..: 1 1 2 2 1 1 1 1 2 1 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 5 levels \"ethnicity: Black\",..: 3 1 1 5 5 3 5 5 3 1 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 4 levels \"disease state: Active sarcoidosis\",..: 4 1 4 4 4 4 4 4 4 2 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 1 level \"tissue: whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Direct ex-vivo human blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Blood was taken into Tempus and frozen\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNon-active sarcoidosis was extracted using the MagMAX-96 Blood RNon-active sarcoidosis Isolation Kit and globi\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified cRNon-active sarcoidosis were prepared using the IllumiNon-active sarcoidosis CustomPre\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:42] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard IllumiNon-active sarcoidosis hybridization protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard IllumiNon-active sarcoidosis scanning protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using 75th percentile normalisation in GeneSpring 11.5\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing.1      : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:42] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 4 levels \"Active sarcoidosis\",..: 4 1 4 4 4 4 4 4 4 2 ...\n",
      "  .. ..$ ethnicity.ch1          : Factor w/ 5 levels \"Black\",\"Central Asia\",..: 3 1 1 5 5 3 5 5 3 1 ...\n",
      "  .. ..$ gender.ch1             : Factor w/ 3 levels \"F\",\"M\",\"None\": 1 1 2 2 1 1 1 1 2 1 ...\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE42826:'data.frame':\t102 obs. of  36 variables:\n",
      "  .. ..$ title                  : Factor w/ 102 levels \"Cancer Sample 22 test set\",..: 67 74 2 62 83 85 87 51 60 68 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 102 levels \"GSM1050774\",\"GSM1050775\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:102] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 1 level \"Whole Blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 2 levels \"gender: F\",\"gender: M\": 2 2 1 2 1 2 1 1 1 2 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 5 levels \"ethnicity: Black\",..: 5 1 5 2 5 5 5 5 5 5 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 6 levels \"disease state: Active Sarcoid\",..: 4 1 3 5 1 1 4 2 2 4 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 1 level \"tissue: whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Direct ex-vivo human blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Blood was taken into Tempus and frozen\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNon-active sarcoidosis was extracted using the MagMAX-96 Blood RNon-active sarcoidosis Isolation Kit and globi\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified cRNon-active sarcoidosis were prepared using the IllumiNon-active sarcoidosis CustomPre\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:102] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard IllumiNon-active sarcoidosis hybridization protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard IllumiNon-active sarcoidosis scanning protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:102] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 6 levels \"Active Sarcoid\",..: 4 1 3 5 1 1 4 2 2 4 ...\n",
      "  .. ..$ ethnicity.ch1          : Factor w/ 5 levels \"Black\",\"Indian subcontinent\",..: 5 1 5 2 5 5 5 5 5 5 ...\n",
      "  .. ..$ gender.ch1             : Factor w/ 2 levels \"F\",\"M\": 2 2 1 2 1 2 1 1 1 2 ...\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE42827:'data.frame':\t10 obs. of  38 variables:\n",
      "  .. ..$ title                  : Factor w/ 10 levels \"Pneumonia_Pre-treatment 2101\",..: 10 5 2 4 3 8 6 7 9 1\n",
      "  .. ..$ geo_accession          : Factor w/ 10 levels \"GSM1050876\",\"GSM1050877\",..: 1 2 3 4 5 6 7 8 9 10\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ channel_count          : int [1:10] 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ source_name_ch1        : Factor w/ 1 level \"Whole Blood\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 2 levels \"treatment: Pre-treatment\",..: 2 1 1 1 1 2 2 2 2 1\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 5 levels \"subject: 2101\",..: 5 5 2 4 3 3 1 2 4 1\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 1 level \"tissue: whole blood\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 1 level \"disease state: pneumonia\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Direct ex-vivo human blood\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Blood was taken into Tempus and frozen\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNA was extracted using the MagMAX-96 Blood RNA Isolation Kit and globin reduced using GLOBINclear\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified cRNA were prepared using the Illumina CustomPrep RNA amplification kit\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ taxid_ch1              : int [1:10] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridization protocol\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ description            : Factor w/ 10 levels \"Sample 1\",\"Sample 10\",..: 1 3 4 5 6 7 8 9 10 2\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using 75th percentile normalisation in GeneSpring 11.5\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ data_processing.1      : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ data_row_count         : int [1:10] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 1 level \"pneumonia\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ subject.ch1            : int [1:10] 2132 2132 2102 2121 2105 2105 2101 2102 2121 2101\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 1 1 1 1 1 1 1 1 1\n",
      "  .. ..$ treatment.ch1          : Factor w/ 2 levels \"Pre-treatment\",..: 2 1 1 1 1 2 2 2 2 1\n",
      "  ..$ GSE42830:'data.frame':\t95 obs. of  36 variables:\n",
      "  .. ..$ title                  : Factor w/ 95 levels \"Cancer Sample 2 training set\",..: 9 1 61 62 24 73 91 38 7 10 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 95 levels \"GSM1050928\",\"GSM1050929\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:95] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 1 level \"Whole Blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 3 levels \"gender: F\",\"gender: M\",..: 1 1 2 1 2 2 2 1 2 1 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 4 levels \"ethnicity: Afro-Carribbean\",..: 2 2 1 2 2 2 1 2 1 1 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 6 levels \"disease state: Active Sarcoid\",..: 3 4 1 5 3 5 6 3 4 3 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 1 level \"tissue: whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Direct ex-vivo human blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Blood was taken into Tempus and frozen\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNA was extracted using the MagMAX-96 Blood RNA Isolation Kit and globin reduced using GLOBINclear\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified cRNA were prepared using the Illumina CustomPrep RNA amplification kit\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:95] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridization protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:95] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 6 levels \"Active Sarcoid\",..: 3 4 1 5 3 5 6 3 4 3 ...\n",
      "  .. ..$ ethnicity.ch1          : Factor w/ 4 levels \"Afro-Carribbean\",..: 2 2 1 2 2 2 1 2 1 1 ...\n",
      "  .. ..$ gender.ch1             : Factor w/ 3 levels \"F\",\"M\",\"None\": 1 1 2 1 2 2 2 1 2 1 ...\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  ..$ GSE42831:'data.frame':\t17 obs. of  37 variables:\n",
      "  .. ..$ title                  : Factor w/ 17 levels \"Sarcoid Sample 1 2100 immunosuppressive treatment time pt. 2\",..: 1 10 11 12 13 14 15 16 17 2 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 17 levels \"GSM1051023\",\"GSM1051024\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:17] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 1 level \"Whole Blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 2 levels \"treament: immunosuppressive treatment\",..: 1 1 1 1 1 1 2 2 2 2 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 3 levels \"response: Good-response\",..: 1 1 2 2 2 1 3 3 3 3 ...\n",
      "  .. ..$ characteristics_ch1.2  : Factor w/ 3 levels \"time (d): 1\",..: 2 3 2 2 2 2 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1.3  : Factor w/ 7 levels \"subject: 2010\",..: 4 2 2 3 7 3 3 2 4 7 ...\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Direct ex-vivo human blood\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Blood was taken into Tempus and frozen\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNA was extracted using the MagMAX-96 Blood RNA Isolation Kit and globin reduced using GLOBINclear\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified cRNA were prepared using the Illumina CustomPrep RNA amplification kit\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:17] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridization protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"The data were normalised using 75th percentile normalisation in GeneSpring 11.5\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing.1      : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:17] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ response.ch1           : Factor w/ 3 levels \"Good-response\",..: 1 1 2 2 2 1 3 3 3 3 ...\n",
      "  .. ..$ subject.ch1            : int [1:17] 2100 2018 2018 2099 6011 2099 2099 2018 2100 6011 ...\n",
      "  .. ..$ time..d..ch1           : int [1:17] 2 3 2 2 2 2 1 1 1 1 ...\n",
      "  .. ..$ treament.ch1           : Factor w/ 2 levels \"immunosuppressive treatment\",..: 1 1 1 1 1 1 2 2 2 2 ...\n",
      "  ..$ GSE42832:'data.frame':\t90 obs. of  33 variables:\n",
      "  .. ..$ title                  : Factor w/ 90 levels \"Control_CD14 Sample 22 cell separation\",..: 86 12 81 48 43 85 59 20 75 76 ...\n",
      "  .. ..$ geo_accession          : Factor w/ 90 levels \"GSM1051040\",\"GSM1051041\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. ..$ status                 : Factor w/ 1 level \"Public on Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ submission_date        : Factor w/ 1 level \"Dec 10 2012\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ last_update_date       : Factor w/ 1 level \"Oct 31 2013\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ type                   : Factor w/ 1 level \"RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ channel_count          : int [1:90] 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ source_name_ch1        : Factor w/ 6 levels \"CD14\",\"CD15\",..: 6 3 5 4 3 5 6 4 3 4 ...\n",
      "  .. ..$ organism_ch1           : Factor w/ 1 level \"Homo sapiens\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ characteristics_ch1    : Factor w/ 3 levels \"disease state: Control\",..: 3 1 3 2 2 3 2 1 3 3 ...\n",
      "  .. ..$ characteristics_ch1.1  : Factor w/ 6 levels \"cell type: CD14\",..: 6 3 5 4 3 5 6 4 3 4 ...\n",
      "  .. ..$ treatment_protocol_ch1 : Factor w/ 1 level \"Whole blood stored in Tempus, PBMCs obtained using Lymphoprep then leucocytes purified using Miltenyi whole blo\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ growth_protocol_ch1    : Factor w/ 1 level \"Direct ex vivo human blood frozen and purified leucocytes after RNA extraction frozen\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ molecule_ch1           : Factor w/ 1 level \"total RNA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ extract_protocol_ch1   : Factor w/ 1 level \"RNA was extracted using the Qiagen RNeasy Mini Kit.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_ch1              : Factor w/ 1 level \"Biotin\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ label_protocol_ch1     : Factor w/ 1 level \"Biotinylated, amplified antisense cRNA were prepared using NuGEN WT-Ovation RNA Amplification and Encore BiotinIL Module.\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ taxid_ch1              : int [1:90] 9606 9606 9606 9606 9606 9606 9606 9606 9606 9606 ...\n",
      "  .. ..$ hyb_protocol           : Factor w/ 1 level \"Standard Illumina hybridization protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ scan_protocol          : Factor w/ 1 level \"Standard Illumina scanning protocol\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_processing        : Factor w/ 1 level \"Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have onl\"| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ platform_id            : Factor w/ 1 level \"GPL10558\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_name           : Factor w/ 1 level \"Chloe,,Bloom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_institute      : Factor w/ 1 level \"MRC National Institute for Medical Research\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_address        : Factor w/ 1 level \"The Ridgeway\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_city           : Factor w/ 1 level \"London\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_zip.postal_code: Factor w/ 1 level \"NW7 1AA\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ contact_country        : Factor w/ 1 level \"United Kingdom\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ supplementary_file     : Factor w/ 1 level \"NONE\": 1 1 1 1 1 1 1 1 1 1 ...\n",
      "  .. ..$ data_row_count         : int [1:90] 47323 47323 47323 47323 47323 47323 47323 47323 47323 47323 ...\n",
      "  .. ..$ cell.type.ch1          : Factor w/ 5 levels \"CD14\",\"CD15\",..: NA 3 5 4 3 5 NA 4 3 4 ...\n",
      "  .. ..$ disease.state.ch1      : Factor w/ 3 levels \"Control\",\"Sarcoid\",..: 3 1 3 2 2 3 2 1 3 3 ...\n",
      "  .. ..$ tissue.ch1             : Factor w/ 1 level \"whole blood\": 1 NA NA NA NA NA 1 NA NA NA ...\n"
     ]
    }
   ],
   "source": [
    "str(test.data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Format the test phenotype data\n",
    "\n",
    "\"status\" is the TB variable. \"hiv.status\" is the HIV variable. These are all I care about."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$GSE19491</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'characteristics_ch1.4'</li>\n",
       "\t<li>'characteristics_ch1.5'</li>\n",
       "\t<li>'characteristics_ch1.6'</li>\n",
       "\t<li>'characteristics_ch1.7'</li>\n",
       "\t<li>'characteristics_ch1.8'</li>\n",
       "\t<li>'characteristics_ch1.9'</li>\n",
       "\t<li>'characteristics_ch1.10'</li>\n",
       "\t<li>'characteristics_ch1.11'</li>\n",
       "\t<li>'characteristics_ch1.12'</li>\n",
       "\t<li>'characteristics_ch1.13'</li>\n",
       "\t<li>'characteristics_ch1.14'</li>\n",
       "\t<li>'characteristics_ch1.15'</li>\n",
       "\t<li>'characteristics_ch1.16'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'description'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_email'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_state'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'relation'</li>\n",
       "\t<li>'age.ch1'</li>\n",
       "\t<li>'bal_culture.ch1'</li>\n",
       "\t<li>'bal_smear.ch1'</li>\n",
       "\t<li>'bcg.vaccinated.ch1'</li>\n",
       "\t<li>'ethnicity.ch1'</li>\n",
       "\t<li>'exposure_latent.ch1'</li>\n",
       "\t<li>'gender.ch1'</li>\n",
       "\t<li>'geographical.region.ch1'</li>\n",
       "\t<li>'healthy.control.ch1'</li>\n",
       "\t<li>'illness.ch1'</li>\n",
       "\t<li>'index_case_disease_site.ch1'</li>\n",
       "\t<li>'isolate.sensitivity.ch1'</li>\n",
       "\t<li>'isolate_sensitivity.ch1'</li>\n",
       "\t<li>'modal.x.ray.grade.ch1'</li>\n",
       "\t<li>'region.of.birth.ch1'</li>\n",
       "\t<li>'smear.of.index.case.ch1'</li>\n",
       "\t<li>'sputum.smear.status.ch1'</li>\n",
       "\t<li>'sputum_culture.ch1'</li>\n",
       "\t<li>'sputum_smear.ch1'</li>\n",
       "\t<li>'time.post.initiation.of.treatment.ch1'</li>\n",
       "\t<li>'tst.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE37250</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'description'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_email'</li>\n",
       "\t<li>'contact_laboratory'</li>\n",
       "\t<li>'contact_department'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'geographical.region.ch1'</li>\n",
       "\t<li>'hiv.status.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE39939</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_email'</li>\n",
       "\t<li>'contact_laboratory'</li>\n",
       "\t<li>'contact_department'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'geographical.region.ch1'</li>\n",
       "\t<li>'hiv.status.ch1'</li>\n",
       "\t<li>'illness.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE39940</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_email'</li>\n",
       "\t<li>'contact_laboratory'</li>\n",
       "\t<li>'contact_department'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'barcode.ch1'</li>\n",
       "\t<li>'disease.status.ch1'</li>\n",
       "\t<li>'geographical.region.ch1'</li>\n",
       "\t<li>'hiv.status.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42825</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'data_processing.1'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'ethnicity.ch1'</li>\n",
       "\t<li>'gender.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42826</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'ethnicity.ch1'</li>\n",
       "\t<li>'gender.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42827</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'description'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'data_processing.1'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'subject.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "\t<li>'treatment.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42830</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'ethnicity.ch1'</li>\n",
       "\t<li>'gender.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42831</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'characteristics_ch1.2'</li>\n",
       "\t<li>'characteristics_ch1.3'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'data_processing.1'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'response.ch1'</li>\n",
       "\t<li>'subject.ch1'</li>\n",
       "\t<li>'time..d..ch1'</li>\n",
       "\t<li>'treament.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$GSE42832</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'title'</li>\n",
       "\t<li>'geo_accession'</li>\n",
       "\t<li>'status'</li>\n",
       "\t<li>'submission_date'</li>\n",
       "\t<li>'last_update_date'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'channel_count'</li>\n",
       "\t<li>'source_name_ch1'</li>\n",
       "\t<li>'organism_ch1'</li>\n",
       "\t<li>'characteristics_ch1'</li>\n",
       "\t<li>'characteristics_ch1.1'</li>\n",
       "\t<li>'treatment_protocol_ch1'</li>\n",
       "\t<li>'growth_protocol_ch1'</li>\n",
       "\t<li>'molecule_ch1'</li>\n",
       "\t<li>'extract_protocol_ch1'</li>\n",
       "\t<li>'label_ch1'</li>\n",
       "\t<li>'label_protocol_ch1'</li>\n",
       "\t<li>'taxid_ch1'</li>\n",
       "\t<li>'hyb_protocol'</li>\n",
       "\t<li>'scan_protocol'</li>\n",
       "\t<li>'data_processing'</li>\n",
       "\t<li>'platform_id'</li>\n",
       "\t<li>'contact_name'</li>\n",
       "\t<li>'contact_institute'</li>\n",
       "\t<li>'contact_address'</li>\n",
       "\t<li>'contact_city'</li>\n",
       "\t<li>'contact_zip.postal_code'</li>\n",
       "\t<li>'contact_country'</li>\n",
       "\t<li>'supplementary_file'</li>\n",
       "\t<li>'data_row_count'</li>\n",
       "\t<li>'cell.type.ch1'</li>\n",
       "\t<li>'disease.state.ch1'</li>\n",
       "\t<li>'tissue.ch1'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$GSE19491] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'characteristics\\_ch1.4'\n",
       "\\item 'characteristics\\_ch1.5'\n",
       "\\item 'characteristics\\_ch1.6'\n",
       "\\item 'characteristics\\_ch1.7'\n",
       "\\item 'characteristics\\_ch1.8'\n",
       "\\item 'characteristics\\_ch1.9'\n",
       "\\item 'characteristics\\_ch1.10'\n",
       "\\item 'characteristics\\_ch1.11'\n",
       "\\item 'characteristics\\_ch1.12'\n",
       "\\item 'characteristics\\_ch1.13'\n",
       "\\item 'characteristics\\_ch1.14'\n",
       "\\item 'characteristics\\_ch1.15'\n",
       "\\item 'characteristics\\_ch1.16'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'description'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_email'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_state'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'relation'\n",
       "\\item 'age.ch1'\n",
       "\\item 'bal\\_culture.ch1'\n",
       "\\item 'bal\\_smear.ch1'\n",
       "\\item 'bcg.vaccinated.ch1'\n",
       "\\item 'ethnicity.ch1'\n",
       "\\item 'exposure\\_latent.ch1'\n",
       "\\item 'gender.ch1'\n",
       "\\item 'geographical.region.ch1'\n",
       "\\item 'healthy.control.ch1'\n",
       "\\item 'illness.ch1'\n",
       "\\item 'index\\_case\\_disease\\_site.ch1'\n",
       "\\item 'isolate.sensitivity.ch1'\n",
       "\\item 'isolate\\_sensitivity.ch1'\n",
       "\\item 'modal.x.ray.grade.ch1'\n",
       "\\item 'region.of.birth.ch1'\n",
       "\\item 'smear.of.index.case.ch1'\n",
       "\\item 'sputum.smear.status.ch1'\n",
       "\\item 'sputum\\_culture.ch1'\n",
       "\\item 'sputum\\_smear.ch1'\n",
       "\\item 'time.post.initiation.of.treatment.ch1'\n",
       "\\item 'tst.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE37250] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'description'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_email'\n",
       "\\item 'contact\\_laboratory'\n",
       "\\item 'contact\\_department'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'geographical.region.ch1'\n",
       "\\item 'hiv.status.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE39939] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_email'\n",
       "\\item 'contact\\_laboratory'\n",
       "\\item 'contact\\_department'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'geographical.region.ch1'\n",
       "\\item 'hiv.status.ch1'\n",
       "\\item 'illness.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE39940] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_email'\n",
       "\\item 'contact\\_laboratory'\n",
       "\\item 'contact\\_department'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'barcode.ch1'\n",
       "\\item 'disease.status.ch1'\n",
       "\\item 'geographical.region.ch1'\n",
       "\\item 'hiv.status.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42825] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'data\\_processing.1'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'ethnicity.ch1'\n",
       "\\item 'gender.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42826] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'ethnicity.ch1'\n",
       "\\item 'gender.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42827] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'description'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'data\\_processing.1'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'subject.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\item 'treatment.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42830] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'ethnicity.ch1'\n",
       "\\item 'gender.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42831] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'characteristics\\_ch1.2'\n",
       "\\item 'characteristics\\_ch1.3'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'data\\_processing.1'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'response.ch1'\n",
       "\\item 'subject.ch1'\n",
       "\\item 'time..d..ch1'\n",
       "\\item 'treament.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GSE42832] \\begin{enumerate*}\n",
       "\\item 'title'\n",
       "\\item 'geo\\_accession'\n",
       "\\item 'status'\n",
       "\\item 'submission\\_date'\n",
       "\\item 'last\\_update\\_date'\n",
       "\\item 'type'\n",
       "\\item 'channel\\_count'\n",
       "\\item 'source\\_name\\_ch1'\n",
       "\\item 'organism\\_ch1'\n",
       "\\item 'characteristics\\_ch1'\n",
       "\\item 'characteristics\\_ch1.1'\n",
       "\\item 'treatment\\_protocol\\_ch1'\n",
       "\\item 'growth\\_protocol\\_ch1'\n",
       "\\item 'molecule\\_ch1'\n",
       "\\item 'extract\\_protocol\\_ch1'\n",
       "\\item 'label\\_ch1'\n",
       "\\item 'label\\_protocol\\_ch1'\n",
       "\\item 'taxid\\_ch1'\n",
       "\\item 'hyb\\_protocol'\n",
       "\\item 'scan\\_protocol'\n",
       "\\item 'data\\_processing'\n",
       "\\item 'platform\\_id'\n",
       "\\item 'contact\\_name'\n",
       "\\item 'contact\\_institute'\n",
       "\\item 'contact\\_address'\n",
       "\\item 'contact\\_city'\n",
       "\\item 'contact\\_zip.postal\\_code'\n",
       "\\item 'contact\\_country'\n",
       "\\item 'supplementary\\_file'\n",
       "\\item 'data\\_row\\_count'\n",
       "\\item 'cell.type.ch1'\n",
       "\\item 'disease.state.ch1'\n",
       "\\item 'tissue.ch1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$GSE19491\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'characteristics_ch1.4'\n",
       "15. 'characteristics_ch1.5'\n",
       "16. 'characteristics_ch1.6'\n",
       "17. 'characteristics_ch1.7'\n",
       "18. 'characteristics_ch1.8'\n",
       "19. 'characteristics_ch1.9'\n",
       "20. 'characteristics_ch1.10'\n",
       "21. 'characteristics_ch1.11'\n",
       "22. 'characteristics_ch1.12'\n",
       "23. 'characteristics_ch1.13'\n",
       "24. 'characteristics_ch1.14'\n",
       "25. 'characteristics_ch1.15'\n",
       "26. 'characteristics_ch1.16'\n",
       "27. 'molecule_ch1'\n",
       "28. 'extract_protocol_ch1'\n",
       "29. 'label_ch1'\n",
       "30. 'label_protocol_ch1'\n",
       "31. 'taxid_ch1'\n",
       "32. 'hyb_protocol'\n",
       "33. 'scan_protocol'\n",
       "34. 'description'\n",
       "35. 'data_processing'\n",
       "36. 'platform_id'\n",
       "37. 'contact_name'\n",
       "38. 'contact_email'\n",
       "39. 'contact_institute'\n",
       "40. 'contact_address'\n",
       "41. 'contact_city'\n",
       "42. 'contact_state'\n",
       "43. 'contact_zip.postal_code'\n",
       "44. 'contact_country'\n",
       "45. 'supplementary_file'\n",
       "46. 'data_row_count'\n",
       "47. 'relation'\n",
       "48. 'age.ch1'\n",
       "49. 'bal_culture.ch1'\n",
       "50. 'bal_smear.ch1'\n",
       "51. 'bcg.vaccinated.ch1'\n",
       "52. 'ethnicity.ch1'\n",
       "53. 'exposure_latent.ch1'\n",
       "54. 'gender.ch1'\n",
       "55. 'geographical.region.ch1'\n",
       "56. 'healthy.control.ch1'\n",
       "57. 'illness.ch1'\n",
       "58. 'index_case_disease_site.ch1'\n",
       "59. 'isolate.sensitivity.ch1'\n",
       "60. 'isolate_sensitivity.ch1'\n",
       "61. 'modal.x.ray.grade.ch1'\n",
       "62. 'region.of.birth.ch1'\n",
       "63. 'smear.of.index.case.ch1'\n",
       "64. 'sputum.smear.status.ch1'\n",
       "65. 'sputum_culture.ch1'\n",
       "66. 'sputum_smear.ch1'\n",
       "67. 'time.post.initiation.of.treatment.ch1'\n",
       "68. 'tst.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE37250\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'molecule_ch1'\n",
       "15. 'extract_protocol_ch1'\n",
       "16. 'label_ch1'\n",
       "17. 'label_protocol_ch1'\n",
       "18. 'taxid_ch1'\n",
       "19. 'hyb_protocol'\n",
       "20. 'scan_protocol'\n",
       "21. 'description'\n",
       "22. 'data_processing'\n",
       "23. 'platform_id'\n",
       "24. 'contact_name'\n",
       "25. 'contact_email'\n",
       "26. 'contact_laboratory'\n",
       "27. 'contact_department'\n",
       "28. 'contact_institute'\n",
       "29. 'contact_address'\n",
       "30. 'contact_city'\n",
       "31. 'contact_zip.postal_code'\n",
       "32. 'contact_country'\n",
       "33. 'supplementary_file'\n",
       "34. 'data_row_count'\n",
       "35. 'disease.state.ch1'\n",
       "36. 'geographical.region.ch1'\n",
       "37. 'hiv.status.ch1'\n",
       "38. 'tissue.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE39939\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'molecule_ch1'\n",
       "14. 'extract_protocol_ch1'\n",
       "15. 'label_ch1'\n",
       "16. 'label_protocol_ch1'\n",
       "17. 'taxid_ch1'\n",
       "18. 'hyb_protocol'\n",
       "19. 'scan_protocol'\n",
       "20. 'data_processing'\n",
       "21. 'platform_id'\n",
       "22. 'contact_name'\n",
       "23. 'contact_email'\n",
       "24. 'contact_laboratory'\n",
       "25. 'contact_department'\n",
       "26. 'contact_institute'\n",
       "27. 'contact_address'\n",
       "28. 'contact_city'\n",
       "29. 'contact_zip.postal_code'\n",
       "30. 'contact_country'\n",
       "31. 'supplementary_file'\n",
       "32. 'data_row_count'\n",
       "33. 'geographical.region.ch1'\n",
       "34. 'hiv.status.ch1'\n",
       "35. 'illness.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE39940\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'molecule_ch1'\n",
       "15. 'extract_protocol_ch1'\n",
       "16. 'label_ch1'\n",
       "17. 'label_protocol_ch1'\n",
       "18. 'taxid_ch1'\n",
       "19. 'hyb_protocol'\n",
       "20. 'scan_protocol'\n",
       "21. 'data_processing'\n",
       "22. 'platform_id'\n",
       "23. 'contact_name'\n",
       "24. 'contact_email'\n",
       "25. 'contact_laboratory'\n",
       "26. 'contact_department'\n",
       "27. 'contact_institute'\n",
       "28. 'contact_address'\n",
       "29. 'contact_city'\n",
       "30. 'contact_zip.postal_code'\n",
       "31. 'contact_country'\n",
       "32. 'supplementary_file'\n",
       "33. 'data_row_count'\n",
       "34. 'barcode.ch1'\n",
       "35. 'disease.status.ch1'\n",
       "36. 'geographical.region.ch1'\n",
       "37. 'hiv.status.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42825\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'treatment_protocol_ch1'\n",
       "15. 'growth_protocol_ch1'\n",
       "16. 'molecule_ch1'\n",
       "17. 'extract_protocol_ch1'\n",
       "18. 'label_ch1'\n",
       "19. 'label_protocol_ch1'\n",
       "20. 'taxid_ch1'\n",
       "21. 'hyb_protocol'\n",
       "22. 'scan_protocol'\n",
       "23. 'data_processing'\n",
       "24. 'data_processing.1'\n",
       "25. 'platform_id'\n",
       "26. 'contact_name'\n",
       "27. 'contact_institute'\n",
       "28. 'contact_address'\n",
       "29. 'contact_city'\n",
       "30. 'contact_zip.postal_code'\n",
       "31. 'contact_country'\n",
       "32. 'supplementary_file'\n",
       "33. 'data_row_count'\n",
       "34. 'disease.state.ch1'\n",
       "35. 'ethnicity.ch1'\n",
       "36. 'gender.ch1'\n",
       "37. 'tissue.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42826\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'treatment_protocol_ch1'\n",
       "15. 'growth_protocol_ch1'\n",
       "16. 'molecule_ch1'\n",
       "17. 'extract_protocol_ch1'\n",
       "18. 'label_ch1'\n",
       "19. 'label_protocol_ch1'\n",
       "20. 'taxid_ch1'\n",
       "21. 'hyb_protocol'\n",
       "22. 'scan_protocol'\n",
       "23. 'data_processing'\n",
       "24. 'platform_id'\n",
       "25. 'contact_name'\n",
       "26. 'contact_institute'\n",
       "27. 'contact_address'\n",
       "28. 'contact_city'\n",
       "29. 'contact_zip.postal_code'\n",
       "30. 'contact_country'\n",
       "31. 'supplementary_file'\n",
       "32. 'data_row_count'\n",
       "33. 'disease.state.ch1'\n",
       "34. 'ethnicity.ch1'\n",
       "35. 'gender.ch1'\n",
       "36. 'tissue.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42827\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'treatment_protocol_ch1'\n",
       "15. 'growth_protocol_ch1'\n",
       "16. 'molecule_ch1'\n",
       "17. 'extract_protocol_ch1'\n",
       "18. 'label_ch1'\n",
       "19. 'label_protocol_ch1'\n",
       "20. 'taxid_ch1'\n",
       "21. 'hyb_protocol'\n",
       "22. 'scan_protocol'\n",
       "23. 'description'\n",
       "24. 'data_processing'\n",
       "25. 'data_processing.1'\n",
       "26. 'platform_id'\n",
       "27. 'contact_name'\n",
       "28. 'contact_institute'\n",
       "29. 'contact_address'\n",
       "30. 'contact_city'\n",
       "31. 'contact_zip.postal_code'\n",
       "32. 'contact_country'\n",
       "33. 'supplementary_file'\n",
       "34. 'data_row_count'\n",
       "35. 'disease.state.ch1'\n",
       "36. 'subject.ch1'\n",
       "37. 'tissue.ch1'\n",
       "38. 'treatment.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42830\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'treatment_protocol_ch1'\n",
       "15. 'growth_protocol_ch1'\n",
       "16. 'molecule_ch1'\n",
       "17. 'extract_protocol_ch1'\n",
       "18. 'label_ch1'\n",
       "19. 'label_protocol_ch1'\n",
       "20. 'taxid_ch1'\n",
       "21. 'hyb_protocol'\n",
       "22. 'scan_protocol'\n",
       "23. 'data_processing'\n",
       "24. 'platform_id'\n",
       "25. 'contact_name'\n",
       "26. 'contact_institute'\n",
       "27. 'contact_address'\n",
       "28. 'contact_city'\n",
       "29. 'contact_zip.postal_code'\n",
       "30. 'contact_country'\n",
       "31. 'supplementary_file'\n",
       "32. 'data_row_count'\n",
       "33. 'disease.state.ch1'\n",
       "34. 'ethnicity.ch1'\n",
       "35. 'gender.ch1'\n",
       "36. 'tissue.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42831\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'characteristics_ch1.2'\n",
       "13. 'characteristics_ch1.3'\n",
       "14. 'treatment_protocol_ch1'\n",
       "15. 'growth_protocol_ch1'\n",
       "16. 'molecule_ch1'\n",
       "17. 'extract_protocol_ch1'\n",
       "18. 'label_ch1'\n",
       "19. 'label_protocol_ch1'\n",
       "20. 'taxid_ch1'\n",
       "21. 'hyb_protocol'\n",
       "22. 'scan_protocol'\n",
       "23. 'data_processing'\n",
       "24. 'data_processing.1'\n",
       "25. 'platform_id'\n",
       "26. 'contact_name'\n",
       "27. 'contact_institute'\n",
       "28. 'contact_address'\n",
       "29. 'contact_city'\n",
       "30. 'contact_zip.postal_code'\n",
       "31. 'contact_country'\n",
       "32. 'supplementary_file'\n",
       "33. 'data_row_count'\n",
       "34. 'response.ch1'\n",
       "35. 'subject.ch1'\n",
       "36. 'time..d..ch1'\n",
       "37. 'treament.ch1'\n",
       "\n",
       "\n",
       "\n",
       "$GSE42832\n",
       ":   1. 'title'\n",
       "2. 'geo_accession'\n",
       "3. 'status'\n",
       "4. 'submission_date'\n",
       "5. 'last_update_date'\n",
       "6. 'type'\n",
       "7. 'channel_count'\n",
       "8. 'source_name_ch1'\n",
       "9. 'organism_ch1'\n",
       "10. 'characteristics_ch1'\n",
       "11. 'characteristics_ch1.1'\n",
       "12. 'treatment_protocol_ch1'\n",
       "13. 'growth_protocol_ch1'\n",
       "14. 'molecule_ch1'\n",
       "15. 'extract_protocol_ch1'\n",
       "16. 'label_ch1'\n",
       "17. 'label_protocol_ch1'\n",
       "18. 'taxid_ch1'\n",
       "19. 'hyb_protocol'\n",
       "20. 'scan_protocol'\n",
       "21. 'data_processing'\n",
       "22. 'platform_id'\n",
       "23. 'contact_name'\n",
       "24. 'contact_institute'\n",
       "25. 'contact_address'\n",
       "26. 'contact_city'\n",
       "27. 'contact_zip.postal_code'\n",
       "28. 'contact_country'\n",
       "29. 'supplementary_file'\n",
       "30. 'data_row_count'\n",
       "31. 'cell.type.ch1'\n",
       "32. 'disease.state.ch1'\n",
       "33. 'tissue.ch1'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$GSE19491\n",
       " [1] \"title\"                                \n",
       " [2] \"geo_accession\"                        \n",
       " [3] \"status\"                               \n",
       " [4] \"submission_date\"                      \n",
       " [5] \"last_update_date\"                     \n",
       " [6] \"type\"                                 \n",
       " [7] \"channel_count\"                        \n",
       " [8] \"source_name_ch1\"                      \n",
       " [9] \"organism_ch1\"                         \n",
       "[10] \"characteristics_ch1\"                  \n",
       "[11] \"characteristics_ch1.1\"                \n",
       "[12] \"characteristics_ch1.2\"                \n",
       "[13] \"characteristics_ch1.3\"                \n",
       "[14] \"characteristics_ch1.4\"                \n",
       "[15] \"characteristics_ch1.5\"                \n",
       "[16] \"characteristics_ch1.6\"                \n",
       "[17] \"characteristics_ch1.7\"                \n",
       "[18] \"characteristics_ch1.8\"                \n",
       "[19] \"characteristics_ch1.9\"                \n",
       "[20] \"characteristics_ch1.10\"               \n",
       "[21] \"characteristics_ch1.11\"               \n",
       "[22] \"characteristics_ch1.12\"               \n",
       "[23] \"characteristics_ch1.13\"               \n",
       "[24] \"characteristics_ch1.14\"               \n",
       "[25] \"characteristics_ch1.15\"               \n",
       "[26] \"characteristics_ch1.16\"               \n",
       "[27] \"molecule_ch1\"                         \n",
       "[28] \"extract_protocol_ch1\"                 \n",
       "[29] \"label_ch1\"                            \n",
       "[30] \"label_protocol_ch1\"                   \n",
       "[31] \"taxid_ch1\"                            \n",
       "[32] \"hyb_protocol\"                         \n",
       "[33] \"scan_protocol\"                        \n",
       "[34] \"description\"                          \n",
       "[35] \"data_processing\"                      \n",
       "[36] \"platform_id\"                          \n",
       "[37] \"contact_name\"                         \n",
       "[38] \"contact_email\"                        \n",
       "[39] \"contact_institute\"                    \n",
       "[40] \"contact_address\"                      \n",
       "[41] \"contact_city\"                         \n",
       "[42] \"contact_state\"                        \n",
       "[43] \"contact_zip.postal_code\"              \n",
       "[44] \"contact_country\"                      \n",
       "[45] \"supplementary_file\"                   \n",
       "[46] \"data_row_count\"                       \n",
       "[47] \"relation\"                             \n",
       "[48] \"age.ch1\"                              \n",
       "[49] \"bal_culture.ch1\"                      \n",
       "[50] \"bal_smear.ch1\"                        \n",
       "[51] \"bcg.vaccinated.ch1\"                   \n",
       "[52] \"ethnicity.ch1\"                        \n",
       "[53] \"exposure_latent.ch1\"                  \n",
       "[54] \"gender.ch1\"                           \n",
       "[55] \"geographical.region.ch1\"              \n",
       "[56] \"healthy.control.ch1\"                  \n",
       "[57] \"illness.ch1\"                          \n",
       "[58] \"index_case_disease_site.ch1\"          \n",
       "[59] \"isolate.sensitivity.ch1\"              \n",
       "[60] \"isolate_sensitivity.ch1\"              \n",
       "[61] \"modal.x.ray.grade.ch1\"                \n",
       "[62] \"region.of.birth.ch1\"                  \n",
       "[63] \"smear.of.index.case.ch1\"              \n",
       "[64] \"sputum.smear.status.ch1\"              \n",
       "[65] \"sputum_culture.ch1\"                   \n",
       "[66] \"sputum_smear.ch1\"                     \n",
       "[67] \"time.post.initiation.of.treatment.ch1\"\n",
       "[68] \"tst.ch1\"                              \n",
       "\n",
       "$GSE37250\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"molecule_ch1\"           \n",
       "[15] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[17] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[19] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[21] \"description\"             \"data_processing\"        \n",
       "[23] \"platform_id\"             \"contact_name\"           \n",
       "[25] \"contact_email\"           \"contact_laboratory\"     \n",
       "[27] \"contact_department\"      \"contact_institute\"      \n",
       "[29] \"contact_address\"         \"contact_city\"           \n",
       "[31] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[33] \"supplementary_file\"      \"data_row_count\"         \n",
       "[35] \"disease.state.ch1\"       \"geographical.region.ch1\"\n",
       "[37] \"hiv.status.ch1\"          \"tissue.ch1\"             \n",
       "\n",
       "$GSE39939\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"molecule_ch1\"            \"extract_protocol_ch1\"   \n",
       "[15] \"label_ch1\"               \"label_protocol_ch1\"     \n",
       "[17] \"taxid_ch1\"               \"hyb_protocol\"           \n",
       "[19] \"scan_protocol\"           \"data_processing\"        \n",
       "[21] \"platform_id\"             \"contact_name\"           \n",
       "[23] \"contact_email\"           \"contact_laboratory\"     \n",
       "[25] \"contact_department\"      \"contact_institute\"      \n",
       "[27] \"contact_address\"         \"contact_city\"           \n",
       "[29] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[31] \"supplementary_file\"      \"data_row_count\"         \n",
       "[33] \"geographical.region.ch1\" \"hiv.status.ch1\"         \n",
       "[35] \"illness.ch1\"            \n",
       "\n",
       "$GSE39940\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"molecule_ch1\"           \n",
       "[15] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[17] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[19] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[21] \"data_processing\"         \"platform_id\"            \n",
       "[23] \"contact_name\"            \"contact_email\"          \n",
       "[25] \"contact_laboratory\"      \"contact_department\"     \n",
       "[27] \"contact_institute\"       \"contact_address\"        \n",
       "[29] \"contact_city\"            \"contact_zip.postal_code\"\n",
       "[31] \"contact_country\"         \"supplementary_file\"     \n",
       "[33] \"data_row_count\"          \"barcode.ch1\"            \n",
       "[35] \"disease.status.ch1\"      \"geographical.region.ch1\"\n",
       "[37] \"hiv.status.ch1\"         \n",
       "\n",
       "$GSE42825\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"treatment_protocol_ch1\" \n",
       "[15] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[17] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[19] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[21] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[23] \"data_processing\"         \"data_processing.1\"      \n",
       "[25] \"platform_id\"             \"contact_name\"           \n",
       "[27] \"contact_institute\"       \"contact_address\"        \n",
       "[29] \"contact_city\"            \"contact_zip.postal_code\"\n",
       "[31] \"contact_country\"         \"supplementary_file\"     \n",
       "[33] \"data_row_count\"          \"disease.state.ch1\"      \n",
       "[35] \"ethnicity.ch1\"           \"gender.ch1\"             \n",
       "[37] \"tissue.ch1\"             \n",
       "\n",
       "$GSE42826\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"treatment_protocol_ch1\" \n",
       "[15] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[17] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[19] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[21] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[23] \"data_processing\"         \"platform_id\"            \n",
       "[25] \"contact_name\"            \"contact_institute\"      \n",
       "[27] \"contact_address\"         \"contact_city\"           \n",
       "[29] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[31] \"supplementary_file\"      \"data_row_count\"         \n",
       "[33] \"disease.state.ch1\"       \"ethnicity.ch1\"          \n",
       "[35] \"gender.ch1\"              \"tissue.ch1\"             \n",
       "\n",
       "$GSE42827\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"treatment_protocol_ch1\" \n",
       "[15] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[17] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[19] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[21] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[23] \"description\"             \"data_processing\"        \n",
       "[25] \"data_processing.1\"       \"platform_id\"            \n",
       "[27] \"contact_name\"            \"contact_institute\"      \n",
       "[29] \"contact_address\"         \"contact_city\"           \n",
       "[31] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[33] \"supplementary_file\"      \"data_row_count\"         \n",
       "[35] \"disease.state.ch1\"       \"subject.ch1\"            \n",
       "[37] \"tissue.ch1\"              \"treatment.ch1\"          \n",
       "\n",
       "$GSE42830\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"treatment_protocol_ch1\" \n",
       "[15] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[17] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[19] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[21] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[23] \"data_processing\"         \"platform_id\"            \n",
       "[25] \"contact_name\"            \"contact_institute\"      \n",
       "[27] \"contact_address\"         \"contact_city\"           \n",
       "[29] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[31] \"supplementary_file\"      \"data_row_count\"         \n",
       "[33] \"disease.state.ch1\"       \"ethnicity.ch1\"          \n",
       "[35] \"gender.ch1\"              \"tissue.ch1\"             \n",
       "\n",
       "$GSE42831\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"characteristics_ch1.2\"  \n",
       "[13] \"characteristics_ch1.3\"   \"treatment_protocol_ch1\" \n",
       "[15] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[17] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[19] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[21] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[23] \"data_processing\"         \"data_processing.1\"      \n",
       "[25] \"platform_id\"             \"contact_name\"           \n",
       "[27] \"contact_institute\"       \"contact_address\"        \n",
       "[29] \"contact_city\"            \"contact_zip.postal_code\"\n",
       "[31] \"contact_country\"         \"supplementary_file\"     \n",
       "[33] \"data_row_count\"          \"response.ch1\"           \n",
       "[35] \"subject.ch1\"             \"time..d..ch1\"           \n",
       "[37] \"treament.ch1\"           \n",
       "\n",
       "$GSE42832\n",
       " [1] \"title\"                   \"geo_accession\"          \n",
       " [3] \"status\"                  \"submission_date\"        \n",
       " [5] \"last_update_date\"        \"type\"                   \n",
       " [7] \"channel_count\"           \"source_name_ch1\"        \n",
       " [9] \"organism_ch1\"            \"characteristics_ch1\"    \n",
       "[11] \"characteristics_ch1.1\"   \"treatment_protocol_ch1\" \n",
       "[13] \"growth_protocol_ch1\"     \"molecule_ch1\"           \n",
       "[15] \"extract_protocol_ch1\"    \"label_ch1\"              \n",
       "[17] \"label_protocol_ch1\"      \"taxid_ch1\"              \n",
       "[19] \"hyb_protocol\"            \"scan_protocol\"          \n",
       "[21] \"data_processing\"         \"platform_id\"            \n",
       "[23] \"contact_name\"            \"contact_institute\"      \n",
       "[25] \"contact_address\"         \"contact_city\"           \n",
       "[27] \"contact_zip.postal_code\" \"contact_country\"        \n",
       "[29] \"supplementary_file\"      \"data_row_count\"         \n",
       "[31] \"cell.type.ch1\"           \"disease.state.ch1\"      \n",
       "[33] \"tissue.ch1\"             \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$pheno.list, function (x) {colnames(x)})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Format GSE19491\n",
    "\n",
    "- I think some of the TB and control samples may be longitudinal, and in the Duffy et al paper were excluded. Treated TB and anything control shouldn't show recent infection, so for now I will just include them. If needed I can dive deeper into the analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# For GSE19491\n",
    "# Exclude source_name_ch1 in here\n",
    "\"Whole blood from patient with active TB 12 months after treatment started (i.e. after treatment completed)\"\n",
    "\"Whole blood from patient with active TB 2 months after treatment started\"\n",
    "\n",
    "# illness.ch1 is the key phenotype\n",
    "# anything containing \"Control\" (i.e. grepl) should be \"control\". anything containing Latent or LATENT TB should be LTBI\n",
    "# turn PTB into TB\n",
    "# keep everything else\n",
    "#\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE19491 = droplevels(test.data$pheno.list$GSE19491[!(test.data$pheno.list$GSE19491$source_name_ch1 %in% \n",
    "                                                                c(\"Whole blood from patient with active TB 12 months after treatment started (i.e. after treatment completed)\",\n",
    "                                                                \"Whole blood from patient with active TB 2 months after treatment started\")),])\n",
    "test.data$pheno.list$GSE19491$status = as.character(test.data$pheno.list$GSE19491$illness.ch1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE19491$status = as.character(ifelse(grepl(\"Control\", test.data$pheno.list$GSE19491$status),\n",
    "                                              \"control\", ifelse(grepl(\"Latent|LATENT\", test.data$pheno.list$GSE19491$status),\n",
    "                                                               \"LTBI\", ifelse(grepl(\"PTB\", test.data$pheno.list$GSE19491$status),\n",
    "                                                                             \"TB\", \n",
    "                                                                              ifelse(test.data$pheno.list$GSE19491$status == \"ASLE\", \"SLE\",\n",
    "                                                                                    ifelse(is.na(test.data$pheno.list$GSE19491$status), \"control\",\n",
    "                                                                                          test.data$pheno.list$GSE19491$status))))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "0"
      ],
      "text/latex": [
       "0"
      ],
      "text/markdown": [
       "0"
      ],
      "text/plain": [
       "[1] 0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "test.data$pheno.list$GSE19491$status = ifelse(is.na(test.data$pheno.list$GSE19491$status), \"control\", test.data$pheno.list$GSE19491$status)\n",
    "sum(is.na(test.data$pheno.list$GSE19491$status))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'TB'</li>\n",
       "\t<li>'LTBI'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'SLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Strep'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Staph'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'Still'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'PSLE'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "\t<li>'control'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'TB'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'TB'\n",
       "\\item 'LTBI'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'SLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Strep'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Staph'\n",
       "\\item 'Still'\n",
       "\\item 'Still'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'PSLE'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\item 'control'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'control'\n",
       "2. 'control'\n",
       "3. 'control'\n",
       "4. 'control'\n",
       "5. 'control'\n",
       "6. 'control'\n",
       "7. 'control'\n",
       "8. 'control'\n",
       "9. 'control'\n",
       "10. 'control'\n",
       "11. 'control'\n",
       "12. 'control'\n",
       "13. 'TB'\n",
       "14. 'TB'\n",
       "15. 'TB'\n",
       "16. 'TB'\n",
       "17. 'TB'\n",
       "18. 'TB'\n",
       "19. 'TB'\n",
       "20. 'LTBI'\n",
       "21. 'TB'\n",
       "22. 'control'\n",
       "23. 'TB'\n",
       "24. 'LTBI'\n",
       "25. 'control'\n",
       "26. 'control'\n",
       "27. 'LTBI'\n",
       "28. 'TB'\n",
       "29. 'control'\n",
       "30. 'LTBI'\n",
       "31. 'TB'\n",
       "32. 'LTBI'\n",
       "33. 'LTBI'\n",
       "34. 'control'\n",
       "35. 'TB'\n",
       "36. 'LTBI'\n",
       "37. 'control'\n",
       "38. 'TB'\n",
       "39. 'LTBI'\n",
       "40. 'TB'\n",
       "41. 'LTBI'\n",
       "42. 'TB'\n",
       "43. 'LTBI'\n",
       "44. 'TB'\n",
       "45. 'LTBI'\n",
       "46. 'TB'\n",
       "47. 'TB'\n",
       "48. 'LTBI'\n",
       "49. 'control'\n",
       "50. 'TB'\n",
       "51. 'control'\n",
       "52. 'control'\n",
       "53. 'control'\n",
       "54. 'LTBI'\n",
       "55. 'control'\n",
       "56. 'LTBI'\n",
       "57. 'TB'\n",
       "58. 'LTBI'\n",
       "59. 'control'\n",
       "60. 'LTBI'\n",
       "61. 'LTBI'\n",
       "62. 'TB'\n",
       "63. 'TB'\n",
       "64. 'TB'\n",
       "65. 'TB'\n",
       "66. 'LTBI'\n",
       "67. 'LTBI'\n",
       "68. 'LTBI'\n",
       "69. 'LTBI'\n",
       "70. 'LTBI'\n",
       "71. 'LTBI'\n",
       "72. 'LTBI'\n",
       "73. 'LTBI'\n",
       "74. 'LTBI'\n",
       "75. 'LTBI'\n",
       "76. 'LTBI'\n",
       "77. 'TB'\n",
       "78. 'TB'\n",
       "79. 'TB'\n",
       "80. 'TB'\n",
       "81. 'TB'\n",
       "82. 'LTBI'\n",
       "83. 'LTBI'\n",
       "84. 'LTBI'\n",
       "85. 'LTBI'\n",
       "86. 'TB'\n",
       "87. 'TB'\n",
       "88. 'TB'\n",
       "89. 'TB'\n",
       "90. 'LTBI'\n",
       "91. 'LTBI'\n",
       "92. 'LTBI'\n",
       "93. 'LTBI'\n",
       "94. 'LTBI'\n",
       "95. 'TB'\n",
       "96. 'TB'\n",
       "97. 'TB'\n",
       "98. 'TB'\n",
       "99. 'LTBI'\n",
       "100. 'LTBI'\n",
       "101. 'LTBI'\n",
       "102. 'LTBI'\n",
       "103. 'LTBI'\n",
       "104. 'LTBI'\n",
       "105. 'TB'\n",
       "106. 'TB'\n",
       "107. 'TB'\n",
       "108. 'LTBI'\n",
       "109. 'LTBI'\n",
       "110. 'LTBI'\n",
       "111. 'LTBI'\n",
       "112. 'LTBI'\n",
       "113. 'TB'\n",
       "114. 'TB'\n",
       "115. 'TB'\n",
       "116. 'TB'\n",
       "117. 'TB'\n",
       "118. 'TB'\n",
       "119. 'TB'\n",
       "120. 'TB'\n",
       "121. 'TB'\n",
       "122. 'TB'\n",
       "123. 'TB'\n",
       "124. 'TB'\n",
       "125. 'TB'\n",
       "126. 'TB'\n",
       "127. 'TB'\n",
       "128. 'TB'\n",
       "129. 'control'\n",
       "130. 'control'\n",
       "131. 'control'\n",
       "132. 'control'\n",
       "133. 'TB'\n",
       "134. 'TB'\n",
       "135. 'TB'\n",
       "136. 'TB'\n",
       "137. 'control'\n",
       "138. 'control'\n",
       "139. 'control'\n",
       "140. 'control'\n",
       "141. 'TB'\n",
       "142. 'TB'\n",
       "143. 'TB'\n",
       "144. 'TB'\n",
       "145. 'TB'\n",
       "146. 'TB'\n",
       "147. 'TB'\n",
       "148. 'TB'\n",
       "149. 'control'\n",
       "150. 'control'\n",
       "151. 'control'\n",
       "152. 'control'\n",
       "153. 'control'\n",
       "154. 'control'\n",
       "155. 'control'\n",
       "156. 'control'\n",
       "157. 'TB'\n",
       "158. 'TB'\n",
       "159. 'TB'\n",
       "160. 'TB'\n",
       "161. 'LTBI'\n",
       "162. 'LTBI'\n",
       "163. 'LTBI'\n",
       "164. 'LTBI'\n",
       "165. 'control'\n",
       "166. 'control'\n",
       "167. 'TB'\n",
       "168. 'TB'\n",
       "169. 'LTBI'\n",
       "170. 'LTBI'\n",
       "171. 'LTBI'\n",
       "172. 'TB'\n",
       "173. 'TB'\n",
       "174. 'control'\n",
       "175. 'control'\n",
       "176. 'LTBI'\n",
       "177. 'LTBI'\n",
       "178. 'LTBI'\n",
       "179. 'LTBI'\n",
       "180. 'TB'\n",
       "181. 'TB'\n",
       "182. 'TB'\n",
       "183. 'TB'\n",
       "184. 'control'\n",
       "185. 'control'\n",
       "186. 'LTBI'\n",
       "187. 'LTBI'\n",
       "188. 'LTBI'\n",
       "189. 'LTBI'\n",
       "190. 'TB'\n",
       "191. 'TB'\n",
       "192. 'TB'\n",
       "193. 'TB'\n",
       "194. 'control'\n",
       "195. 'control'\n",
       "196. 'control'\n",
       "197. 'LTBI'\n",
       "198. 'LTBI'\n",
       "199. 'LTBI'\n",
       "200. 'TB'\n",
       "201. 'TB'\n",
       "202. 'LTBI'\n",
       "203. 'TB'\n",
       "204. 'TB'\n",
       "205. 'LTBI'\n",
       "206. 'control'\n",
       "207. 'TB'\n",
       "208. 'LTBI'\n",
       "209. 'control'\n",
       "210. 'control'\n",
       "211. 'Still'\n",
       "212. 'Still'\n",
       "213. 'Still'\n",
       "214. 'Still'\n",
       "215. 'Still'\n",
       "216. 'Still'\n",
       "217. 'Still'\n",
       "218. 'Still'\n",
       "219. 'Still'\n",
       "220. 'Still'\n",
       "221. 'Still'\n",
       "222. 'Still'\n",
       "223. 'Still'\n",
       "224. 'Still'\n",
       "225. 'Still'\n",
       "226. 'Still'\n",
       "227. 'Still'\n",
       "228. 'Still'\n",
       "229. 'Still'\n",
       "230. 'Still'\n",
       "231. 'Still'\n",
       "232. 'Still'\n",
       "233. 'Still'\n",
       "234. 'Still'\n",
       "235. 'Still'\n",
       "236. 'Still'\n",
       "237. 'Still'\n",
       "238. 'SLE'\n",
       "239. 'SLE'\n",
       "240. 'SLE'\n",
       "241. 'SLE'\n",
       "242. 'SLE'\n",
       "243. 'SLE'\n",
       "244. 'SLE'\n",
       "245. 'SLE'\n",
       "246. 'SLE'\n",
       "247. 'SLE'\n",
       "248. 'SLE'\n",
       "249. 'SLE'\n",
       "250. 'SLE'\n",
       "251. 'SLE'\n",
       "252. 'SLE'\n",
       "253. 'SLE'\n",
       "254. 'SLE'\n",
       "255. 'SLE'\n",
       "256. 'SLE'\n",
       "257. 'SLE'\n",
       "258. 'SLE'\n",
       "259. 'SLE'\n",
       "260. 'SLE'\n",
       "261. 'SLE'\n",
       "262. 'SLE'\n",
       "263. 'SLE'\n",
       "264. 'SLE'\n",
       "265. 'SLE'\n",
       "266. 'PSLE'\n",
       "267. 'PSLE'\n",
       "268. 'PSLE'\n",
       "269. 'PSLE'\n",
       "270. 'PSLE'\n",
       "271. 'PSLE'\n",
       "272. 'PSLE'\n",
       "273. 'PSLE'\n",
       "274. 'PSLE'\n",
       "275. 'PSLE'\n",
       "276. 'PSLE'\n",
       "277. 'PSLE'\n",
       "278. 'PSLE'\n",
       "279. 'PSLE'\n",
       "280. 'PSLE'\n",
       "281. 'PSLE'\n",
       "282. 'PSLE'\n",
       "283. 'PSLE'\n",
       "284. 'PSLE'\n",
       "285. 'PSLE'\n",
       "286. 'PSLE'\n",
       "287. 'PSLE'\n",
       "288. 'PSLE'\n",
       "289. 'PSLE'\n",
       "290. 'PSLE'\n",
       "291. 'PSLE'\n",
       "292. 'PSLE'\n",
       "293. 'PSLE'\n",
       "294. 'PSLE'\n",
       "295. 'PSLE'\n",
       "296. 'PSLE'\n",
       "297. 'PSLE'\n",
       "298. 'PSLE'\n",
       "299. 'Still'\n",
       "300. 'Still'\n",
       "301. 'Staph'\n",
       "302. 'Staph'\n",
       "303. 'Staph'\n",
       "304. 'Staph'\n",
       "305. 'Staph'\n",
       "306. 'Staph'\n",
       "307. 'Staph'\n",
       "308. 'Staph'\n",
       "309. 'Staph'\n",
       "310. 'Staph'\n",
       "311. 'Staph'\n",
       "312. 'Staph'\n",
       "313. 'Staph'\n",
       "314. 'Staph'\n",
       "315. 'Strep'\n",
       "316. 'Strep'\n",
       "317. 'Strep'\n",
       "318. 'Strep'\n",
       "319. 'Strep'\n",
       "320. 'Strep'\n",
       "321. 'Strep'\n",
       "322. 'Strep'\n",
       "323. 'Strep'\n",
       "324. 'Strep'\n",
       "325. 'Strep'\n",
       "326. 'Strep'\n",
       "327. 'Staph'\n",
       "328. 'Staph'\n",
       "329. 'Staph'\n",
       "330. 'Staph'\n",
       "331. 'Staph'\n",
       "332. 'Staph'\n",
       "333. 'Staph'\n",
       "334. 'Staph'\n",
       "335. 'Staph'\n",
       "336. 'Staph'\n",
       "337. 'Staph'\n",
       "338. 'Staph'\n",
       "339. 'Staph'\n",
       "340. 'Staph'\n",
       "341. 'Staph'\n",
       "342. 'Staph'\n",
       "343. 'Staph'\n",
       "344. 'Staph'\n",
       "345. 'Staph'\n",
       "346. 'Staph'\n",
       "347. 'Staph'\n",
       "348. 'Staph'\n",
       "349. 'Staph'\n",
       "350. 'Staph'\n",
       "351. 'Staph'\n",
       "352. 'Staph'\n",
       "353. 'Still'\n",
       "354. 'Still'\n",
       "355. 'PSLE'\n",
       "356. 'PSLE'\n",
       "357. 'PSLE'\n",
       "358. 'PSLE'\n",
       "359. 'PSLE'\n",
       "360. 'PSLE'\n",
       "361. 'PSLE'\n",
       "362. 'PSLE'\n",
       "363. 'PSLE'\n",
       "364. 'PSLE'\n",
       "365. 'PSLE'\n",
       "366. 'PSLE'\n",
       "367. 'PSLE'\n",
       "368. 'PSLE'\n",
       "369. 'PSLE'\n",
       "370. 'PSLE'\n",
       "371. 'PSLE'\n",
       "372. 'PSLE'\n",
       "373. 'PSLE'\n",
       "374. 'PSLE'\n",
       "375. 'PSLE'\n",
       "376. 'PSLE'\n",
       "377. 'PSLE'\n",
       "378. 'PSLE'\n",
       "379. 'PSLE'\n",
       "380. 'PSLE'\n",
       "381. 'PSLE'\n",
       "382. 'PSLE'\n",
       "383. 'PSLE'\n",
       "384. 'PSLE'\n",
       "385. 'PSLE'\n",
       "386. 'PSLE'\n",
       "387. 'PSLE'\n",
       "388. 'PSLE'\n",
       "389. 'PSLE'\n",
       "390. 'PSLE'\n",
       "391. 'PSLE'\n",
       "392. 'PSLE'\n",
       "393. 'PSLE'\n",
       "394. 'PSLE'\n",
       "395. 'PSLE'\n",
       "396. 'PSLE'\n",
       "397. 'PSLE'\n",
       "398. 'PSLE'\n",
       "399. 'PSLE'\n",
       "400. 'PSLE'\n",
       "401. 'PSLE'\n",
       "402. 'PSLE'\n",
       "403. 'PSLE'\n",
       "404. 'control'\n",
       "405. 'control'\n",
       "406. 'control'\n",
       "407. 'control'\n",
       "408. 'control'\n",
       "409. 'control'\n",
       "410. 'control'\n",
       "411. 'control'\n",
       "412. 'control'\n",
       "413. 'control'\n",
       "414. 'control'\n",
       "415. 'control'\n",
       "416. 'control'\n",
       "417. 'control'\n",
       "418. 'control'\n",
       "419. 'control'\n",
       "420. 'control'\n",
       "421. 'control'\n",
       "422. 'control'\n",
       "423. 'control'\n",
       "424. 'control'\n",
       "425. 'control'\n",
       "426. 'control'\n",
       "427. 'control'\n",
       "428. 'control'\n",
       "429. 'control'\n",
       "430. 'control'\n",
       "431. 'control'\n",
       "432. 'control'\n",
       "433. 'control'\n",
       "434. 'control'\n",
       "435. 'control'\n",
       "436. 'control'\n",
       "437. 'control'\n",
       "438. 'control'\n",
       "439. 'control'\n",
       "440. 'control'\n",
       "441. 'control'\n",
       "442. 'control'\n",
       "443. 'control'\n",
       "444. 'control'\n",
       "445. 'control'\n",
       "446. 'control'\n",
       "447. 'control'\n",
       "448. 'control'\n",
       "449. 'control'\n",
       "450. 'control'\n",
       "451. 'control'\n",
       "452. 'control'\n",
       "453. 'control'\n",
       "454. 'control'\n",
       "455. 'control'\n",
       "456. 'control'\n",
       "457. 'control'\n",
       "458. 'control'\n",
       "459. 'control'\n",
       "460. 'control'\n",
       "461. 'control'\n",
       "462. 'control'\n",
       "463. 'control'\n",
       "464. 'control'\n",
       "465. 'control'\n",
       "466. 'control'\n",
       "467. 'control'\n",
       "468. 'control'\n",
       "469. 'control'\n",
       "470. 'control'\n",
       "471. 'control'\n",
       "472. 'control'\n",
       "473. 'control'\n",
       "474. 'control'\n",
       "475. 'control'\n",
       "476. 'control'\n",
       "477. 'control'\n",
       "478. 'control'\n",
       "479. 'control'\n",
       "480. 'control'\n",
       "481. 'control'\n",
       "482. 'control'\n",
       "483. 'control'\n",
       "484. 'control'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "  [8] \"control\" \"control\" \"control\" \"control\" \"control\" \"TB\"      \"TB\"     \n",
       " [15] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"LTBI\"    \"TB\"     \n",
       " [22] \"control\" \"TB\"      \"LTBI\"    \"control\" \"control\" \"LTBI\"    \"TB\"     \n",
       " [29] \"control\" \"LTBI\"    \"TB\"      \"LTBI\"    \"LTBI\"    \"control\" \"TB\"     \n",
       " [36] \"LTBI\"    \"control\" \"TB\"      \"LTBI\"    \"TB\"      \"LTBI\"    \"TB\"     \n",
       " [43] \"LTBI\"    \"TB\"      \"LTBI\"    \"TB\"      \"TB\"      \"LTBI\"    \"control\"\n",
       " [50] \"TB\"      \"control\" \"control\" \"control\" \"LTBI\"    \"control\" \"LTBI\"   \n",
       " [57] \"TB\"      \"LTBI\"    \"control\" \"LTBI\"    \"LTBI\"    \"TB\"      \"TB\"     \n",
       " [64] \"TB\"      \"TB\"      \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"   \n",
       " [71] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"     \n",
       " [78] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"LTBI\"    \"LTBI\"    \"LTBI\"   \n",
       " [85] \"LTBI\"    \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"LTBI\"    \"LTBI\"   \n",
       " [92] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"      \"TB\"      \"TB\"      \"TB\"     \n",
       " [99] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"     \n",
       "[106] \"TB\"      \"TB\"      \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"   \n",
       "[113] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"     \n",
       "[120] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"     \n",
       "[127] \"TB\"      \"TB\"      \"control\" \"control\" \"control\" \"control\" \"TB\"     \n",
       "[134] \"TB\"      \"TB\"      \"TB\"      \"control\" \"control\" \"control\" \"control\"\n",
       "[141] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"TB\"     \n",
       "[148] \"TB\"      \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[155] \"control\" \"control\" \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"LTBI\"   \n",
       "[162] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"control\" \"control\" \"TB\"      \"TB\"     \n",
       "[169] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"      \"TB\"      \"control\" \"control\"\n",
       "[176] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"      \"TB\"      \"TB\"     \n",
       "[183] \"TB\"      \"control\" \"control\" \"LTBI\"    \"LTBI\"    \"LTBI\"    \"LTBI\"   \n",
       "[190] \"TB\"      \"TB\"      \"TB\"      \"TB\"      \"control\" \"control\" \"control\"\n",
       "[197] \"LTBI\"    \"LTBI\"    \"LTBI\"    \"TB\"      \"TB\"      \"LTBI\"    \"TB\"     \n",
       "[204] \"TB\"      \"LTBI\"    \"control\" \"TB\"      \"LTBI\"    \"control\" \"control\"\n",
       "[211] \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"  \n",
       "[218] \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"  \n",
       "[225] \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"  \n",
       "[232] \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"Still\"   \"SLE\"    \n",
       "[239] \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"    \n",
       "[246] \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"    \n",
       "[253] \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"    \n",
       "[260] \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"SLE\"     \"PSLE\"   \n",
       "[267] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[274] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[281] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[288] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[295] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"Still\"   \"Still\"   \"Staph\"  \n",
       "[302] \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"  \n",
       "[309] \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Strep\"  \n",
       "[316] \"Strep\"   \"Strep\"   \"Strep\"   \"Strep\"   \"Strep\"   \"Strep\"   \"Strep\"  \n",
       "[323] \"Strep\"   \"Strep\"   \"Strep\"   \"Strep\"   \"Staph\"   \"Staph\"   \"Staph\"  \n",
       "[330] \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"  \n",
       "[337] \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"  \n",
       "[344] \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"   \"Staph\"  \n",
       "[351] \"Staph\"   \"Staph\"   \"Still\"   \"Still\"   \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[358] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[365] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[372] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[379] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[386] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[393] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"   \n",
       "[400] \"PSLE\"    \"PSLE\"    \"PSLE\"    \"PSLE\"    \"control\" \"control\" \"control\"\n",
       "[407] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[414] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[421] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[428] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[435] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[442] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[449] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[456] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[463] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[470] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[477] \"control\" \"control\" \"control\" \"control\" \"control\" \"control\" \"control\"\n",
       "[484] \"control\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "test.data$pheno.list$GSE19491$status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>TRUE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "\t<li>FALSE</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. FALSE\n",
       "2. FALSE\n",
       "3. FALSE\n",
       "4. FALSE\n",
       "5. FALSE\n",
       "6. FALSE\n",
       "7. FALSE\n",
       "8. FALSE\n",
       "9. FALSE\n",
       "10. FALSE\n",
       "11. FALSE\n",
       "12. FALSE\n",
       "13. TRUE\n",
       "14. TRUE\n",
       "15. TRUE\n",
       "16. TRUE\n",
       "17. TRUE\n",
       "18. TRUE\n",
       "19. TRUE\n",
       "20. FALSE\n",
       "21. TRUE\n",
       "22. FALSE\n",
       "23. TRUE\n",
       "24. FALSE\n",
       "25. FALSE\n",
       "26. FALSE\n",
       "27. FALSE\n",
       "28. TRUE\n",
       "29. FALSE\n",
       "30. FALSE\n",
       "31. TRUE\n",
       "32. FALSE\n",
       "33. FALSE\n",
       "34. FALSE\n",
       "35. TRUE\n",
       "36. FALSE\n",
       "37. FALSE\n",
       "38. TRUE\n",
       "39. FALSE\n",
       "40. TRUE\n",
       "41. FALSE\n",
       "42. TRUE\n",
       "43. FALSE\n",
       "44. TRUE\n",
       "45. FALSE\n",
       "46. TRUE\n",
       "47. TRUE\n",
       "48. FALSE\n",
       "49. FALSE\n",
       "50. TRUE\n",
       "51. FALSE\n",
       "52. FALSE\n",
       "53. FALSE\n",
       "54. FALSE\n",
       "55. FALSE\n",
       "56. FALSE\n",
       "57. TRUE\n",
       "58. FALSE\n",
       "59. FALSE\n",
       "60. FALSE\n",
       "61. FALSE\n",
       "62. TRUE\n",
       "63. TRUE\n",
       "64. TRUE\n",
       "65. TRUE\n",
       "66. FALSE\n",
       "67. FALSE\n",
       "68. FALSE\n",
       "69. FALSE\n",
       "70. FALSE\n",
       "71. FALSE\n",
       "72. FALSE\n",
       "73. FALSE\n",
       "74. FALSE\n",
       "75. FALSE\n",
       "76. FALSE\n",
       "77. TRUE\n",
       "78. TRUE\n",
       "79. TRUE\n",
       "80. TRUE\n",
       "81. TRUE\n",
       "82. FALSE\n",
       "83. FALSE\n",
       "84. FALSE\n",
       "85. FALSE\n",
       "86. TRUE\n",
       "87. TRUE\n",
       "88. TRUE\n",
       "89. TRUE\n",
       "90. FALSE\n",
       "91. FALSE\n",
       "92. FALSE\n",
       "93. FALSE\n",
       "94. FALSE\n",
       "95. TRUE\n",
       "96. TRUE\n",
       "97. TRUE\n",
       "98. TRUE\n",
       "99. FALSE\n",
       "100. FALSE\n",
       "101. FALSE\n",
       "102. FALSE\n",
       "103. FALSE\n",
       "104. FALSE\n",
       "105. TRUE\n",
       "106. TRUE\n",
       "107. TRUE\n",
       "108. FALSE\n",
       "109. FALSE\n",
       "110. FALSE\n",
       "111. FALSE\n",
       "112. FALSE\n",
       "113. TRUE\n",
       "114. TRUE\n",
       "115. TRUE\n",
       "116. TRUE\n",
       "117. TRUE\n",
       "118. TRUE\n",
       "119. TRUE\n",
       "120. TRUE\n",
       "121. TRUE\n",
       "122. TRUE\n",
       "123. TRUE\n",
       "124. TRUE\n",
       "125. TRUE\n",
       "126. TRUE\n",
       "127. TRUE\n",
       "128. TRUE\n",
       "129. FALSE\n",
       "130. FALSE\n",
       "131. FALSE\n",
       "132. FALSE\n",
       "133. TRUE\n",
       "134. TRUE\n",
       "135. TRUE\n",
       "136. TRUE\n",
       "137. FALSE\n",
       "138. FALSE\n",
       "139. FALSE\n",
       "140. FALSE\n",
       "141. TRUE\n",
       "142. TRUE\n",
       "143. TRUE\n",
       "144. TRUE\n",
       "145. TRUE\n",
       "146. TRUE\n",
       "147. TRUE\n",
       "148. TRUE\n",
       "149. FALSE\n",
       "150. FALSE\n",
       "151. FALSE\n",
       "152. FALSE\n",
       "153. FALSE\n",
       "154. FALSE\n",
       "155. FALSE\n",
       "156. FALSE\n",
       "157. TRUE\n",
       "158. TRUE\n",
       "159. TRUE\n",
       "160. TRUE\n",
       "161. FALSE\n",
       "162. FALSE\n",
       "163. FALSE\n",
       "164. FALSE\n",
       "165. FALSE\n",
       "166. FALSE\n",
       "167. TRUE\n",
       "168. TRUE\n",
       "169. FALSE\n",
       "170. FALSE\n",
       "171. FALSE\n",
       "172. TRUE\n",
       "173. TRUE\n",
       "174. FALSE\n",
       "175. FALSE\n",
       "176. FALSE\n",
       "177. FALSE\n",
       "178. FALSE\n",
       "179. FALSE\n",
       "180. TRUE\n",
       "181. TRUE\n",
       "182. TRUE\n",
       "183. TRUE\n",
       "184. FALSE\n",
       "185. FALSE\n",
       "186. FALSE\n",
       "187. FALSE\n",
       "188. FALSE\n",
       "189. FALSE\n",
       "190. TRUE\n",
       "191. TRUE\n",
       "192. TRUE\n",
       "193. TRUE\n",
       "194. FALSE\n",
       "195. FALSE\n",
       "196. FALSE\n",
       "197. FALSE\n",
       "198. FALSE\n",
       "199. FALSE\n",
       "200. TRUE\n",
       "201. TRUE\n",
       "202. FALSE\n",
       "203. TRUE\n",
       "204. TRUE\n",
       "205. FALSE\n",
       "206. FALSE\n",
       "207. TRUE\n",
       "208. FALSE\n",
       "209. FALSE\n",
       "210. FALSE\n",
       "211. FALSE\n",
       "212. FALSE\n",
       "213. FALSE\n",
       "214. FALSE\n",
       "215. FALSE\n",
       "216. FALSE\n",
       "217. FALSE\n",
       "218. FALSE\n",
       "219. FALSE\n",
       "220. FALSE\n",
       "221. FALSE\n",
       "222. FALSE\n",
       "223. FALSE\n",
       "224. FALSE\n",
       "225. FALSE\n",
       "226. FALSE\n",
       "227. FALSE\n",
       "228. FALSE\n",
       "229. FALSE\n",
       "230. FALSE\n",
       "231. FALSE\n",
       "232. FALSE\n",
       "233. FALSE\n",
       "234. FALSE\n",
       "235. FALSE\n",
       "236. FALSE\n",
       "237. FALSE\n",
       "238. FALSE\n",
       "239. FALSE\n",
       "240. FALSE\n",
       "241. FALSE\n",
       "242. FALSE\n",
       "243. FALSE\n",
       "244. FALSE\n",
       "245. FALSE\n",
       "246. FALSE\n",
       "247. FALSE\n",
       "248. FALSE\n",
       "249. FALSE\n",
       "250. FALSE\n",
       "251. FALSE\n",
       "252. FALSE\n",
       "253. FALSE\n",
       "254. FALSE\n",
       "255. FALSE\n",
       "256. FALSE\n",
       "257. FALSE\n",
       "258. FALSE\n",
       "259. FALSE\n",
       "260. FALSE\n",
       "261. FALSE\n",
       "262. FALSE\n",
       "263. FALSE\n",
       "264. FALSE\n",
       "265. FALSE\n",
       "266. FALSE\n",
       "267. FALSE\n",
       "268. FALSE\n",
       "269. FALSE\n",
       "270. FALSE\n",
       "271. FALSE\n",
       "272. FALSE\n",
       "273. FALSE\n",
       "274. FALSE\n",
       "275. FALSE\n",
       "276. FALSE\n",
       "277. FALSE\n",
       "278. FALSE\n",
       "279. FALSE\n",
       "280. FALSE\n",
       "281. FALSE\n",
       "282. FALSE\n",
       "283. FALSE\n",
       "284. FALSE\n",
       "285. FALSE\n",
       "286. FALSE\n",
       "287. FALSE\n",
       "288. FALSE\n",
       "289. FALSE\n",
       "290. FALSE\n",
       "291. FALSE\n",
       "292. FALSE\n",
       "293. FALSE\n",
       "294. FALSE\n",
       "295. FALSE\n",
       "296. FALSE\n",
       "297. FALSE\n",
       "298. FALSE\n",
       "299. FALSE\n",
       "300. FALSE\n",
       "301. FALSE\n",
       "302. FALSE\n",
       "303. FALSE\n",
       "304. FALSE\n",
       "305. FALSE\n",
       "306. FALSE\n",
       "307. FALSE\n",
       "308. FALSE\n",
       "309. FALSE\n",
       "310. FALSE\n",
       "311. FALSE\n",
       "312. FALSE\n",
       "313. FALSE\n",
       "314. FALSE\n",
       "315. FALSE\n",
       "316. FALSE\n",
       "317. FALSE\n",
       "318. FALSE\n",
       "319. FALSE\n",
       "320. FALSE\n",
       "321. FALSE\n",
       "322. FALSE\n",
       "323. FALSE\n",
       "324. FALSE\n",
       "325. FALSE\n",
       "326. FALSE\n",
       "327. FALSE\n",
       "328. FALSE\n",
       "329. FALSE\n",
       "330. FALSE\n",
       "331. FALSE\n",
       "332. FALSE\n",
       "333. FALSE\n",
       "334. FALSE\n",
       "335. FALSE\n",
       "336. FALSE\n",
       "337. FALSE\n",
       "338. FALSE\n",
       "339. FALSE\n",
       "340. FALSE\n",
       "341. FALSE\n",
       "342. FALSE\n",
       "343. FALSE\n",
       "344. FALSE\n",
       "345. FALSE\n",
       "346. FALSE\n",
       "347. FALSE\n",
       "348. FALSE\n",
       "349. FALSE\n",
       "350. FALSE\n",
       "351. FALSE\n",
       "352. FALSE\n",
       "353. FALSE\n",
       "354. FALSE\n",
       "355. FALSE\n",
       "356. FALSE\n",
       "357. FALSE\n",
       "358. FALSE\n",
       "359. FALSE\n",
       "360. FALSE\n",
       "361. FALSE\n",
       "362. FALSE\n",
       "363. FALSE\n",
       "364. FALSE\n",
       "365. FALSE\n",
       "366. FALSE\n",
       "367. FALSE\n",
       "368. FALSE\n",
       "369. FALSE\n",
       "370. FALSE\n",
       "371. FALSE\n",
       "372. FALSE\n",
       "373. FALSE\n",
       "374. FALSE\n",
       "375. FALSE\n",
       "376. FALSE\n",
       "377. FALSE\n",
       "378. FALSE\n",
       "379. FALSE\n",
       "380. FALSE\n",
       "381. FALSE\n",
       "382. FALSE\n",
       "383. FALSE\n",
       "384. FALSE\n",
       "385. FALSE\n",
       "386. FALSE\n",
       "387. FALSE\n",
       "388. FALSE\n",
       "389. FALSE\n",
       "390. FALSE\n",
       "391. FALSE\n",
       "392. FALSE\n",
       "393. FALSE\n",
       "394. FALSE\n",
       "395. FALSE\n",
       "396. FALSE\n",
       "397. FALSE\n",
       "398. FALSE\n",
       "399. FALSE\n",
       "400. FALSE\n",
       "401. FALSE\n",
       "402. FALSE\n",
       "403. FALSE\n",
       "404. FALSE\n",
       "405. FALSE\n",
       "406. FALSE\n",
       "407. FALSE\n",
       "408. FALSE\n",
       "409. FALSE\n",
       "410. FALSE\n",
       "411. FALSE\n",
       "412. FALSE\n",
       "413. FALSE\n",
       "414. FALSE\n",
       "415. FALSE\n",
       "416. FALSE\n",
       "417. FALSE\n",
       "418. FALSE\n",
       "419. FALSE\n",
       "420. FALSE\n",
       "421. FALSE\n",
       "422. FALSE\n",
       "423. FALSE\n",
       "424. FALSE\n",
       "425. FALSE\n",
       "426. FALSE\n",
       "427. FALSE\n",
       "428. FALSE\n",
       "429. FALSE\n",
       "430. FALSE\n",
       "431. FALSE\n",
       "432. FALSE\n",
       "433. FALSE\n",
       "434. FALSE\n",
       "435. FALSE\n",
       "436. FALSE\n",
       "437. FALSE\n",
       "438. FALSE\n",
       "439. FALSE\n",
       "440. FALSE\n",
       "441. FALSE\n",
       "442. FALSE\n",
       "443. FALSE\n",
       "444. FALSE\n",
       "445. FALSE\n",
       "446. FALSE\n",
       "447. FALSE\n",
       "448. FALSE\n",
       "449. FALSE\n",
       "450. FALSE\n",
       "451. FALSE\n",
       "452. FALSE\n",
       "453. FALSE\n",
       "454. FALSE\n",
       "455. FALSE\n",
       "456. FALSE\n",
       "457. FALSE\n",
       "458. FALSE\n",
       "459. FALSE\n",
       "460. FALSE\n",
       "461. FALSE\n",
       "462. FALSE\n",
       "463. FALSE\n",
       "464. FALSE\n",
       "465. FALSE\n",
       "466. FALSE\n",
       "467. FALSE\n",
       "468. FALSE\n",
       "469. FALSE\n",
       "470. FALSE\n",
       "471. FALSE\n",
       "472. FALSE\n",
       "473. FALSE\n",
       "474. FALSE\n",
       "475. FALSE\n",
       "476. FALSE\n",
       "477. FALSE\n",
       "478. FALSE\n",
       "479. FALSE\n",
       "480. FALSE\n",
       "481. FALSE\n",
       "482. FALSE\n",
       "483. FALSE\n",
       "484. FALSE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [13]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE  TRUE FALSE\n",
       " [25] FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE FALSE\n",
       " [37] FALSE  TRUE FALSE  TRUE FALSE  TRUE FALSE  TRUE FALSE  TRUE  TRUE FALSE\n",
       " [49] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE\n",
       " [61] FALSE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       " [73] FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE\n",
       " [85] FALSE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE\n",
       " [97]  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE FALSE\n",
       "[109] FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE\n",
       "[121]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE\n",
       "[133]  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE  TRUE\n",
       "[145]  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[157]  TRUE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE\n",
       "[169] FALSE FALSE FALSE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE\n",
       "[181]  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE\n",
       "[193]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE  TRUE  TRUE\n",
       "[205] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[217] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[229] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[241] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[253] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[265] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[277] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[289] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[301] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[313] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[325] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[337] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[349] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[361] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[373] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[385] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[397] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[409] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[421] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[433] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[445] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[457] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[469] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[481] FALSE FALSE FALSE FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "test.data$pheno.list$GSE19491$status == \"TB\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "control    LTBI    PSLE     SLE   Staph   Still   Strep      TB \n",
       "    133      69      82      28      40      31      12      89 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE19491$status)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE19491$hiv.status = \"negative\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                          title       geo_accession    status         \n",
       " CON_LON_test035             :  1   GSM484368:  1   Length:484        \n",
       " CON_LON_test035_long_0_022  :  1   GSM484369:  1   Class :character  \n",
       " CON_LON_test035_sep_CD4_034 :  1   GSM484370:  1   Mode  :character  \n",
       " CON_LON_test035_sep_CD8_035 :  1   GSM484371:  1                     \n",
       " CON_LON_test035_sep_Mono_033:  1   GSM484372:  1                     \n",
       " CON_LON_test035_sep_Neut_032:  1   GSM484373:  1                     \n",
       " (Other)                     :478   (Other)  :478                     \n",
       "    submission_date    last_update_date  type     channel_count\n",
       " Dec 11 2009: 61    Aug 11 2010:387     RNA:484   Min.   :1    \n",
       " Dec 12 2009:149    Jan 04 2018: 97               1st Qu.:1    \n",
       " Jun 02 2010:193                                  Median :1    \n",
       " Jun 03 2010: 81                                  Mean   :1    \n",
       "                                                  3rd Qu.:1    \n",
       "                                                  Max.   :1    \n",
       "                                                               \n",
       "                                source_name_ch1       organism_ch1\n",
       " Human whole blood                      :274    Homo sapiens:484  \n",
       " Whole Blood from patient with Latent TB: 69                      \n",
       " Whole Blood from patient with Active TB: 54                      \n",
       " Whole Blood from healthy control       : 24                      \n",
       " Whole blood from healthy control       : 12                      \n",
       " CD4+ cells from patient with Active TB :  7                      \n",
       " (Other)                                : 44                      \n",
       " characteristics_ch1    characteristics_ch1.1\n",
       " age: 18: 27         gender: Female:286      \n",
       " age: 16: 20         gender: Male  :198      \n",
       " age: 17: 20                                 \n",
       " age: 28: 15                                 \n",
       " age: 14: 14                                 \n",
       " age: 24: 14                                 \n",
       " (Other):374                                 \n",
       "                 characteristics_ch1.2             characteristics_ch1.3\n",
       " ethnicity: Caucasian       :105       illness: PTB           : 89      \n",
       " ethnicity: Hispanic        : 97       illness: PSLE          : 82      \n",
       " ethnicity: Black           : 86       illness: Staph         : 40      \n",
       " ethnicity: White           : 52       illness: Latent        : 38      \n",
       " ethnicity: African American: 46       illness: Control (BCG+): 34      \n",
       " ethnicity: South Asian     : 44       illness: LATENT TB     : 31      \n",
       " (Other)                    : 54       (Other)                :170      \n",
       "                                 characteristics_ch1.4\n",
       "                                            :274      \n",
       " geographical region: London                :140      \n",
       " geographical region: South Africa          : 51      \n",
       " time post initiation of treatment: 0_months: 19      \n",
       "                                                      \n",
       "                                                      \n",
       "                                                      \n",
       "                   characteristics_ch1.5\n",
       "                              :274      \n",
       " bcg vaccinated: Yes          :121      \n",
       " bcg vaccinated: No           : 53      \n",
       " bcg vaccinated: Not_Known    : 12      \n",
       " sputum smear status: n.a     : 12      \n",
       " sputum smear status: Positive:  6      \n",
       " (Other)                      :  6      \n",
       "                              characteristics_ch1.6\n",
       "                                         :274      \n",
       " isolate sensitivity: Fully_Sensitive    :  6      \n",
       " isolate sensitivity: n.a                : 12      \n",
       " isolate sensitivity: Resistant Isoniazid:  1      \n",
       " region of birth: non_UK_born            : 51      \n",
       " region of birth: non_UK_Born            :113      \n",
       " region of birth: UK_Born                : 27      \n",
       "         characteristics_ch1.7                     characteristics_ch1.8\n",
       "                    :274                                      :293      \n",
       " tst: n.a           : 45       exposure_latent: n.a           :100      \n",
       " tst: 0             : 24       exposure_latent: Contact_Screen: 31      \n",
       " bcg vaccinated: Yes: 17       sputum_smear: n.a              : 31      \n",
       " tst: 15            : 13       sputum_smear: 3+               :  7      \n",
       " tst: 20            : 13       exposure_latent: New_Entrant   :  6      \n",
       " (Other)            : 98       (Other)                        : 16      \n",
       "                      characteristics_ch1.9\n",
       "                                 :293      \n",
       " index_case_disease_site: n.a    : 98      \n",
       " sputum_culture: n.a             : 31      \n",
       " index_case_disease_site: pulm   : 27      \n",
       " sputum_culture: M.tuberculosis  : 20      \n",
       " index_case_disease_site: unknown:  8      \n",
       " (Other)                         :  7      \n",
       "                  characteristics_ch1.10\n",
       "                             :344       \n",
       " smear of index case:        :  4       \n",
       " smear of index case: n.a    :109       \n",
       " smear of index case: Sm-    :  1       \n",
       " smear of index case: Sm+    : 19       \n",
       " smear of index case: unknown:  7       \n",
       "                                        \n",
       "                   characteristics_ch1.11          characteristics_ch1.12\n",
       "                              :344                            :344       \n",
       " modal x-ray grade:           : 16        sputum_smear: +     :  9       \n",
       " modal x-ray grade: Advanced  : 23        sputum_smear: ++    :  1       \n",
       " modal x-ray grade: Minimal   : 18        sputum_smear: +++   :  7       \n",
       " modal x-ray grade: Moderate  : 25        sputum_smear: n.a   : 90       \n",
       " modal x-ray grade: n.a       : 24        sputum_smear: nil   : 30       \n",
       " modal x-ray grade: No disease: 34        sputum_smear: scanty:  3       \n",
       "                       characteristics_ch1.13       characteristics_ch1.14\n",
       "                                  :344                         :344       \n",
       " sputum_culture: n.a              : 89        bal_smear: +     :  2       \n",
       " sputum_culture: no_growth_6_weeks:  3        bal_smear: ++    :  1       \n",
       " sputum_culture: tuberculosis     : 48        bal_smear: +++   :  1       \n",
       "                                              bal_smear: n.a   :104       \n",
       "                                              bal_smear: nil   : 29       \n",
       "                                              bal_smear: scanty:  3       \n",
       "                    characteristics_ch1.15\n",
       "                               :344       \n",
       " bal_culture: n.a              :104       \n",
       " bal_culture: no_growth_6_weeks:  7       \n",
       " bal_culture: tuberculosis     : 29       \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                        characteristics_ch1.16    molecule_ch1\n",
       "                                                   :344        total RNA:484  \n",
       " isolate_sensitivity: n.a                          : 79                       \n",
       " isolate_sensitivity: Fully_sensitive              : 45                       \n",
       " isolate_sensitivity: Resistant Isoniazid + Ethambu:  5                       \n",
       " isolate_sensitivity: Resistant Streptomycin, Ethio:  5                       \n",
       " isolate_sensitivity: Resistant Isoniazid          :  3                       \n",
       " (Other)                                           :  3                       \n",
       "                                                                                                                                              extract_protocol_ch1\n",
       " Ambion Temput Isolation kit                                                                                                                            :274      \n",
       " PerfectPure RNA Blood kit (5 PRIME Inc, Gaithersburg, MD, USA)                                                                                         : 42      \n",
       " Qiagen RNeasy Mini Kit following manufacturer's protocol                                                                                               : 43      \n",
       " Qiagen RNeasy Mini Kit following manufacturer's protocols                                                                                              :  1      \n",
       " RNA was isolated using the MagMAX-96 Blood RNA Isolation Kit (Applied Biosystems/Ambion, Austin, TX, USA) according to the manufacturer's instructions:124      \n",
       "                                                                                                                                                                  \n",
       "                                                                                                                                                                  \n",
       "  label_ch1  \n",
       " biotin:484  \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "                                                                                                                                                                                                    label_protocol_ch1\n",
       " Biotinylated cRNA were generated following manufacturer protocols using the Illumina TotalPrep Amplification Kit from Ambion                                                                                :124     \n",
       " Biotynlated cRNA were generated following manufacturer protocols from 200 - 250ng of the globin-reduced RNA using the Illumina CustomPrep RNA amplification kit (Applied Biosystems/Ambion, Austin, TX, USA): 42     \n",
       " Biotynlated cRNA were generated following manufacturer protocols using the Illumina TotalPrep Amplification Kit from Ambion.                                                                                :318     \n",
       "                                                                                                                                                                                                                      \n",
       "                                                                                                                                                                                                                      \n",
       "                                                                                                                                                                                                                      \n",
       "                                                                                                                                                                                                                      \n",
       "   taxid_ch1   \n",
       " Min.   :9606  \n",
       " 1st Qu.:9606  \n",
       " Median :9606  \n",
       " Mean   :9606  \n",
       " 3rd Qu.:9606  \n",
       " Max.   :9606  \n",
       "               \n",
       "                                                           hyb_protocol\n",
       " Hybridize 12-Sample Beadchip                                    :210  \n",
       " Hybridized 12-Sample Beadchip according to manufacturer protocol:274  \n",
       "                                                                       \n",
       "                                                                       \n",
       "                                                                       \n",
       "                                                                       \n",
       "                                                                       \n",
       "                                                             scan_protocol\n",
       " SentrixScanVersion 2.3.0.13  (This information is in the .XML file):484  \n",
       "                                                                          \n",
       "                                                                          \n",
       "                                                                          \n",
       "                                                                          \n",
       "                                                                          \n",
       "                                                                          \n",
       "                                                                                                                                                                                                                                     description \n",
       " Whole blood was collected in Tempus tube                                                                                                                                                                                                  :274  \n",
       " Whole Blood from patient with Latent TB recruited in South Africa.  The subject tested positive for IGRA test.                                                                                                                            : 31  \n",
       " Whole Blood from healthy control recruited in London. The subject was negative for skin test and IGRA test.                                                                                                                               : 25  \n",
       " Whole Blood from patient with active TB recruited in London. The subject was infected with Mycobacterium tuberculosis and confirmed as culture positive.                                                                                  : 22  \n",
       " Whole Blood from patient with Latent TB  recruited in London .  The subject was an asymptomatic household contact of an active patient or a new entrant from an endemic country and tested positive by both IGRA and tuberculin skin test.: 22  \n",
       " Whole Blood from patient with active TB recruited in South Africa. The subject was infected with Mycobacterium tuberculosis and confirmed as culture positive.                                                                            : 20  \n",
       " (Other)                                                                                                                                                                                                                                   : 90  \n",
       "                                                                                             data_processing\n",
       " The data were analysed using BeadStudio version 1.5.1.3 using the average chip normalization method.:166   \n",
       " The data were analyzed using BeadStudio version 1.5.1.3 using the average chip normalization method.:318   \n",
       "                                                                                                            \n",
       "                                                                                                            \n",
       "                                                                                                            \n",
       "                                                                                                            \n",
       "                                                                                                            \n",
       "  platform_id              contact_name                          contact_email\n",
       " GPL6947:484   Damien,,Chaussabel:484   DChaussabel@benaroyaresearch.org:484  \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                contact_institute      contact_address\n",
       " Baylor Institute for Immunology Research:484     3434 Live Oak:484   \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       " contact_city contact_state contact_zip.postal_code contact_country\n",
       " Dallas:484   TX:484        Min.   :75204           USA:484        \n",
       "                            1st Qu.:75204                          \n",
       "                            Median :75204                          \n",
       "                            Mean   :75204                          \n",
       "                            3rd Qu.:75204                          \n",
       "                            Max.   :75204                          \n",
       "                                                                   \n",
       "                                                                                                    supplementary_file\n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484368/suppl/GSM484368_CON_LON_test035_long_0_022.txt.gz:  1     \n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484369/suppl/GSM484369_CON_LON_test036_long_0_023.txt.gz:  1     \n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484370/suppl/GSM484370_CON_LON_test037_long_0_024.txt.gz:  1     \n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484371/suppl/GSM484371_CON_LON_test038_long_0_025.txt.gz:  1     \n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484372/suppl/GSM484372_CON_LON_test039_long_0_026.txt.gz:  1     \n",
       " ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM484nnn/GSM484373/suppl/GSM484373_CON_LON_test040_long_0_027.txt.gz:  1     \n",
       " (Other)                                                                                                     :478     \n",
       " data_row_count                       relation      age.ch1   \n",
       " Min.   :48791                            :387   18     : 27  \n",
       " 1st Qu.:48803   Reanalyzed by: GSM2886035:  1   16     : 20  \n",
       " Median :48803   Reanalyzed by: GSM2886036:  1   17     : 20  \n",
       " Mean   :48802   Reanalyzed by: GSM2886037:  1   28     : 15  \n",
       " 3rd Qu.:48803   Reanalyzed by: GSM2886038:  1   14     : 14  \n",
       " Max.   :48803   Reanalyzed by: GSM2886039:  1   24     : 14  \n",
       "                 (Other)                  : 92   (Other):374  \n",
       "          bal_culture.ch1 bal_smear.ch1 bcg.vaccinated.ch1\n",
       " n.a              :104    +     :  2    No       : 55     \n",
       " no_growth_6_weeks:  7    ++    :  1    Not Known:  5     \n",
       " tuberculosis     : 29    +++   :  1    Not_Known: 12     \n",
       " NA's             :344    n.a   :104    Yes      :138     \n",
       "                          nil   : 29    NA's     :274     \n",
       "                          scanty:  3                      \n",
       "                          NA's  :344                      \n",
       "          ethnicity.ch1     exposure_latent.ch1  gender.ch1 \n",
       " Caucasian       :105   Contact_Screen: 31      Female:286  \n",
       " Hispanic        : 97   GP_Referral   :  1      Male  :198  \n",
       " Black           : 86   n.a           :100                  \n",
       " White           : 52   New_Entrant   :  6                  \n",
       " African American: 46   Not_Known     :  2                  \n",
       " South Asian     : 44   NA's          :344                  \n",
       " (Other)         : 54                                       \n",
       " geographical.region.ch1      healthy.control.ch1         illness.ch1 \n",
       " London      :140        ASLE           : 17      PTB           : 89  \n",
       " South Africa: 51        pSLE           : 19      PSLE          : 82  \n",
       " NA's        :293        Still          : 22      Staph         : 40  \n",
       "                         Strep and Staph: 23      Latent        : 38  \n",
       "                         NA's           :403      Control (BCG+): 34  \n",
       "                                                  (Other)       :120  \n",
       "                                                  NA's          : 81  \n",
       " index_case_disease_site.ch1        isolate.sensitivity.ch1\n",
       "          :  4               Fully_Sensitive    :  6       \n",
       " extrapulm:  3               n.a                : 12       \n",
       " n.a      : 98               Resistant Isoniazid:  1       \n",
       " pulm     : 27               NA's               :465       \n",
       " unknown  :  8                                             \n",
       " NA's     :344                                             \n",
       "                                                           \n",
       "                  isolate_sensitivity.ch1 modal.x.ray.grade.ch1\n",
       " n.a                          : 79                  : 16       \n",
       " Fully_sensitive              : 45        Advanced  : 23       \n",
       " Resistant Isoniazid + Ethambu:  5        Minimal   : 18       \n",
       " Resistant Streptomycin, Ethio:  5        Moderate  : 25       \n",
       " Resistant Isoniazid          :  3        n.a       : 24       \n",
       " (Other)                      :  3        No disease: 34       \n",
       " NA's                         :344        NA's      :344       \n",
       "  region.of.birth.ch1 smear.of.index.case.ch1 sputum.smear.status.ch1\n",
       " non_UK_born: 51             :  4             n.a     : 12           \n",
       " non_UK_Born:113      n.a    :109             Negative:  1           \n",
       " UK_Born    : 27      Sm-    :  1             Positive:  6           \n",
       " NA's       :293      Sm+    : 19             NA's    :465           \n",
       "                      unknown:  7                                    \n",
       "                      NA's   :344                                    \n",
       "                                                                     \n",
       "         sputum_culture.ch1 sputum_smear.ch1\n",
       " M.tuberculosis   : 20      n.a    :121     \n",
       " n.a              :120      nil    : 30     \n",
       " no_growth_6_weeks:  3      +      :  9     \n",
       " tuberculosis     : 48      +++    :  7     \n",
       " NA's             :293      3+     :  7     \n",
       "                            (Other): 17     \n",
       "                            NA's   :293     \n",
       " time.post.initiation.of.treatment.ch1    tst.ch1     hiv.status       \n",
       " 0_months: 19                          n.a    : 45   Length:484        \n",
       " NA's    :465                          0      : 24   Class :character  \n",
       "                                       15     : 13   Mode  :character  \n",
       "                                       20     : 13                     \n",
       "                                       9      : 10                     \n",
       "                                       (Other): 86                     \n",
       "                                       NA's   :293                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE19491)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                \n",
       "                 CD4+ cells from healthy control\n",
       "  ASLE                                         0\n",
       "  Control                                      0\n",
       "  Control (BCG-)                               0\n",
       "  Control (BCG+)                               4\n",
       "  Latent                                       0\n",
       "  LATENT TB                                    0\n",
       "  PSLE                                         0\n",
       "  PTB                                          0\n",
       "  Staph                                        0\n",
       "  Still                                        0\n",
       "  Strep                                        0\n",
       "                \n",
       "                 CD4+ cells from patient with Active TB\n",
       "  ASLE                                                0\n",
       "  Control                                             0\n",
       "  Control (BCG-)                                      0\n",
       "  Control (BCG+)                                      0\n",
       "  Latent                                              0\n",
       "  LATENT TB                                           0\n",
       "  PSLE                                                0\n",
       "  PTB                                                 7\n",
       "  Staph                                               0\n",
       "  Still                                               0\n",
       "  Strep                                               0\n",
       "                \n",
       "                 CD8+ cells from healthy control\n",
       "  ASLE                                         0\n",
       "  Control                                      0\n",
       "  Control (BCG-)                               0\n",
       "  Control (BCG+)                               4\n",
       "  Latent                                       0\n",
       "  LATENT TB                                    0\n",
       "  PSLE                                         0\n",
       "  PTB                                          0\n",
       "  Staph                                        0\n",
       "  Still                                        0\n",
       "  Strep                                        0\n",
       "                \n",
       "                 CD8+ cells from patient with Active TB Human whole blood\n",
       "  ASLE                                                0                28\n",
       "  Control                                             0                 0\n",
       "  Control (BCG-)                                      0                 0\n",
       "  Control (BCG+)                                      0                 0\n",
       "  Latent                                              0                 0\n",
       "  LATENT TB                                           0                 0\n",
       "  PSLE                                                0                82\n",
       "  PTB                                                 7                 0\n",
       "  Staph                                               0                40\n",
       "  Still                                               0                31\n",
       "  Strep                                               0                12\n",
       "                \n",
       "                 Monocytes from healthy control\n",
       "  ASLE                                        0\n",
       "  Control                                     0\n",
       "  Control (BCG-)                              0\n",
       "  Control (BCG+)                              4\n",
       "  Latent                                      0\n",
       "  LATENT TB                                   0\n",
       "  PSLE                                        0\n",
       "  PTB                                         0\n",
       "  Staph                                       0\n",
       "  Still                                       0\n",
       "  Strep                                       0\n",
       "                \n",
       "                 Monocytes from patient with Active TB\n",
       "  ASLE                                               0\n",
       "  Control                                            0\n",
       "  Control (BCG-)                                     0\n",
       "  Control (BCG+)                                     0\n",
       "  Latent                                             0\n",
       "  LATENT TB                                          0\n",
       "  PSLE                                               0\n",
       "  PTB                                                7\n",
       "  Staph                                              0\n",
       "  Still                                              0\n",
       "  Strep                                              0\n",
       "                \n",
       "                 Neutrophils from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                       0\n",
       "  Control (BCG-)                                0\n",
       "  Control (BCG+)                                4\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Neutrophils from patient with Active TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                               0\n",
       "  LATENT TB                                            0\n",
       "  PSLE                                                 0\n",
       "  PTB                                                  7\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0\n",
       "                \n",
       "                 Whole blood from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                      12\n",
       "  Control (BCG-)                                0\n",
       "  Control (BCG+)                                0\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Whole Blood from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                       0\n",
       "  Control (BCG-)                                6\n",
       "  Control (BCG+)                               18\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Whole Blood from patient with Active TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                               0\n",
       "  LATENT TB                                            0\n",
       "  PSLE                                                 0\n",
       "  PTB                                                 54\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0\n",
       "                \n",
       "                 Whole blood from patient with active TB before treatment\n",
       "  ASLE                                                                  0\n",
       "  Control                                                               0\n",
       "  Control (BCG-)                                                        0\n",
       "  Control (BCG+)                                                        0\n",
       "  Latent                                                                0\n",
       "  LATENT TB                                                             0\n",
       "  PSLE                                                                  0\n",
       "  PTB                                                                   7\n",
       "  Staph                                                                 0\n",
       "  Still                                                                 0\n",
       "  Strep                                                                 0\n",
       "                \n",
       "                 Whole Blood from patient with Latent TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                              38\n",
       "  LATENT TB                                           31\n",
       "  PSLE                                                 0\n",
       "  PTB                                                  0\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE19491$illness.ch1, test.data$pheno.list$GSE19491$source_name_ch1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                \n",
       "                 CD4+ cells from healthy control\n",
       "  ASLE                                         0\n",
       "  Control                                      0\n",
       "  Control (BCG-)                               0\n",
       "  Control (BCG+)                               4\n",
       "  Latent                                       0\n",
       "  LATENT TB                                    0\n",
       "  PSLE                                         0\n",
       "  PTB                                          0\n",
       "  Staph                                        0\n",
       "  Still                                        0\n",
       "  Strep                                        0\n",
       "                \n",
       "                 CD4+ cells from patient with Active TB\n",
       "  ASLE                                                0\n",
       "  Control                                             0\n",
       "  Control (BCG-)                                      0\n",
       "  Control (BCG+)                                      0\n",
       "  Latent                                              0\n",
       "  LATENT TB                                           0\n",
       "  PSLE                                                0\n",
       "  PTB                                                 7\n",
       "  Staph                                               0\n",
       "  Still                                               0\n",
       "  Strep                                               0\n",
       "                \n",
       "                 CD8+ cells from healthy control\n",
       "  ASLE                                         0\n",
       "  Control                                      0\n",
       "  Control (BCG-)                               0\n",
       "  Control (BCG+)                               4\n",
       "  Latent                                       0\n",
       "  LATENT TB                                    0\n",
       "  PSLE                                         0\n",
       "  PTB                                          0\n",
       "  Staph                                        0\n",
       "  Still                                        0\n",
       "  Strep                                        0\n",
       "                \n",
       "                 CD8+ cells from patient with Active TB Human whole blood\n",
       "  ASLE                                                0                28\n",
       "  Control                                             0                 0\n",
       "  Control (BCG-)                                      0                 0\n",
       "  Control (BCG+)                                      0                 0\n",
       "  Latent                                              0                 0\n",
       "  LATENT TB                                           0                 0\n",
       "  PSLE                                                0                82\n",
       "  PTB                                                 7                 0\n",
       "  Staph                                               0                40\n",
       "  Still                                               0                31\n",
       "  Strep                                               0                12\n",
       "                \n",
       "                 Monocytes from healthy control\n",
       "  ASLE                                        0\n",
       "  Control                                     0\n",
       "  Control (BCG-)                              0\n",
       "  Control (BCG+)                              4\n",
       "  Latent                                      0\n",
       "  LATENT TB                                   0\n",
       "  PSLE                                        0\n",
       "  PTB                                         0\n",
       "  Staph                                       0\n",
       "  Still                                       0\n",
       "  Strep                                       0\n",
       "                \n",
       "                 Monocytes from patient with Active TB\n",
       "  ASLE                                               0\n",
       "  Control                                            0\n",
       "  Control (BCG-)                                     0\n",
       "  Control (BCG+)                                     0\n",
       "  Latent                                             0\n",
       "  LATENT TB                                          0\n",
       "  PSLE                                               0\n",
       "  PTB                                                7\n",
       "  Staph                                              0\n",
       "  Still                                              0\n",
       "  Strep                                              0\n",
       "                \n",
       "                 Neutrophils from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                       0\n",
       "  Control (BCG-)                                0\n",
       "  Control (BCG+)                                4\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Neutrophils from patient with Active TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                               0\n",
       "  LATENT TB                                            0\n",
       "  PSLE                                                 0\n",
       "  PTB                                                  7\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0\n",
       "                \n",
       "                 Whole blood from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                      12\n",
       "  Control (BCG-)                                0\n",
       "  Control (BCG+)                                0\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Whole Blood from healthy control\n",
       "  ASLE                                          0\n",
       "  Control                                       0\n",
       "  Control (BCG-)                                6\n",
       "  Control (BCG+)                               18\n",
       "  Latent                                        0\n",
       "  LATENT TB                                     0\n",
       "  PSLE                                          0\n",
       "  PTB                                           0\n",
       "  Staph                                         0\n",
       "  Still                                         0\n",
       "  Strep                                         0\n",
       "                \n",
       "                 Whole Blood from patient with Active TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                               0\n",
       "  LATENT TB                                            0\n",
       "  PSLE                                                 0\n",
       "  PTB                                                 54\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0\n",
       "                \n",
       "                 Whole blood from patient with active TB 12 months after treatment started (i.e. after treatment completed)\n",
       "  ASLE                                                                                                                    0\n",
       "  Control                                                                                                                 0\n",
       "  Control (BCG-)                                                                                                          0\n",
       "  Control (BCG+)                                                                                                          0\n",
       "  Latent                                                                                                                  0\n",
       "  LATENT TB                                                                                                               0\n",
       "  PSLE                                                                                                                    0\n",
       "  PTB                                                                                                                     7\n",
       "  Staph                                                                                                                   0\n",
       "  Still                                                                                                                   0\n",
       "  Strep                                                                                                                   0\n",
       "                \n",
       "                 Whole blood from patient with active TB 2 months after treatment started\n",
       "  ASLE                                                                                  0\n",
       "  Control                                                                               0\n",
       "  Control (BCG-)                                                                        0\n",
       "  Control (BCG+)                                                                        0\n",
       "  Latent                                                                                0\n",
       "  LATENT TB                                                                             0\n",
       "  PSLE                                                                                  0\n",
       "  PTB                                                                                   7\n",
       "  Staph                                                                                 0\n",
       "  Still                                                                                 0\n",
       "  Strep                                                                                 0\n",
       "                \n",
       "                 Whole blood from patient with active TB before treatment\n",
       "  ASLE                                                                  0\n",
       "  Control                                                               0\n",
       "  Control (BCG-)                                                        0\n",
       "  Control (BCG+)                                                        0\n",
       "  Latent                                                                0\n",
       "  LATENT TB                                                             0\n",
       "  PSLE                                                                  0\n",
       "  PTB                                                                   7\n",
       "  Staph                                                                 0\n",
       "  Still                                                                 0\n",
       "  Strep                                                                 0\n",
       "                \n",
       "                 Whole Blood from patient with Latent TB\n",
       "  ASLE                                                 0\n",
       "  Control                                              0\n",
       "  Control (BCG-)                                       0\n",
       "  Control (BCG+)                                       0\n",
       "  Latent                                              38\n",
       "  LATENT TB                                           31\n",
       "  PSLE                                                 0\n",
       "  PTB                                                  0\n",
       "  Staph                                                0\n",
       "  Still                                                0\n",
       "  Strep                                                0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE19491$illness.ch1, test.data$pheno.list$GSE19491$source_name_ch1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Format GSE37250\n",
    "\n",
    "- Other disease is a general indication and is not specified, so I will keep it as \"other_disease\".\n",
    "- Some individuals are HIV positive in this cohort. I will keep this with TB and examine HIV status later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE37250$status = as.character(test.data$pheno.list$GSE37250$disease.state.ch1)\n",
    "test.data$pheno.list$GSE37250$status = ifelse(test.data$pheno.list$GSE37250$status == \"active tuberculosis\", \"TB\",\n",
    "                                              ifelse(test.data$pheno.list$GSE37250$status == \"latent TB infection\", \"LTBI\", \"other_disease\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "         LTBI other_disease            TB \n",
       "          167           175           195 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE37250$status)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "negative positive \n",
       "     263      274 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "test.data$pheno.list$GSE37250$hiv.status = as.character(test.data$pheno.list$GSE37250$hiv.status.ch1)\n",
    "test.data$pheno.list$GSE37250$hiv.status = ifelse(test.data$pheno.list$GSE37250$hiv.status == \"HIV negative\", \"negative\", \"positive\")\n",
    "table(test.data$pheno.list$GSE37250$hiv.status)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                                  title       geo_accession\n",
       " active tuberculosis HIV negative Malawi WB_1_M_39619:  1   GSM914353:  1  \n",
       " active tuberculosis HIV negative Malawi WB_1_M_39668:  1   GSM914354:  1  \n",
       " active tuberculosis HIV negative Malawi WB_1_M_39827:  1   GSM914355:  1  \n",
       " active tuberculosis HIV negative Malawi WB_1_M_39896:  1   GSM914356:  1  \n",
       " active tuberculosis HIV negative Malawi WB_1_M_39956:  1   GSM914357:  1  \n",
       " active tuberculosis HIV negative Malawi WB_1_M_39985:  1   GSM914358:  1  \n",
       " (Other)                                             :531   (Other)  :531  \n",
       "    status             submission_date    last_update_date  type    \n",
       " Length:537         Apr 13 2012:537    Dec 17 2018:537     RNA:537  \n",
       " Class :character                                                   \n",
       " Mode  :character                                                   \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       " channel_count                                    source_name_ch1\n",
       " Min.   :1     Whole Blood from patient with Active TB    :195   \n",
       " 1st Qu.:1     Whole Blood from patient with LTBI         :167   \n",
       " Median :1     Whole Blood from patient with other disease:175   \n",
       " Mean   :1                                                       \n",
       " 3rd Qu.:1                                                       \n",
       " Max.   :1                                                       \n",
       "                                                                 \n",
       "       organism_ch1                         characteristics_ch1\n",
       " Homo sapiens:537   disease state: active tuberculosis:195     \n",
       "                    disease state: latent TB infection:167     \n",
       "                    disease state: other disease      :175     \n",
       "                                                               \n",
       "                                                               \n",
       "                                                               \n",
       "                                                               \n",
       "              characteristics_ch1.1                       characteristics_ch1.2\n",
       " hiv status: HIV negative:263       geographical region: Malawi      :237      \n",
       " hiv status: HIV positive:274       geographical region: South Africa:300      \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "         characteristics_ch1.3    molecule_ch1\n",
       " tissue: whole blood:537       total RNA:537  \n",
       "                                              \n",
       "                                              \n",
       "                                              \n",
       "                                              \n",
       "                                              \n",
       "                                              \n",
       "                                                    extract_protocol_ch1\n",
       " Standard PAXgene blood RNA extraction kit protocol (Qiagen).:537      \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "  label_ch1  \n",
       " biotin:537  \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "                                                                label_protocol_ch1\n",
       " Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).:537     \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "   taxid_ch1                                       hyb_protocol\n",
       " Min.   :9606   Standard Illumina hybridisation protocol.:537  \n",
       " 1st Qu.:9606                                                  \n",
       " Median :9606                                                  \n",
       " Mean   :9606                                                  \n",
       " 3rd Qu.:9606                                                  \n",
       " Max.   :9606                                                  \n",
       "                                                               \n",
       "                              scan_protocol                    description \n",
       " Standard Illumina scanning protocol.:537   Sample name: 5483347021_A:  1  \n",
       "                                            Sample name: 5483347021_B:  1  \n",
       "                                            Sample name: 5483347021_C:  1  \n",
       "                                            Sample name: 5483347021_D:  1  \n",
       "                                            Sample name: 5483347021_E:  1  \n",
       "                                            Sample name: 5483347021_F:  1  \n",
       "                                            (Other)                  :531  \n",
       "                                                                                          data_processing\n",
       " The data were normalised using Genome Studio, with background subtracted and quantile normalised.:537   \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "   platform_id            contact_name                 contact_email\n",
       " GPL10558:537   Victoria,,Wright:537   v.wright@imperial.ac.uk:537  \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                contact_laboratory   contact_department\n",
       " Wright Fleming Institute:537      Paediatrics:537     \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "               contact_institute      contact_address contact_city\n",
       " Imperial College London:537     Norfolk Place:537    London:537  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       " contact_zip.postal_code       contact_country supplementary_file\n",
       " W2 1PG:537              United Kingdom:537    NONE:537          \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       " data_row_count            disease.state.ch1 geographical.region.ch1\n",
       " Min.   :47323   active tuberculosis:195     Malawi      :237       \n",
       " 1st Qu.:47323   latent TB infection:167     South Africa:300       \n",
       " Median :47323   other disease      :175                            \n",
       " Mean   :47323                                                      \n",
       " 3rd Qu.:47323                                                      \n",
       " Max.   :47323                                                      \n",
       "                                                                    \n",
       "      hiv.status.ch1       tissue.ch1   hiv.status       \n",
       " HIV negative:263    whole blood:537   Length:537        \n",
       " HIV positive:274                      Class :character  \n",
       "                                       Mode  :character  \n",
       "                                                         \n",
       "                                                         \n",
       "                                                         \n",
       "                                                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE37250)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39619 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39668 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39827 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39896 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39956 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_39985 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_40029 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_40176 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_40541 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_40553 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_41546 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_41770 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_43286 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_43551 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_44574 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_45205 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_45806 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_45842 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_46375 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_47431 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_48995 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_49487 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_49748 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_49750 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_49879 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_49985 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_50243 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_50349 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_54650 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_55841 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_56253 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_56643 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_56746 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_57253 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_57256 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_57755 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_58408 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_60390 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_60636 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_61073 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_61234 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_62891 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_63420 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_63877 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_64180 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_64274 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_64284 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_66845 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_66851 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_66933 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV negative Malawi WB_1_M_68875 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20024 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20054 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20055 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20056 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20058 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20063 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20064 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20065 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20067 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20069 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20071 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20076 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20077 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20078 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20081 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20082 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20086 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20087 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20088 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20090 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20091 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20092 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20096 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20097 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20098 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20100 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20129 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20271 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20275 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20301 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20302 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20303 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20308 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20310 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20354 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20357 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20406 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20412 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20433 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20455 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20503 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20511 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20512 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20513 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20694 \n",
       "                                                          1 \n",
       "active tuberculosis HIV negative South Africa WB_1_SA_20696 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_39463 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_39912 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_40181 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_40231 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_41193 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_41530 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_41725 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_41913 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_42482 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_42488 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_43475 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_43546 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_43890 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_43904 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_44308 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_44379 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_44943 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_45797 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_46619 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_48857 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_49485 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_50347 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_52894 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_53505 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_54008 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_54380 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_54732 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_56615 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_56617 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_57657 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_58187 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_58972 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_58974 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_59188 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_60378 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_60633 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_61806 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_61953 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_62711 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_62897 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_62934 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_63566 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_63702 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_64177 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_66174 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_66481 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_66513 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_67166 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_68169 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_68524 \n",
       "                                                          1 \n",
       "       active tuberculosis HIV positive Malawi WB_2_M_70040 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20059 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20060 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20062 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20066 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20068 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20072 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20073 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20074 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20079 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20083 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20084 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20085 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20093 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20094 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20099 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20132 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20135 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20309 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20311 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20313 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20316 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20320 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20326 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20329 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20331 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20336 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20337 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20343 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20344 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20347 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20349 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20351 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20371 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20373 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20381 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20472 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20478 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20487 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20518 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20583 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20613 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20623 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20642 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20706 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20728 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20733 \n",
       "                                                          1 \n",
       "active tuberculosis HIV positive South Africa WB_2_SA_20758 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39570 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39628 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39633 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39639 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39682 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39731 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39741 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39764 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39786 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39797 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39837 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39923 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_39935 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40220 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40225 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40449 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40508 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40547 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40667 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40790 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40910 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_40913 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_41177 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_41182 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_41439 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_41828 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_43816 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_43945 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_44954 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_44958 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_45131 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_46165 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_49481 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_49704 \n",
       "                                                          1 \n",
       "       latent TB infection HIV negative Malawi WB_3_M_50458 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20022 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20026 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20028 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20029 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20031 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20033 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20035 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20036 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20037 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20038 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20041 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20045 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20047 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20052 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20362 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20364 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20367 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20369 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20372 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20377 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20401 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20402 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20427 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20429 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20430 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20431 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20435 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20436 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20440 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20451 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20453 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20491 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20492 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20496 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20527 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20528 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20531 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20536 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20537 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20540 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20542 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20543 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20545 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20546 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20548 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20550 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20551 \n",
       "                                                          1 \n",
       "latent TB infection HIV negative South Africa WB_3_SA_20552 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39569 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39792 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39843 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39848 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39861 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_39878 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_41483 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_41861 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_41865 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_41869 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_41988 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_42052 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_42151 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_42155 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_42166 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_43072 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_43153 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_44431 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_44555 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_44946 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_45135 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_45604 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_45692 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_45976 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46162 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46346 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46362 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46370 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46605 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_46925 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_47819 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_47892 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_48851 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_49701 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_51573 \n",
       "                                                          1 \n",
       "       latent TB infection HIV positive Malawi WB_4_M_52910 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20050 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20306 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20315 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20318 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20321 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20323 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20324 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20325 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20332 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20339 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20340 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20348 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20370 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20374 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20375 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20376 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20386 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20390 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20425 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20444 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20457 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20464 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20471 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20473 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20475 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20479 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20480 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20498 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20501 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20510 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20519 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20520 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20524 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20525 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20526 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20584 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20586 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20587 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20590 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20592 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20594 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20596 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20603 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20605 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20608 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20614 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20616 \n",
       "                                                          1 \n",
       "latent TB infection HIV positive South Africa WB_4_SA_20617 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_50707 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_54180 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_54181 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_54779 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_55329 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_56646 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_58411 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_59181 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_61150 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_61637 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_64368 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_64832 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_67039 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_67228 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_67534 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_67644 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_68527 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_68935 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_68936 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_68969 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_69710 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_69950 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_70283 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_71534 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_72307 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_72333 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_72828 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73290 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73435 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73550 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73577 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73689 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_73767 \n",
       "                                                          1 \n",
       "             other disease HIV negative Malawi WB_6_M_74459 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20102 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20122 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20125 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20151 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20158 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20160 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20161 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20166 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20167 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20168 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20171 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20172 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20179 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20183 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20184 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20185 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20189 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20190 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20193 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20200 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20204 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20206 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20211 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20213 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20214 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20215 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20216 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20217 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20218 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20219 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20220 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20222 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20224 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20232 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20233 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20234 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20235 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20238 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20240 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20241 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20244 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20245 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20246 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20249 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20250 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20251 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20255 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20257 \n",
       "                                                          1 \n",
       "      other disease HIV negative South Africa WB_6_SA_20263 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_53954 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_54646 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_54651 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_54778 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_54897 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_55728 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_55925 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_55971 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_56880 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_58793 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_60088 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_64189 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_64285 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_64419 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_65173 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_65339 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_65948 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_66600 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_67538 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_68495 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_68791 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_68878 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_69616 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_70039 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_70453 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_71982 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_72114 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_72653 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_73687 \n",
       "                                                          1 \n",
       "             other disease HIV positive Malawi WB_5_M_74083 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20103 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20104 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20107 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20120 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20123 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20126 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20134 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20139 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20141 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20142 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20143 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20159 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20175 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20187 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20192 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20197 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20205 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20221 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20228 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20236 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20237 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20242 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20248 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20254 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20259 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20262 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20266 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20267 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20269 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20270 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20273 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20280 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20282 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20286 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20287 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20293 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20294 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20295 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20297 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20300 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20651 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20652 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20654 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20655 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20658 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20661 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20662 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20664 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20665 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20667 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20669 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20670 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20673 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20674 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20675 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20677 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20678 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20679 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20683 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20685 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20686 \n",
       "                                                          1 \n",
       "      other disease HIV positive South Africa WB_5_SA_20688 \n",
       "                                                          1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE37250$title)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Remainder except GSE42831 ('GSE42825', 'GSE42826', 'GSE42827', 'GSE42830', 'GSE42832')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'GSE19491'</li>\n",
       "\t<li>'GSE37250'</li>\n",
       "\t<li>'GSE39939'</li>\n",
       "\t<li>'GSE39940'</li>\n",
       "\t<li>'GSE42825'</li>\n",
       "\t<li>'GSE42826'</li>\n",
       "\t<li>'GSE42827'</li>\n",
       "\t<li>'GSE42830'</li>\n",
       "\t<li>'GSE42831'</li>\n",
       "\t<li>'GSE42832'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GSE19491'\n",
       "\\item 'GSE37250'\n",
       "\\item 'GSE39939'\n",
       "\\item 'GSE39940'\n",
       "\\item 'GSE42825'\n",
       "\\item 'GSE42826'\n",
       "\\item 'GSE42827'\n",
       "\\item 'GSE42830'\n",
       "\\item 'GSE42831'\n",
       "\\item 'GSE42832'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GSE19491'\n",
       "2. 'GSE37250'\n",
       "3. 'GSE39939'\n",
       "4. 'GSE39940'\n",
       "5. 'GSE42825'\n",
       "6. 'GSE42826'\n",
       "7. 'GSE42827'\n",
       "8. 'GSE42830'\n",
       "9. 'GSE42831'\n",
       "10. 'GSE42832'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GSE19491\" \"GSE37250\" \"GSE39939\" \"GSE39940\" \"GSE42825\" \"GSE42826\"\n",
       " [7] \"GSE42827\" \"GSE42830\" \"GSE42831\" \"GSE42832\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(test.data$pheno.list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "to.analyze = c('GSE42825', 'GSE42826', 'GSE42827', 'GSE42830', 'GSE42832')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "for (gse in to.analyze) {\n",
    "    pheno = test.data$pheno.list[[gse]]\n",
    "    pheno$status = as.character(pheno$disease.state.ch1)\n",
    "    pheno$status = ifelse(grepl(\"arcoid\", pheno$status), \"sarcoidosis\",\n",
    "                         ifelse(grepl(\"Control\", pheno$status), \"control\",\n",
    "                               ifelse(grepl(\"lung cancer\", pheno$status), \"lung_cancer\",\n",
    "                                     ifelse(grepl(\"Pneumonia|Baseline\", pheno$status), \"pneumonia\", pheno$status))))\n",
    "    pheno$hiv.status = as.character(\"negative\")\n",
    "    test.data$pheno.list[[gse]] = pheno\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[1]]\n",
       "\n",
       "    control sarcoidosis          TB \n",
       "         23          11           8 \n",
       "\n",
       "[[2]]\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         52           8           6          25          11 \n",
       "\n",
       "[[3]]\n",
       "\n",
       "pneumonia \n",
       "       10 \n",
       "\n",
       "[[4]]\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         38           8           8          25          16 \n",
       "\n",
       "[[5]]\n",
       "\n",
       "    control sarcoidosis          TB \n",
       "         30          30          30 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(to.analyze, function(x) {table(test.data$pheno.list[[x]][,\"status\"])})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[1]]\n",
       "\n",
       "    Active sarcoidosis                Control Non-active sarcoidosis \n",
       "                     6                     23                      5 \n",
       "                    TB \n",
       "                     8 \n",
       "\n",
       "[[2]]\n",
       "\n",
       "        Active Sarcoid                Control            lung cancer \n",
       "                    16                     52                      8 \n",
       "Non-active sarcoidosis              Pneumonia                     TB \n",
       "                     9                      6                     11 \n",
       "\n",
       "[[3]]\n",
       "\n",
       "pneumonia \n",
       "       10 \n",
       "\n",
       "[[4]]\n",
       "\n",
       "        Active Sarcoid               Baseline                Control \n",
       "                    17                      8                     38 \n",
       "           lung cancer Non-active sarcoidosis                     TB \n",
       "                     8                      8                     16 \n",
       "\n",
       "[[5]]\n",
       "\n",
       "Control Sarcoid      TB \n",
       "     30      30      30 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(to.analyze, function(x) {table(test.data$pheno.list[[x]][,\"disease.state.ch1\"])})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                              title      geo_accession    status         \n",
       " Pneumonia Sample 18 training set:1   GSM1050945:1     Length:8          \n",
       " Pneumonia Sample 33 training set:1   GSM1050960:1     Class :character  \n",
       " Pneumonia Sample 36 training set:1   GSM1050963:1     Mode  :character  \n",
       " Pneumonia Sample 38 training set:1   GSM1050965:1                       \n",
       " Pneumonia Sample 47 training set:1   GSM1050974:1                       \n",
       " Pneumonia Sample 63 training set:1   GSM1050990:1                       \n",
       " (Other)                         :2   (Other)   :2                       \n",
       "    submission_date    last_update_date  type   channel_count\n",
       " Dec 10 2012:8      Oct 31 2013:8       RNA:8   Min.   :1    \n",
       "                                                1st Qu.:1    \n",
       "                                                Median :1    \n",
       "                                                Mean   :1    \n",
       "                                                3rd Qu.:1    \n",
       "                                                Max.   :1    \n",
       "                                                             \n",
       "    source_name_ch1       organism_ch1 characteristics_ch1\n",
       " Whole Blood:8      Homo sapiens:8     gender: F:2        \n",
       "                                       gender: M:6        \n",
       "                                                          \n",
       "                                                          \n",
       "                                                          \n",
       "                                                          \n",
       "                                                          \n",
       "                    characteristics_ch1.1             characteristics_ch1.2\n",
       " ethnicity: Caucasian          :6         disease state: Baseline:8        \n",
       " ethnicity: Indian subcontinent:1                                          \n",
       " ethnicity: Middle Eastern     :1                                          \n",
       "                                                                           \n",
       "                                                                           \n",
       "                                                                           \n",
       "                                                                           \n",
       "         characteristics_ch1.3                treatment_protocol_ch1\n",
       " tissue: whole blood:8         Direct ex-vivo human blood:8         \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                             growth_protocol_ch1    molecule_ch1\n",
       " Blood was taken into Tempus and frozen:8        total RNA:8    \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                                         extract_protocol_ch1\n",
       " RNA was extracted using the MagMAX-96 Blood RNA Isolation Kit and globin reduced using GLOBINclear:8        \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "  label_ch1\n",
       " biotin:8  \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "                                                                                      label_protocol_ch1\n",
       " Biotinylated, amplified cRNA were prepared using the Illumina CustomPrep RNA amplification kit:8       \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "   taxid_ch1                                      hyb_protocol\n",
       " Min.   :9606   Standard Illumina hybridization protocol:8    \n",
       " 1st Qu.:9606                                                 \n",
       " Median :9606                                                 \n",
       " Mean   :9606                                                 \n",
       " 3rd Qu.:9606                                                 \n",
       " Max.   :9606                                                 \n",
       "                                                              \n",
       "                             scan_protocol\n",
       " Standard Illumina scanning protocol:8    \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                                                                                                                    data_processing\n",
       " Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have only had background subtraction:8     \n",
       "                                                                                                                                                   \n",
       "                                                                                                                                                   \n",
       "                                                                                                                                                   \n",
       "                                                                                                                                                   \n",
       "                                                                                                                                                   \n",
       "                                                                                                                                                   \n",
       "   platform_id       contact_name\n",
       " GPL10558:8    Chloe,,Bloom:8    \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                   contact_institute     contact_address\n",
       " MRC National Institute for Medical Research:8       The Ridgeway:8     \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       " contact_city contact_zip.postal_code       contact_country supplementary_file\n",
       " London:8     NW7 1AA:8               United Kingdom:8      NONE:8            \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       " data_row_count  disease.state.ch1             ethnicity.ch1 gender.ch1\n",
       " Min.   :47323   Baseline:8        Caucasian          :6     F:2       \n",
       " 1st Qu.:47323                     Indian subcontinent:1     M:6       \n",
       " Median :47323                     Middle Eastern     :1               \n",
       " Mean   :47323                                                         \n",
       " 3rd Qu.:47323                                                         \n",
       " Max.   :47323                                                         \n",
       "                                                                       \n",
       "       tissue.ch1  hiv.status       \n",
       " whole blood:8    Length:8          \n",
       "                  Class :character  \n",
       "                  Mode  :character  \n",
       "                                    \n",
       "                                    \n",
       "                                    \n",
       "                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(droplevels(test.data$pheno.list$GSE42830[test.data$pheno.list$GSE42830$disease.state.ch1 == \"Baseline\",]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                              title       geo_accession    status         \n",
       " Control Sample 10 validation set: 1   GSM1050732: 1    Length:42         \n",
       " Control Sample 11 validation set: 1   GSM1050733: 1    Class :character  \n",
       " Control Sample 12 validation set: 1   GSM1050734: 1    Mode  :character  \n",
       " Control Sample 13 validation set: 1   GSM1050735: 1                      \n",
       " Control Sample 14 validation set: 1   GSM1050736: 1                      \n",
       " Control Sample 15 validation set: 1   GSM1050737: 1                      \n",
       " (Other)                         :36   (Other)   :36                      \n",
       "    submission_date    last_update_date  type    channel_count\n",
       " Dec 10 2012:42     Oct 31 2013:42      RNA:42   Min.   :1    \n",
       "                                                 1st Qu.:1    \n",
       "                                                 Median :1    \n",
       "                                                 Mean   :1    \n",
       "                                                 3rd Qu.:1    \n",
       "                                                 Max.   :1    \n",
       "                                                              \n",
       "    source_name_ch1       organism_ch1   characteristics_ch1\n",
       " Whole Blood:42     Homo sapiens:42    gender: F   :28      \n",
       "                                       gender: M   :13      \n",
       "                                       gender: None: 1      \n",
       "                                                            \n",
       "                                                            \n",
       "                                                            \n",
       "                                                            \n",
       "                    characteristics_ch1.1\n",
       " ethnicity: Black              :15       \n",
       " ethnicity: Central Asia       : 1       \n",
       " ethnicity: Indian subcontinent: 6       \n",
       " ethnicity: None               : 1       \n",
       " ethnicity: White              :19       \n",
       "                                         \n",
       "                                         \n",
       "                           characteristics_ch1.2         characteristics_ch1.3\n",
       " disease state: Active sarcoidosis    : 6        tissue: whole blood:42       \n",
       " disease state: Control               :23                                     \n",
       " disease state: Non-active sarcoidosis: 5                                     \n",
       " disease state: TB                    : 8                                     \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                treatment_protocol_ch1\n",
       " Direct ex-vivo human blood:42        \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                             growth_protocol_ch1    molecule_ch1\n",
       " Blood was taken into Tempus and frozen:42       total RNA:42   \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                                                                                 extract_protocol_ch1\n",
       " RNon-active sarcoidosis was extracted using the MagMAX-96 Blood RNon-active sarcoidosis Isolation Kit and globin reduced using GLOBINclear:42       \n",
       "                                                                                                                                                     \n",
       "                                                                                                                                                     \n",
       "                                                                                                                                                     \n",
       "                                                                                                                                                     \n",
       "                                                                                                                                                     \n",
       "                                                                                                                                                     \n",
       "  label_ch1 \n",
       " biotin:42  \n",
       "            \n",
       "            \n",
       "            \n",
       "            \n",
       "            \n",
       "            \n",
       "                                                                                                                                                  label_protocol_ch1\n",
       " Biotinylated, amplified cRNon-active sarcoidosis were prepared using the IllumiNon-active sarcoidosis CustomPrep RNon-active sarcoidosis amplification kit:42      \n",
       "                                                                                                                                                                    \n",
       "                                                                                                                                                                    \n",
       "                                                                                                                                                                    \n",
       "                                                                                                                                                                    \n",
       "                                                                                                                                                                    \n",
       "                                                                                                                                                                    \n",
       "   taxid_ch1   \n",
       " Min.   :9606  \n",
       " 1st Qu.:9606  \n",
       " Median :9606  \n",
       " Mean   :9606  \n",
       " 3rd Qu.:9606  \n",
       " Max.   :9606  \n",
       "               \n",
       "                                                       hyb_protocol\n",
       " Standard IllumiNon-active sarcoidosis hybridization protocol:42   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                 scan_protocol\n",
       " Standard IllumiNon-active sarcoidosis scanning protocol:42   \n",
       "                                                              \n",
       "                                                              \n",
       "                                                              \n",
       "                                                              \n",
       "                                                              \n",
       "                                                              \n",
       "                                                                        data_processing\n",
       " The data were normalised using 75th percentile normalisation in GeneSpring 11.5:42    \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                                                                   data_processing.1\n",
       " Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have only had background subtraction:42     \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "   platform_id       contact_name\n",
       " GPL10558:42   Chloe,,Bloom:42   \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                   contact_institute     contact_address\n",
       " MRC National Institute for Medical Research:42      The Ridgeway:42    \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       " contact_city contact_zip.postal_code       contact_country supplementary_file\n",
       " London:42    NW7 1AA:42              United Kingdom:42     NONE:42           \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       " data_row_count               disease.state.ch1             ethnicity.ch1\n",
       " Min.   :47323   Active sarcoidosis    : 6      Black              :15   \n",
       " 1st Qu.:47323   Control               :23      Central Asia       : 1   \n",
       " Median :47323   Non-active sarcoidosis: 5      Indian subcontinent: 6   \n",
       " Mean   :47323   TB                    : 8      None               : 1   \n",
       " 3rd Qu.:47323                                  White              :19   \n",
       " Max.   :47323                                                           \n",
       "                                                                         \n",
       " gender.ch1       tissue.ch1  hiv.status       \n",
       " F   :28    whole blood:42   Length:42         \n",
       " M   :13                     Class :character  \n",
       " None: 1                     Mode  :character  \n",
       "                                               \n",
       "                                               \n",
       "                                               \n",
       "                                               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE42825)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Format GSE42831\n",
    "\n",
    "- All samples are sarcoidosis samples. Just take sarcoidosis pre-treatment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE42831$status = as.character(\"sarcoidosis\")\n",
    "test.data$pheno.list$GSE42831$hiv.status = as.character(\"negative\")\n",
    "test.data$pheno.list$GSE42831 = droplevels(test.data$pheno.list$GSE42831[test.data$pheno.list$GSE42831$treament.ch1 == \"Pre-treatment\",])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                                             title      geo_accession\n",
       " Sarcoid Sample 10 6011 Pre-treatment time pt. 1:1   GSM1051029:1    \n",
       " Sarcoid Sample 11 2010 Pre-treatment time pt. 1:1   GSM1051030:1    \n",
       " Sarcoid Sample 14 2140 Pre-treatment time pt. 1:1   GSM1051031:1    \n",
       " Sarcoid Sample 15 6009 Pre-treatment time pt. 1:1   GSM1051032:1    \n",
       " Sarcoid Sample 7 2099 Pre-treatment time pt. 1 :1   GSM1051033:1    \n",
       " Sarcoid Sample 8 2018 Pre-treatment time pt. 1 :1   GSM1051036:1    \n",
       " Sarcoid Sample 9 2100 Pre-treatment time pt. 1 :1   GSM1051037:1    \n",
       "    status             submission_date    last_update_date  type  \n",
       " Length:7           Dec 10 2012:7      Oct 31 2013:7       RNA:7  \n",
       " Class :character                                                 \n",
       " Mode  :character                                                 \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       " channel_count    source_name_ch1       organism_ch1\n",
       " Min.   :1     Whole Blood:7      Homo sapiens:7    \n",
       " 1st Qu.:1                                          \n",
       " Median :1                                          \n",
       " Mean   :1                                          \n",
       " 3rd Qu.:1                                          \n",
       " Max.   :1                                          \n",
       "                                                    \n",
       "              characteristics_ch1             characteristics_ch1.1\n",
       " treament: Pre-treatment:7        response: Pre-treatment:7        \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       "                                                                   \n",
       " characteristics_ch1.2   characteristics_ch1.3\n",
       " time (d): 1:7         subject: 2010:1        \n",
       "                       subject: 2018:1        \n",
       "                       subject: 2099:1        \n",
       "                       subject: 2100:1        \n",
       "                       subject: 2140:1        \n",
       "                       subject: 6009:1        \n",
       "                       subject: 6011:1        \n",
       "                treatment_protocol_ch1\n",
       " Direct ex-vivo human blood:7         \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                                      \n",
       "                             growth_protocol_ch1    molecule_ch1\n",
       " Blood was taken into Tempus and frozen:7        total RNA:7    \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                \n",
       "                                                                                         extract_protocol_ch1\n",
       " RNA was extracted using the MagMAX-96 Blood RNA Isolation Kit and globin reduced using GLOBINclear:7        \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "                                                                                                             \n",
       "  label_ch1\n",
       " biotin:7  \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "           \n",
       "                                                                                      label_protocol_ch1\n",
       " Biotinylated, amplified cRNA were prepared using the Illumina CustomPrep RNA amplification kit:7       \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "                                                                                                        \n",
       "   taxid_ch1                                      hyb_protocol\n",
       " Min.   :9606   Standard Illumina hybridization protocol:7    \n",
       " 1st Qu.:9606                                                 \n",
       " Median :9606                                                 \n",
       " Mean   :9606                                                 \n",
       " 3rd Qu.:9606                                                 \n",
       " Max.   :9606                                                 \n",
       "                                                              \n",
       "                             scan_protocol\n",
       " Standard Illumina scanning protocol:7    \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                          \n",
       "                                                                        data_processing\n",
       " The data were normalised using 75th percentile normalisation in GeneSpring 11.5:7     \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                       \n",
       "                                                                                                                                   data_processing.1\n",
       " Background subtracted, normalised and log2 transformed for normalised values and non-normalised values have only had background subtraction:7      \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "                                                                                                                                                    \n",
       "   platform_id       contact_name\n",
       " GPL10558:7    Chloe,,Bloom:7    \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                 \n",
       "                                   contact_institute     contact_address\n",
       " MRC National Institute for Medical Research:7       The Ridgeway:7     \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       " contact_city contact_zip.postal_code       contact_country supplementary_file\n",
       " London:7     NW7 1AA:7               United Kingdom:7      NONE:7            \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       "                                                                              \n",
       " data_row_count         response.ch1  subject.ch1    time..d..ch1\n",
       " Min.   :47323   Pre-treatment:7     Min.   :2010   Min.   :1    \n",
       " 1st Qu.:47323                       1st Qu.:2058   1st Qu.:1    \n",
       " Median :47323                       Median :2100   Median :1    \n",
       " Mean   :47323                       Mean   :3198   Mean   :1    \n",
       " 3rd Qu.:47323                       3rd Qu.:4074   3rd Qu.:1    \n",
       " Max.   :47323                       Max.   :6011   Max.   :1    \n",
       "                                                                 \n",
       "        treament.ch1  hiv.status       \n",
       " Pre-treatment:7     Length:7          \n",
       "                     Class :character  \n",
       "                     Mode  :character  \n",
       "                                       \n",
       "                                       \n",
       "                                       \n",
       "                                       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE42831)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Sarcoid Sample 10 6011 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       "Sarcoid Sample 11 2010 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       "Sarcoid Sample 14 2140 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       "Sarcoid Sample 15 6009 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       " Sarcoid Sample 7 2099 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       " Sarcoid Sample 8 2018 Pre-treatment time pt. 1 \n",
       "                                              1 \n",
       " Sarcoid Sample 9 2100 Pre-treatment time pt. 1 \n",
       "                                              1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE42831$title)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Format GSE39939"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "              title        geo_accession                   status   \n",
       " WB_20_KIL_100091:  1   GSM1249177:  1   Public on May 01 2014:157  \n",
       " WB_20_KIL_100360:  1   GSM1249178:  1                              \n",
       " WB_20_KIL_100398:  1   GSM1249179:  1                              \n",
       " WB_20_KIL_100406:  1   GSM1249180:  1                              \n",
       " WB_20_KIL_500086:  1   GSM1249181:  1                              \n",
       " WB_20_KIL_500119:  1   GSM1249182:  1                              \n",
       " (Other)         :151   (Other)   :151                              \n",
       "    submission_date    last_update_date  type     channel_count\n",
       " Oct 23 2013:157    May 01 2014:157     RNA:157   Min.   :1    \n",
       "                                                  1st Qu.:1    \n",
       "                                                  Median :1    \n",
       "                                                  Mean   :1    \n",
       "                                                  3rd Qu.:1    \n",
       "                                                  Max.   :1    \n",
       "                                                               \n",
       "                                    source_name_ch1       organism_ch1\n",
       " Whole Blood from patient with Active TB    :79     Homo sapiens:157  \n",
       " Whole Blood from patient with LTBI         :14                       \n",
       " Whole Blood from patient with other disease:64                       \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                       characteristics_ch1\n",
       " illness: active tuberculosis (culture confirmed):35      \n",
       " illness: active tuberculosis (culture negative) :44      \n",
       " illness: latent TB infection                    :14      \n",
       " illness: other disease                          :55      \n",
       " illness: other disease (IGRA +)                 : 9      \n",
       "                                                          \n",
       "                                                          \n",
       "              characteristics_ch1.1                characteristics_ch1.2\n",
       " hiv status: HIV negative:103       geographical region: Kenya:157      \n",
       " hiv status: HIV positive: 54                                           \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "    molecule_ch1\n",
       " total RNA:157  \n",
       "                \n",
       "                \n",
       "                \n",
       "                \n",
       "                \n",
       "                \n",
       "                                                    extract_protocol_ch1\n",
       " Standard PAXgene blood RNA extraction kit protocol (Qiagen).:157      \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "  label_ch1  \n",
       " biotin:157  \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "                                                                label_protocol_ch1\n",
       " Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).:157     \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "   taxid_ch1                                       hyb_protocol\n",
       " Min.   :9606   Standard Illumina hybridisation protocol.:157  \n",
       " 1st Qu.:9606                                                  \n",
       " Median :9606                                                  \n",
       " Mean   :9606                                                  \n",
       " 3rd Qu.:9606                                                  \n",
       " Max.   :9606                                                  \n",
       "                                                               \n",
       "                              scan_protocol\n",
       " Standard Illumina scanning protocol.:157  \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                                                                          data_processing\n",
       " The data were normalised using Genome Studio, with background subtracted and quantile normalised.:157   \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "   platform_id            contact_name                 contact_email\n",
       " GPL10558:157   Victoria,,Wright:157   v.wright@imperial.ac.uk:157  \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                contact_laboratory   contact_department\n",
       " Wright Fleming Institute:157      Paediatrics:157     \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "               contact_institute      contact_address contact_city\n",
       " Imperial College London:157     Norfolk Place:157    London:157  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       " contact_zip.postal_code       contact_country supplementary_file\n",
       " W2 1PG:157              United Kingdom:157    NONE:157          \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       " data_row_count  geographical.region.ch1      hiv.status.ch1\n",
       " Min.   :47323   Kenya:157               HIV negative:103   \n",
       " 1st Qu.:47323                           HIV positive: 54   \n",
       " Median :47323                                              \n",
       " Mean   :47323                                              \n",
       " 3rd Qu.:47323                                              \n",
       " Max.   :47323                                              \n",
       "                                                            \n",
       "                                  illness.ch1\n",
       " active tuberculosis (culture confirmed):35  \n",
       " active tuberculosis (culture negative) :44  \n",
       " latent TB infection                    :14  \n",
       " other disease                          :55  \n",
       " other disease (IGRA +)                 : 9  \n",
       "                                             \n",
       "                                             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE39939)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE39939$status = as.character(test.data$pheno.list$GSE39939$illness.ch1)\n",
    "test.data$pheno.list$GSE39939$status = ifelse(grepl(\"active tuberculosis\", test.data$pheno.list$GSE39939$status), \"TB\",\n",
    "                                             ifelse(grepl(\"other disease\", test.data$pheno.list$GSE39939$status), \"other_disease\",\n",
    "                                                    \"LTBI\"))\n",
    "test.data$pheno.list$GSE39939$hiv.status = as.character(test.data$pheno.list$GSE39939$hiv.status.ch1)\n",
    "test.data$pheno.list$GSE39939$hiv.status = ifelse(test.data$pheno.list$GSE39939$hiv.status == \"HIV negative\", \"negative\", \"positive\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'negative'</li>\n",
       "\t<li>'positive'</li>\n",
       "\t<li>'negative'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'negative'\n",
       "\\item 'positive'\n",
       "\\item 'negative'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'positive'\n",
       "2. 'positive'\n",
       "3. 'positive'\n",
       "4. 'positive'\n",
       "5. 'positive'\n",
       "6. 'positive'\n",
       "7. 'positive'\n",
       "8. 'positive'\n",
       "9. 'positive'\n",
       "10. 'positive'\n",
       "11. 'negative'\n",
       "12. 'negative'\n",
       "13. 'negative'\n",
       "14. 'negative'\n",
       "15. 'negative'\n",
       "16. 'negative'\n",
       "17. 'negative'\n",
       "18. 'negative'\n",
       "19. 'negative'\n",
       "20. 'negative'\n",
       "21. 'negative'\n",
       "22. 'negative'\n",
       "23. 'negative'\n",
       "24. 'negative'\n",
       "25. 'negative'\n",
       "26. 'negative'\n",
       "27. 'negative'\n",
       "28. 'negative'\n",
       "29. 'negative'\n",
       "30. 'negative'\n",
       "31. 'negative'\n",
       "32. 'negative'\n",
       "33. 'negative'\n",
       "34. 'negative'\n",
       "35. 'negative'\n",
       "36. 'negative'\n",
       "37. 'negative'\n",
       "38. 'negative'\n",
       "39. 'negative'\n",
       "40. 'negative'\n",
       "41. 'negative'\n",
       "42. 'negative'\n",
       "43. 'negative'\n",
       "44. 'negative'\n",
       "45. 'negative'\n",
       "46. 'negative'\n",
       "47. 'negative'\n",
       "48. 'negative'\n",
       "49. 'negative'\n",
       "50. 'negative'\n",
       "51. 'negative'\n",
       "52. 'negative'\n",
       "53. 'negative'\n",
       "54. 'negative'\n",
       "55. 'negative'\n",
       "56. 'negative'\n",
       "57. 'negative'\n",
       "58. 'negative'\n",
       "59. 'negative'\n",
       "60. 'negative'\n",
       "61. 'negative'\n",
       "62. 'negative'\n",
       "63. 'negative'\n",
       "64. 'negative'\n",
       "65. 'positive'\n",
       "66. 'positive'\n",
       "67. 'positive'\n",
       "68. 'positive'\n",
       "69. 'positive'\n",
       "70. 'positive'\n",
       "71. 'positive'\n",
       "72. 'positive'\n",
       "73. 'positive'\n",
       "74. 'positive'\n",
       "75. 'positive'\n",
       "76. 'positive'\n",
       "77. 'positive'\n",
       "78. 'positive'\n",
       "79. 'positive'\n",
       "80. 'positive'\n",
       "81. 'positive'\n",
       "82. 'positive'\n",
       "83. 'positive'\n",
       "84. 'positive'\n",
       "85. 'positive'\n",
       "86. 'positive'\n",
       "87. 'positive'\n",
       "88. 'positive'\n",
       "89. 'positive'\n",
       "90. 'positive'\n",
       "91. 'negative'\n",
       "92. 'negative'\n",
       "93. 'negative'\n",
       "94. 'negative'\n",
       "95. 'negative'\n",
       "96. 'negative'\n",
       "97. 'negative'\n",
       "98. 'negative'\n",
       "99. 'negative'\n",
       "100. 'negative'\n",
       "101. 'negative'\n",
       "102. 'negative'\n",
       "103. 'negative'\n",
       "104. 'negative'\n",
       "105. 'positive'\n",
       "106. 'positive'\n",
       "107. 'negative'\n",
       "108. 'negative'\n",
       "109. 'positive'\n",
       "110. 'negative'\n",
       "111. 'positive'\n",
       "112. 'positive'\n",
       "113. 'negative'\n",
       "114. 'negative'\n",
       "115. 'positive'\n",
       "116. 'negative'\n",
       "117. 'negative'\n",
       "118. 'negative'\n",
       "119. 'negative'\n",
       "120. 'negative'\n",
       "121. 'negative'\n",
       "122. 'positive'\n",
       "123. 'positive'\n",
       "124. 'positive'\n",
       "125. 'positive'\n",
       "126. 'positive'\n",
       "127. 'negative'\n",
       "128. 'negative'\n",
       "129. 'positive'\n",
       "130. 'positive'\n",
       "131. 'negative'\n",
       "132. 'positive'\n",
       "133. 'negative'\n",
       "134. 'negative'\n",
       "135. 'positive'\n",
       "136. 'negative'\n",
       "137. 'negative'\n",
       "138. 'negative'\n",
       "139. 'negative'\n",
       "140. 'positive'\n",
       "141. 'negative'\n",
       "142. 'negative'\n",
       "143. 'negative'\n",
       "144. 'negative'\n",
       "145. 'negative'\n",
       "146. 'negative'\n",
       "147. 'negative'\n",
       "148. 'positive'\n",
       "149. 'negative'\n",
       "150. 'negative'\n",
       "151. 'negative'\n",
       "152. 'negative'\n",
       "153. 'negative'\n",
       "154. 'negative'\n",
       "155. 'negative'\n",
       "156. 'positive'\n",
       "157. 'negative'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"positive\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       "  [7] \"positive\" \"positive\" \"positive\" \"positive\" \"negative\" \"negative\"\n",
       " [13] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [19] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [25] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [31] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [37] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [43] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [49] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [55] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [61] \"negative\" \"negative\" \"negative\" \"negative\" \"positive\" \"positive\"\n",
       " [67] \"positive\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       " [73] \"positive\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       " [79] \"positive\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       " [85] \"positive\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       " [91] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       " [97] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       "[103] \"negative\" \"negative\" \"positive\" \"positive\" \"negative\" \"negative\"\n",
       "[109] \"positive\" \"negative\" \"positive\" \"positive\" \"negative\" \"negative\"\n",
       "[115] \"positive\" \"negative\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       "[121] \"negative\" \"positive\" \"positive\" \"positive\" \"positive\" \"positive\"\n",
       "[127] \"negative\" \"negative\" \"positive\" \"positive\" \"negative\" \"positive\"\n",
       "[133] \"negative\" \"negative\" \"positive\" \"negative\" \"negative\" \"negative\"\n",
       "[139] \"negative\" \"positive\" \"negative\" \"negative\" \"negative\" \"negative\"\n",
       "[145] \"negative\" \"negative\" \"negative\" \"positive\" \"negative\" \"negative\"\n",
       "[151] \"negative\" \"negative\" \"negative\" \"negative\" \"negative\" \"positive\"\n",
       "[157] \"negative\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "test.data$pheno.list$GSE39939$hiv.status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "         LTBI other_disease            TB \n",
       "           14            64            79 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE39939$status)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Format GSE39940"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           title       geo_accession                   status   \n",
       " WB_20_M_40012:  1   GSM981627:  1   Public on May 01 2014:334  \n",
       " WB_20_M_40182:  1   GSM981628:  1                              \n",
       " WB_20_M_40263:  1   GSM981629:  1                              \n",
       " WB_20_M_40376:  1   GSM981630:  1                              \n",
       " WB_20_M_40384:  1   GSM981631:  1                              \n",
       " WB_20_M_40385:  1   GSM981632:  1                              \n",
       " (Other)      :328   (Other)  :328                              \n",
       "    submission_date    last_update_date  type     channel_count\n",
       " Aug 07 2012:334    May 01 2014:334     RNA:334   Min.   :1    \n",
       "                                                  1st Qu.:1    \n",
       "                                                  Median :1    \n",
       "                                                  Mean   :1    \n",
       "                                                  3rd Qu.:1    \n",
       "                                                  Max.   :1    \n",
       "                                                               \n",
       "                                    source_name_ch1       organism_ch1\n",
       " Whole Blood from patient with Active TB    :111    Homo sapiens:334  \n",
       " Whole Blood from patient with LTBI         : 54                      \n",
       " Whole Blood from patient with other disease:169                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                                                                      \n",
       "                          characteristics_ch1\n",
       " disease status: active tuberculosis:111     \n",
       " disease status: latent TB infection: 54     \n",
       " disease status: other disease      :169     \n",
       "                                             \n",
       "                                             \n",
       "                                             \n",
       "                                             \n",
       "              characteristics_ch1.1                       characteristics_ch1.2\n",
       " hiv status: HIV negative:227       geographical region: Malawi      :177      \n",
       " hiv status: HIV positive:107       geographical region: South Africa:157      \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "                                                                               \n",
       "           characteristics_ch1.3    molecule_ch1\n",
       " barcode: 6116725094_A:  1       total RNA:334  \n",
       " barcode: 6116725094_B:  1                      \n",
       " barcode: 6116725094_C:  1                      \n",
       " barcode: 6116725094_D:  1                      \n",
       " barcode: 6116725094_E:  1                      \n",
       " barcode: 6116725094_F:  1                      \n",
       " (Other)              :328                      \n",
       "                                                    extract_protocol_ch1\n",
       " Standard PAXgene blood RNA extraction kit protocol (Qiagen).:334      \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "                                                                        \n",
       "  label_ch1  \n",
       " biotin:334  \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "             \n",
       "                                                                label_protocol_ch1\n",
       " Standard Illumina TotalPrep RNA Amplification kits (Applied Biosystems).:334     \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "                                                                                  \n",
       "   taxid_ch1                                       hyb_protocol\n",
       " Min.   :9606   Standard Illumina hybridisation protocol.:334  \n",
       " 1st Qu.:9606                                                  \n",
       " Median :9606                                                  \n",
       " Mean   :9606                                                  \n",
       " 3rd Qu.:9606                                                  \n",
       " Max.   :9606                                                  \n",
       "                                                               \n",
       "                              scan_protocol\n",
       " Standard Illumina scanning protocol.:334  \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                           \n",
       "                                                                                          data_processing\n",
       " The data were normalised using Genome Studio, with background subtracted and quantile normalised.:334   \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "                                                                                                         \n",
       "   platform_id            contact_name                 contact_email\n",
       " GPL10558:334   Victoria,,Wright:334   v.wright@imperial.ac.uk:334  \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                                                                    \n",
       "                contact_laboratory   contact_department\n",
       " Wright Fleming Institute:334      Paediatrics:334     \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "                                                       \n",
       "               contact_institute      contact_address contact_city\n",
       " Imperial College London:334     Norfolk Place:334    London:334  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       "                                                                  \n",
       " contact_zip.postal_code       contact_country supplementary_file\n",
       " W2 1PG:334              United Kingdom:334    NONE:334          \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       "                                                                 \n",
       " data_row_count        barcode.ch1            disease.status.ch1\n",
       " Min.   :47323   6116725094_A:  1   active tuberculosis:111     \n",
       " 1st Qu.:47323   6116725094_B:  1   latent TB infection: 54     \n",
       " Median :47323   6116725094_C:  1   other disease      :169     \n",
       " Mean   :47323   6116725094_D:  1                               \n",
       " 3rd Qu.:47323   6116725094_E:  1                               \n",
       " Max.   :47323   6116725094_F:  1                               \n",
       "                 (Other)     :328                               \n",
       " geographical.region.ch1      hiv.status.ch1\n",
       " Malawi      :177        HIV negative:227   \n",
       " South Africa:157        HIV positive:107   \n",
       "                                            \n",
       "                                            \n",
       "                                            \n",
       "                                            \n",
       "                                            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "summary(test.data$pheno.list$GSE39940)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE39940$status = as.character(test.data$pheno.list$GSE39940$disease.status.ch1)\n",
    "test.data$pheno.list$GSE39940$status = ifelse(test.data$pheno.list$GSE39940$status == \"active tuberculosis\", \"TB\",\n",
    "                                             ifelse(test.data$pheno.list$GSE39940$status == \"other disease\", \"other_disease\", \"LTBI\"))\n",
    "\n",
    "test.data$pheno.list$GSE39940$hiv.status = as.character(test.data$pheno.list$GSE39940$hiv.status.ch1)\n",
    "test.data$pheno.list$GSE39940$hiv.status = ifelse(test.data$pheno.list$GSE39940$hiv.status == \"HIV negative\", \"negative\", \"positive\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "         LTBI other_disease            TB \n",
       "           54           169           111 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE39940$status)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "               \n",
       "                negative positive\n",
       "  LTBI                54        0\n",
       "  other_disease      103       66\n",
       "  TB                  70       41"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(test.data$pheno.list$GSE39940$status, test.data$pheno.list$GSE39940$hiv.status)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Review that the phenotype formatting is correct and save to csv files (status and GSM identifier)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Review phenotype data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "$GSE19491\n",
       "\n",
       "control    LTBI    PSLE     SLE   Staph   Still   Strep      TB \n",
       "    133      69      82      28      40      31      12      89 \n",
       "\n",
       "$GSE37250\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "          167           175           195 \n",
       "\n",
       "$GSE39939\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "           14            64            79 \n",
       "\n",
       "$GSE39940\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "           54           169           111 \n",
       "\n",
       "$GSE42825\n",
       "\n",
       "    control sarcoidosis          TB \n",
       "         23          11           8 \n",
       "\n",
       "$GSE42826\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         52           8           6          25          11 \n",
       "\n",
       "$GSE42827\n",
       "\n",
       "pneumonia \n",
       "       10 \n",
       "\n",
       "$GSE42830\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         38           8           8          25          16 \n",
       "\n",
       "$GSE42831\n",
       "\n",
       "sarcoidosis \n",
       "          7 \n",
       "\n",
       "$GSE42832\n",
       "\n",
       "    control sarcoidosis          TB \n",
       "         30          30          30 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$pheno.list, function(x) {table(x$status)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "539"
      ],
      "text/latex": [
       "539"
      ],
      "text/markdown": [
       "539"
      ],
      "text/plain": [
       "[1] 539"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Reduce(\"+\", lapply(test.data$pheno.list, function(x) {sum(x$status == \"TB\")}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$GSE19491</dt>\n",
       "\t\t<dd>89</dd>\n",
       "\t<dt>$GSE37250</dt>\n",
       "\t\t<dd>195</dd>\n",
       "\t<dt>$GSE39939</dt>\n",
       "\t\t<dd>79</dd>\n",
       "\t<dt>$GSE39940</dt>\n",
       "\t\t<dd>111</dd>\n",
       "\t<dt>$GSE42825</dt>\n",
       "\t\t<dd>8</dd>\n",
       "\t<dt>$GSE42826</dt>\n",
       "\t\t<dd>11</dd>\n",
       "\t<dt>$GSE42827</dt>\n",
       "\t\t<dd>0</dd>\n",
       "\t<dt>$GSE42830</dt>\n",
       "\t\t<dd>16</dd>\n",
       "\t<dt>$GSE42831</dt>\n",
       "\t\t<dd>0</dd>\n",
       "\t<dt>$GSE42832</dt>\n",
       "\t\t<dd>30</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$GSE19491] 89\n",
       "\\item[\\$GSE37250] 195\n",
       "\\item[\\$GSE39939] 79\n",
       "\\item[\\$GSE39940] 111\n",
       "\\item[\\$GSE42825] 8\n",
       "\\item[\\$GSE42826] 11\n",
       "\\item[\\$GSE42827] 0\n",
       "\\item[\\$GSE42830] 16\n",
       "\\item[\\$GSE42831] 0\n",
       "\\item[\\$GSE42832] 30\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$GSE19491\n",
       ":   89\n",
       "$GSE37250\n",
       ":   195\n",
       "$GSE39939\n",
       ":   79\n",
       "$GSE39940\n",
       ":   111\n",
       "$GSE42825\n",
       ":   8\n",
       "$GSE42826\n",
       ":   11\n",
       "$GSE42827\n",
       ":   0\n",
       "$GSE42830\n",
       ":   16\n",
       "$GSE42831\n",
       ":   0\n",
       "$GSE42832\n",
       ":   30\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$GSE19491\n",
       "[1] 89\n",
       "\n",
       "$GSE37250\n",
       "[1] 195\n",
       "\n",
       "$GSE39939\n",
       "[1] 79\n",
       "\n",
       "$GSE39940\n",
       "[1] 111\n",
       "\n",
       "$GSE42825\n",
       "[1] 8\n",
       "\n",
       "$GSE42826\n",
       "[1] 11\n",
       "\n",
       "$GSE42827\n",
       "[1] 0\n",
       "\n",
       "$GSE42830\n",
       "[1] 16\n",
       "\n",
       "$GSE42831\n",
       "[1] 0\n",
       "\n",
       "$GSE42832\n",
       "[1] 30\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$pheno.list, function(x) {sum(x$status == \"TB\")})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "$GSE19491\n",
       "\n",
       "control    LTBI    PSLE     SLE   Staph   Still   Strep      TB \n",
       "    133      69      82      28      40      31      12      89 \n",
       "\n",
       "$GSE37250\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "          167           175           195 \n",
       "\n",
       "$GSE39939\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "           14            64            79 \n",
       "\n",
       "$GSE39940\n",
       "\n",
       "         LTBI other_disease            TB \n",
       "           54           169           111 \n",
       "\n",
       "$GSE42826\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         52           8           6          25          11 \n",
       "\n",
       "$GSE42827\n",
       "\n",
       "pneumonia \n",
       "       10 \n",
       "\n",
       "$GSE42830\n",
       "\n",
       "    control lung_cancer   pneumonia sarcoidosis          TB \n",
       "         38           8           8          25          16 \n",
       "\n",
       "$GSE42831\n",
       "\n",
       "sarcoidosis \n",
       "          7 \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$pheno.list, function(x) {table(x$status)})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### Combine phenotype data and GSM's across all tables, excluding GSE42825 and GSE42832 based on fewer number of genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$pheno.list$GSE42825 = NULL\n",
    "test.data$pheno.list$GSE42832 = NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.pheno = data.frame()\n",
    "for (GSE in test.data$pheno.list) {\n",
    "    test.pheno = rbind(test.pheno, GSE[,c(\"status\", \"hiv.status\"), drop=F])\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1726</li>\n",
       "\t<li>2</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1726\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1726\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1726    2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "               hiv.status\n",
       "status          negative positive\n",
       "  control            223        0\n",
       "  LTBI               220       84\n",
       "  lung_cancer         16        0\n",
       "  other_disease      223      185\n",
       "  pneumonia           24        0\n",
       "  PSLE                82        0\n",
       "  sarcoidosis         57        0\n",
       "  SLE                 28        0\n",
       "  Staph               40        0\n",
       "  Still               31        0\n",
       "  Strep               12        0\n",
       "  TB                 335      166"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(test.pheno)\n",
    "table(test.pheno)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "write.csv(test.pheno, \"data/Illumina_microarray_GSE_validation/Illumina_microarray_validation_pheno.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## Analyze the number of genes in each microarray and determine how to do overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$GSE19491</dt>\n",
       "\t\t<dd>15836</dd>\n",
       "\t<dt>$GSE37250</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE39939</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE39940</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42825</dt>\n",
       "\t\t<dd>11147</dd>\n",
       "\t<dt>$GSE42826</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42827</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42830</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42831</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42832</dt>\n",
       "\t\t<dd>12296</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$GSE19491] 15836\n",
       "\\item[\\$GSE37250] 17879\n",
       "\\item[\\$GSE39939] 17879\n",
       "\\item[\\$GSE39940] 17879\n",
       "\\item[\\$GSE42825] 11147\n",
       "\\item[\\$GSE42826] 17879\n",
       "\\item[\\$GSE42827] 17879\n",
       "\\item[\\$GSE42830] 17879\n",
       "\\item[\\$GSE42831] 17879\n",
       "\\item[\\$GSE42832] 12296\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$GSE19491\n",
       ":   15836\n",
       "$GSE37250\n",
       ":   17879\n",
       "$GSE39939\n",
       ":   17879\n",
       "$GSE39940\n",
       ":   17879\n",
       "$GSE42825\n",
       ":   11147\n",
       "$GSE42826\n",
       ":   17879\n",
       "$GSE42827\n",
       ":   17879\n",
       "$GSE42830\n",
       ":   17879\n",
       "$GSE42831\n",
       ":   17879\n",
       "$GSE42832\n",
       ":   12296\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$GSE19491\n",
       "[1] 15836\n",
       "\n",
       "$GSE37250\n",
       "[1] 17879\n",
       "\n",
       "$GSE39939\n",
       "[1] 17879\n",
       "\n",
       "$GSE39940\n",
       "[1] 17879\n",
       "\n",
       "$GSE42825\n",
       "[1] 11147\n",
       "\n",
       "$GSE42826\n",
       "[1] 17879\n",
       "\n",
       "$GSE42827\n",
       "[1] 17879\n",
       "\n",
       "$GSE42830\n",
       "[1] 17879\n",
       "\n",
       "$GSE42831\n",
       "[1] 17879\n",
       "\n",
       "$GSE42832\n",
       "[1] 12296\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$exprs, function(x) {dim(x)[1]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$GSE19491</dt>\n",
       "\t\t<dd>15836</dd>\n",
       "\t<dt>$GSE37250</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE39939</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE39940</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42826</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42827</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42830</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "\t<dt>$GSE42831</dt>\n",
       "\t\t<dd>17879</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$GSE19491] 15836\n",
       "\\item[\\$GSE37250] 17879\n",
       "\\item[\\$GSE39939] 17879\n",
       "\\item[\\$GSE39940] 17879\n",
       "\\item[\\$GSE42826] 17879\n",
       "\\item[\\$GSE42827] 17879\n",
       "\\item[\\$GSE42830] 17879\n",
       "\\item[\\$GSE42831] 17879\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$GSE19491\n",
       ":   15836\n",
       "$GSE37250\n",
       ":   17879\n",
       "$GSE39939\n",
       ":   17879\n",
       "$GSE39940\n",
       ":   17879\n",
       "$GSE42826\n",
       ":   17879\n",
       "$GSE42827\n",
       ":   17879\n",
       "$GSE42830\n",
       ":   17879\n",
       "$GSE42831\n",
       ":   17879\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$GSE19491\n",
       "[1] 15836\n",
       "\n",
       "$GSE37250\n",
       "[1] 17879\n",
       "\n",
       "$GSE39939\n",
       "[1] 17879\n",
       "\n",
       "$GSE39940\n",
       "[1] 17879\n",
       "\n",
       "$GSE42826\n",
       "[1] 17879\n",
       "\n",
       "$GSE42827\n",
       "[1] 17879\n",
       "\n",
       "$GSE42830\n",
       "[1] 17879\n",
       "\n",
       "$GSE42831\n",
       "[1] 17879\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$exprs, function(x) {dim(x)[1]})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### Exclude GSE42832 and GSE42825 based on smaller number of genes. Keep 19491 as common denominator since these were in my RNA-seq validation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$exprs$GSE42825 = NULL\n",
    "test.data$exprs$GSE42832 = NULL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "## Merge the expression datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "15821"
      ],
      "text/latex": [
       "15821"
      ],
      "text/markdown": [
       "15821"
      ],
      "text/plain": [
       "[1] 15821"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'EEF1A1'</li>\n",
       "\t<li>'GAPDH'</li>\n",
       "\t<li>'SLC35E2'</li>\n",
       "\t<li>'DUSP22'</li>\n",
       "\t<li>'RPS28'</li>\n",
       "\t<li>'IPO13'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'EEF1A1'\n",
       "\\item 'GAPDH'\n",
       "\\item 'SLC35E2'\n",
       "\\item 'DUSP22'\n",
       "\\item 'RPS28'\n",
       "\\item 'IPO13'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'EEF1A1'\n",
       "2. 'GAPDH'\n",
       "3. 'SLC35E2'\n",
       "4. 'DUSP22'\n",
       "5. 'RPS28'\n",
       "6. 'IPO13'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"EEF1A1\"  \"GAPDH\"   \"SLC35E2\" \"DUSP22\"  \"RPS28\"   \"IPO13\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genes.to.keep = intersect(rownames(test.data$exprs$GSE19491), rownames(test.data$exprs$GSE42830))\n",
    "length(genes.to.keep)\n",
    "head(genes.to.keep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "merge.exprs = function(datasets, seed.data) {\n",
    "    begin = datasets[[seed.data]]\n",
    "    remaining = setdiff(names(datasets), seed.data)\n",
    "    print(remaining)\n",
    "    for (GSE in remaining) {\n",
    "        begin = merge(begin, datasets[[GSE]], by=\"row.names\", all.x=T)#, suffixes = c(\"\",\"\"))\n",
    "        rownames(begin) = begin$Row.names\n",
    "        begin$Row.names = NULL\n",
    "        \n",
    "    }\n",
    "    return(begin)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.data$exprs.fil = lapply(test.data$exprs, function(x) {x[rownames(x) %in% genes.to.keep, ]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$GSE19491</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE37250</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE39939</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE39940</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE42826</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE42827</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE42830</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "\t<dt>$GSE42831</dt>\n",
       "\t\t<dd>15821</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$GSE19491] 15821\n",
       "\\item[\\$GSE37250] 15821\n",
       "\\item[\\$GSE39939] 15821\n",
       "\\item[\\$GSE39940] 15821\n",
       "\\item[\\$GSE42826] 15821\n",
       "\\item[\\$GSE42827] 15821\n",
       "\\item[\\$GSE42830] 15821\n",
       "\\item[\\$GSE42831] 15821\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$GSE19491\n",
       ":   15821\n",
       "$GSE37250\n",
       ":   15821\n",
       "$GSE39939\n",
       ":   15821\n",
       "$GSE39940\n",
       ":   15821\n",
       "$GSE42826\n",
       ":   15821\n",
       "$GSE42827\n",
       ":   15821\n",
       "$GSE42830\n",
       ":   15821\n",
       "$GSE42831\n",
       ":   15821\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$GSE19491\n",
       "[1] 15821\n",
       "\n",
       "$GSE37250\n",
       "[1] 15821\n",
       "\n",
       "$GSE39939\n",
       "[1] 15821\n",
       "\n",
       "$GSE39940\n",
       "[1] 15821\n",
       "\n",
       "$GSE42826\n",
       "[1] 15821\n",
       "\n",
       "$GSE42827\n",
       "[1] 15821\n",
       "\n",
       "$GSE42830\n",
       "[1] 15821\n",
       "\n",
       "$GSE42831\n",
       "[1] 15821\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(test.data$exprs.fil, function(x) {dim(x)[1]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"GSE37250\" \"GSE39939\" \"GSE39940\" \"GSE42826\" \"GSE42827\" \"GSE42830\" \"GSE42831\"\n"
     ]
    }
   ],
   "source": [
    "test.merge = merge.exprs(test.data$exprs.fil, \"GSE19491\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1726"
      ],
      "text/latex": [
       "1726"
      ],
      "text/markdown": [
       "1726"
      ],
      "text/plain": [
       "[1] 1726"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(intersect(rownames(test.pheno), colnames(test.merge)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "test.merge = test.merge[, colnames(test.merge) %in% rownames(test.pheno)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "identical(rownames(test.pheno), colnames(test.merge))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>1726</li>\n",
       "\t<li>2</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 1726\n",
       "\\item 2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 1726\n",
       "2. 2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 1726    2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>15821</li>\n",
       "\t<li>1726</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 15821\n",
       "\\item 1726\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 15821\n",
       "2. 1726\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 15821  1726"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(test.pheno)\n",
    "dim(test.merge)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "write.csv(test.merge, \"data/Illumina_microarray_GSE_validation/Illumina_microarray_validation_exprs.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
